









  
  
  
  
  



<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="shortcut icon" href="/img/igraph4.png" type="image/x-icon" />

    <title>python-igraph manual</title>

    <!-- Bootstrap core CSS -->
    <link href="/bootstrap/css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom styles for this template -->
    <link href='http://fonts.googleapis.com/css?family=Lato:300,400,700'
          rel='stylesheet' type='text/css'>
    <link href="http://netdna.bootstrapcdn.com/font-awesome/3.2.1/css/font-awesome.css" rel="stylesheet">
    <link rel="stylesheet" href="/fonts/font-mfizz/font-mfizz.css">
    <link href="/css/affix.css" rel="stylesheet">
    <link href="/css/syntax.css" rel="stylesheet">
    <link href="/css/other.css" rel="stylesheet">
    <link href="/fonts/fonts.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements 
         and media queries -->
    <!--[if lt IE 9]>
      <script src="/js/html5shiv.js"></script>
      <script src="/js/respond.min.js"></script>
    <![endif]-->

    <link href="/css/epydoc.css" rel="stylesheet">

  </head>

  <body class="python">
  <div id="wrap">

    <div class="navbar navbar-inverse navbar-fixed-top">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle"
                  data-toggle="collapse" data-target=".navbar-collapse">
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="navbar-brand" href="/">
            <img src="/img/igraph4.svg" width=35> igraph
          </a>
        </div>
        <div class="navbar-collapse collapse">
	  
	  <ul class="nav navbar-nav">
	    <li><a href="/python">| python-igraph</a></li>
	  </ul>
	  
          <ul class="nav navbar-nav navbar-right">
            <li class="dropdown">
              <a href="#" class="dropdown-toggle" data-toggle="dropdown"
		 role="button" aria-expanded="false">Products
		<span class="caret"></span></a>
              <ul class="dropdown-menu" role="menu">
		<li>
		  <a href="/r"><i class="icon-fixed-width icon-rlogo"></i>
		    igraph R package</a></li>
		<li>
		  <a href="/python"><i class="icon-fixed-width icon-python"></i>
		    python-igraph</a></li>
		<li>
		  <a href="/c"><i class="icon-fixed-width icon-c"></i>
		    igraph C library</a></li>
		<li class="divider"></li>
		<li>
		  <a href="https://github.com/igraph/igraphdata">
		    <i class="icon-fixed-width icon-rlogo"></i>
		    igraphdata R package <i class="fa fa-github"></i></a></li>
		<li>
		  <a href="http://nexus.igraph.org">
		    <i class="fa fa-times"></i>Nexus network repository</a></li>
              </ul>
            </li>
            <li>
              <a href="/news.html"><i class="icon-rss"></i>
		News</a></li>
	    <li><a href="https://github.com/igraph">
	      <i class="icon-github"></i> On github</a></li>
          </ul>
          <!-- <form class="navbar-form navbar-right"> -->
          <!--   <div class="form-group"> -->
          <!--     <input type="text" placeholder="Search docs" -->
	  <!-- 	     class="form-control" size="15"> -->
          <!--   </div> -->
          <!--   <button type="submit" class="btn btn-success"> -->
          <!--     <span class="glyphicon glyphicon-search"></span></button> -->
          <!-- </form> -->
        </div><!--/.navbar-collapse -->
      </div>
    </div>
    
    
    
    <div class="bs-header-outer"><div class="bs-header">
      <div class="container">
        <h1>python-igraph manual</h1>
        <p class="lead">For using igraph from Python</p>        
      </div>
    </div></div>

    

    <div class="container bs-docs-container">
<div class="row">
  <div class="col-md-9 col-md-offset-1">
  <div class="bs-docs-section">

    
<div bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080"
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="epynavbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="igraph-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="epynavbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="epynavbar" align="center"
            ></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="igraph-module.html">Package&nbsp;igraph</a> ::
        Class&nbsp;GraphBase
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            ></span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class GraphBase</h1><p class="nomargin-top"><span class="codelink"><a href="igraph-pysrc.html#GraphBase">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">GraphBase</strong>
</pre>

<dl><dt>Known Subclasses:</dt>
<dd>
      <ul class="subclass-list">
<li><a href="igraph.Graph-class.html">Graph</a></li>  </ul>
</dd></dl>

<hr />
<p>Low-level representation of a graph.</p>
  <p>Don't use it directly, use <a href="igraph.Graph-class.html" 
  class="link">igraph.Graph</a> instead.</p>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Adjacency" class="summary-sig-name">Adjacency</a>(<span class="summary-sig-arg">matrix</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ADJ_DIRECTED</span>)</span><br />
      Generates a graph from its adjacency matrix.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Adjacency">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Asymmetric_Preference" class="summary-sig-name">Asymmetric_Preference</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">type_dist_matrix</span>,
        <span class="summary-sig-arg">pref_matrix</span>,
        <span class="summary-sig-arg">attribute</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a graph based on asymmetric vertex types and connection 
      probabilities.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Asymmetric_Preference">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Atlas" class="summary-sig-name">Atlas</a>(<span class="summary-sig-arg">idx</span>)</span><br />
      Generates a graph from the Graph Atlas.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Atlas">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Barabasi" class="summary-sig-name">Barabasi</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">m</span>,
        <span class="summary-sig-arg">outpref</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">power</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">zero_appeal</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">implementation</span>=<span class="summary-sig-default">&quot;psumtree&quot;</span>,
        <span class="summary-sig-arg">start_from</span>=<span class="summary-sig-default">None</span>)</span><br />
      Generates a graph based on the Barabasi-Albert model.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Barabasi">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#De_Bruijn" class="summary-sig-name">De_Bruijn</a>(<span class="summary-sig-arg">m</span>,
        <span class="summary-sig-arg">n</span>)</span><br />
      Generates a de Bruijn graph with parameters (m, n)</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.De_Bruijn">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Degree_Sequence" class="summary-sig-name">Degree_Sequence</a>(<span class="summary-sig-arg">out</span>,
        <span class="summary-sig-arg">in</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">method</span>=<span class="summary-sig-default">&quot;simple&quot;</span>)</span><br />
      Generates a graph with a given degree sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Degree_Sequence">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Erdos_Renyi" class="summary-sig-name">Erdos_Renyi</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">p</span>,
        <span class="summary-sig-arg">m</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a graph based on the Erdos-Renyi model.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Erdos_Renyi">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Establishment" class="summary-sig-name">Establishment</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">k</span>,
        <span class="summary-sig-arg">type_dist</span>,
        <span class="summary-sig-arg">pref_matrix</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a graph based on a simple growing model with vertex types.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Establishment">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Famous" class="summary-sig-name">Famous</a>(<span class="summary-sig-arg">name</span>)</span><br />
      Generates a famous graph based on its name.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Famous">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Forest_Fire" class="summary-sig-name">Forest_Fire</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">fw_prob</span>,
        <span class="summary-sig-arg">bw_factor</span>=<span class="summary-sig-default">0.0</span>,
        <span class="summary-sig-arg">ambs</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a graph based on the forest fire model</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Forest_Fire">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Full" class="summary-sig-name">Full</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a full graph (directed or undirected, with or without 
      loops).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Full">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Full_Citation" class="summary-sig-name">Full_Citation</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a full citation graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Full_Citation">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Growing_Random" class="summary-sig-name">Growing_Random</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">m</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">citation</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a growing random graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Growing_Random">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Isoclass" class="summary-sig-name">Isoclass</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">class</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a graph with a given isomorphy class.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Isoclass">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#K_Regular" class="summary-sig-name">K_Regular</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">k</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">multiple</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a k-regular random graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.K_Regular">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Kautz" class="summary-sig-name">Kautz</a>(<span class="summary-sig-arg">m</span>,
        <span class="summary-sig-arg">n</span>)</span><br />
      Generates a Kautz graph with parameters (m, n)</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Kautz">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#LCF" class="summary-sig-name">LCF</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">shifts</span>,
        <span class="summary-sig-arg">repeats</span>)</span><br />
      Generates a graph from LCF notation.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.LCF">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Lattice" class="summary-sig-name">Lattice</a>(<span class="summary-sig-arg">dim</span>,
        <span class="summary-sig-arg">nei</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">mutual</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">circular</span>=<span class="summary-sig-default">True</span>)</span><br />
      Generates a regular lattice.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Lattice">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Preference" class="summary-sig-name">Preference</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">type_dist</span>,
        <span class="summary-sig-arg">pref_matrix</span>,
        <span class="summary-sig-arg">attribute</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a graph based on vertex types and connection probabilities.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Preference">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_DIMACS" class="summary-sig-name">Read_DIMACS</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>)</span><br />
      Reads a graph from a file conforming to the DIMACS minimum-cost flow 
      file format.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_DIMACS">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_DL" class="summary-sig-name">Read_DL</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>)</span><br />
      Reads an UCINET DL file and creates a graph based on it.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_DL">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_Edgelist" class="summary-sig-name">Read_Edgelist</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>)</span><br />
      Reads an edge list from a file and creates a graph based on it.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_Edgelist">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_GML" class="summary-sig-name">Read_GML</a>(<span class="summary-sig-arg">f</span>)</span><br />
      Reads a GML file and creates a graph based on it.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_GML">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_GraphDB" class="summary-sig-name">Read_GraphDB</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>)</span><br />
      Reads a GraphDB format file and creates a graph based on it.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_GraphDB">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_GraphML" class="summary-sig-name">Read_GraphML</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">index</span>=<span class="summary-sig-default">0</span>)</span><br />
      Reads a GraphML format file and creates a graph based on it.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_GraphML">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_Lgl" class="summary-sig-name">Read_Lgl</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">names</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">&quot;if_present&quot;</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>)</span><br />
      Reads an .lgl file used by LGL.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_Lgl">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_Ncol" class="summary-sig-name">Read_Ncol</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">names</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">&quot;if_present&quot;</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>)</span><br />
      Reads an .ncol file used by LGL.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_Ncol">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Read_Pajek" class="summary-sig-name">Read_Pajek</a>(<span class="summary-sig-arg">f</span>)</span><br />
      Reads a Pajek format file and creates a graph based on it.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_Pajek">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Recent_Degree" class="summary-sig-name">Recent_Degree</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">m</span>,
        <span class="summary-sig-arg">window</span>,
        <span class="summary-sig-arg">outpref</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">power</span>=<span class="summary-sig-default">1</span>)</span><br />
      Generates a graph based on a stochastic model where the probability 
      of an edge gaining a new node is proportional to the edges gained in 
      a given time window.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Recent_Degree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Ring" class="summary-sig-name">Ring</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">mutual</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">circular</span>=<span class="summary-sig-default">True</span>)</span><br />
      Generates a ring graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Ring">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#SBM" class="summary-sig-name">SBM</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">pref_matrix</span>,
        <span class="summary-sig-arg">block_sizes</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a graph based on a stochastic blockmodel.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.SBM">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Star" class="summary-sig-name">Star</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;undirected&quot;</span>,
        <span class="summary-sig-arg">center</span>=<span class="summary-sig-default">0</span>)</span><br />
      Generates a star graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Star">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Static_Fitness" class="summary-sig-name">Static_Fitness</a>(<span class="summary-sig-arg">m</span>,
        <span class="summary-sig-arg">fitness_out</span>,
        <span class="summary-sig-arg">fitness_in</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">multiple</span>=<span class="summary-sig-default">False</span>)</span><br />
      Generates a non-growing graph with edge probabilities proportional to
      node fitnesses.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Static_Fitness">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Static_Power_Law" class="summary-sig-name">Static_Power_Law</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">m</span>,
        <span class="summary-sig-arg">exponent_out</span>,
        <span class="summary-sig-arg">exponent_in</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">multiple</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">finite_size_correction</span>=<span class="summary-sig-default">True</span>)</span><br />
      Generates a non-growing graph with prescribed power-law degree 
      distributions.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Static_Power_Law">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Tree" class="summary-sig-name">Tree</a>(<span class="summary-sig-arg">n</span>,
        <span class="summary-sig-arg">children</span>,
        <span class="summary-sig-arg">type</span>=<span class="summary-sig-default">TREE_UNDIRECTED</span>)</span><br />
      Generates a tree in which almost all vertices have the same number of
      children.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Tree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Watts_Strogatz" class="summary-sig-name">Watts_Strogatz</a>(<span class="summary-sig-arg">dim</span>,
        <span class="summary-sig-arg">size</span>,
        <span class="summary-sig-arg">nei</span>,
        <span class="summary-sig-arg">p</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">multiple</span>=<span class="summary-sig-default">False</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Watts_Strogatz">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#Weighted_Adjacency" class="summary-sig-name">Weighted_Adjacency</a>(<span class="summary-sig-arg">matrix</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ADJ_DIRECTED</span>,
        <span class="summary-sig-arg">attr</span>=<span class="summary-sig-default">&quot;weight&quot;</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">True</span>)</span><br />
      Generates a graph from its adjacency matrix.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.Weighted_Adjacency">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#_Random_Bipartite" class="summary-sig-name" onclick="show_private();">_Random_Bipartite</a>(<span class="summary-sig-arg">n1</span>,
        <span class="summary-sig-arg">n2</span>,
        <span class="summary-sig-arg">p</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">m</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">neimode</span>=<span class="summary-sig-default">&quot;all&quot;</span>)</span><br />
      Internal function, undocumented.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase._Random_Bipartite">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__and__"></a><span class="summary-sig-name">__and__</span>(<span class="summary-sig-arg">x</span>,
        <span class="summary-sig-arg">y</span>)</span><br />
      x&amp;y</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__and__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__delitem__"></a><span class="summary-sig-name">__delitem__</span>(<span class="summary-sig-arg">x</span>,
        <span class="summary-sig-arg">y</span>)</span><br />
      del x[y]</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__delitem__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__getitem__"></a><span class="summary-sig-name">__getitem__</span>(<span class="summary-sig-arg">x</span>,
        <span class="summary-sig-arg">y</span>)</span><br />
      x[y]</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__getitem__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#__graph_as_cobject" class="summary-sig-name" onclick="show_private();">__graph_as_cobject</a>()</span><br />
      Returns the igraph graph encapsulated by the Python object as a 
      PyCObject</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__graph_as_cobject">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">...</span>)</span><br />
      x.__init__(...) initializes x; see help(type(x)) for signature</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__invert__"></a><span class="summary-sig-name">__invert__</span>(<span class="summary-sig-arg">x</span>)</span><br />
      ~x</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__invert__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">a new object with type S, a subtype of T</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#__new__" class="summary-sig-name">__new__</a>(<span class="summary-sig-arg">T</span>,
        <span class="summary-sig-arg">S</span>,
        <span class="summary-sig-arg">...</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__new__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__or__"></a><span class="summary-sig-name">__or__</span>(<span class="summary-sig-arg">x</span>,
        <span class="summary-sig-arg">y</span>)</span><br />
      x|y</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__or__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__rand__"></a><span class="summary-sig-name">__rand__</span>(<span class="summary-sig-arg">x</span>,
        <span class="summary-sig-arg">y</span>)</span><br />
      y&amp;x</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__rand__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#__register_destructor" class="summary-sig-name" onclick="show_private();">__register_destructor</a>(<span class="summary-sig-arg">destructor</span>)</span><br />
      Registers a destructor to be called when the object is freed by 
      Python.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__register_destructor">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__ror__"></a><span class="summary-sig-name">__ror__</span>(<span class="summary-sig-arg">x</span>,
        <span class="summary-sig-arg">y</span>)</span><br />
      y|x</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__ror__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="__setitem__"></a><span class="summary-sig-name">__setitem__</span>(<span class="summary-sig-arg">x</span>,
        <span class="summary-sig-arg">i</span>,
        <span class="summary-sig-arg">y</span>)</span><br />
      x[i]=y</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__setitem__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#__str__" class="summary-sig-name">__str__</a>(<span class="summary-sig-arg">x</span>)</span><br />
      str(x)</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.__str__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#add_edges" class="summary-sig-name">add_edges</a>(<span class="summary-sig-arg">es</span>)</span><br />
      Adds edges to the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.add_edges">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#add_vertices" class="summary-sig-name">add_vertices</a>(<span class="summary-sig-arg">n</span>)</span><br />
      Adds vertices to the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.add_vertices">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#all_minimal_st_separators" class="summary-sig-name">all_minimal_st_separators</a>()</span><br />
      Returns a list containing all the minimal s-t separators of a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.all_minimal_st_separators">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#all_st_cuts" class="summary-sig-name">all_st_cuts</a>(<span class="summary-sig-arg">source</span>,
        <span class="summary-sig-arg">target</span>)</span><br />
      Returns all the cuts between the source and target vertices in a 
      directed graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.all_st_cuts">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#all_st_mincuts" class="summary-sig-name">all_st_mincuts</a>(<span class="summary-sig-arg">source</span>,
        <span class="summary-sig-arg">target</span>)</span><br />
      Returns all minimum cuts between the source and target vertices in a 
      directed graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.all_st_mincuts">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#are_connected" class="summary-sig-name">are_connected</a>(<span class="summary-sig-arg">v1</span>,
        <span class="summary-sig-arg">v2</span>)</span><br />
      Decides whether two given vertices are directly connected.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.are_connected">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#articulation_points" class="summary-sig-name">articulation_points</a>()</span><br />
      Returns the list of articulation points in the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.cut_vertices">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#assortativity" class="summary-sig-name">assortativity</a>(<span class="summary-sig-arg">types1</span>,
        <span class="summary-sig-arg">types2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>)</span><br />
      Returns the assortativity of the graph based on numeric properties of
      the vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.assortativity">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#assortativity_degree" class="summary-sig-name">assortativity_degree</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>)</span><br />
      Returns the assortativity of a graph based on vertex degrees.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.assortativity_degree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#assortativity_nominal" class="summary-sig-name">assortativity_nominal</a>(<span class="summary-sig-arg">types</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>)</span><br />
      Returns the assortativity of the graph based on vertex categories.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.assortativity_nominal">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#attributes" class="summary-sig-name">attributes</a>()</span><br />
      Returns:
      the attribute name list of the graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.attributes">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#authority_score" class="summary-sig-name">authority_score</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">scale</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">arpack_options</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">return_eigenvalue</span>=<span class="summary-sig-default">False</span>)</span><br />
      Calculates Kleinberg's authority score for the vertices of the graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.authority_score">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#average_path_length" class="summary-sig-name">average_path_length</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">unconn</span>=<span class="summary-sig-default">True</span>)</span><br />
      Calculates the average path length in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.average_path_length">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#betweenness" class="summary-sig-name">betweenness</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">cutoff</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">nobigint</span>=<span class="summary-sig-default">True</span>)</span><br />
      Calculates or estimates the betweenness of vertices in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.betweenness">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#bfs" class="summary-sig-name">bfs</a>(<span class="summary-sig-arg">vid</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>)</span><br />
      Conducts a breadth first search (BFS) on the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.bfs">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#bfsiter" class="summary-sig-name">bfsiter</a>(<span class="summary-sig-arg">vid</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>,
        <span class="summary-sig-arg">advanced</span>=<span class="summary-sig-default">False</span>)</span><br />
      Constructs a breadth first search (BFS) iterator of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.bfsiter">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#bibcoupling" class="summary-sig-name">bibcoupling</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates bibliographic coupling scores for given vertices in a 
      graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.bibcoupling">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#biconnected_components" class="summary-sig-name">biconnected_components</a>(<span class="summary-sig-arg">return_articulation_points</span>=<span class="summary-sig-default">True</span>)</span><br />
      Calculates the biconnected components of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.biconnected_components">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#bipartite_projection" class="summary-sig-name">bipartite_projection</a>(<span class="summary-sig-arg">types</span>,
        <span class="summary-sig-arg">multiplicity</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">probe1</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">which</span>=<span class="summary-sig-default">-1</span>)</span><br />
      Internal function, undocumented.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.bipartite_projection">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#bipartite_projection_size" class="summary-sig-name">bipartite_projection_size</a>(<span class="summary-sig-arg">types</span>)</span><br />
      Internal function, undocumented.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.bipartite_projection_size">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#canonical_permutation" class="summary-sig-name">canonical_permutation</a>(<span class="summary-sig-arg">sh</span>=<span class="summary-sig-default">&quot;fm&quot;</span>)</span><br />
      Calculates the canonical permutation of a graph using the BLISS 
      isomorphism algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.canonical_permutation">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#clique_number" class="summary-sig-name">clique_number</a>()</span><br />
      Returns the clique number of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.omega">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#cliques" class="summary-sig-name">cliques</a>(<span class="summary-sig-arg">min</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">max</span>=<span class="summary-sig-default">0</span>)</span><br />
      Returns some or all cliques of the graph as a list of tuples.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.cliques">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#closeness" class="summary-sig-name">closeness</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>,
        <span class="summary-sig-arg">cutoff</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">normalized</span>=<span class="summary-sig-default">True</span>)</span><br />
      Calculates the closeness centralities of given vertices in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.closeness">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#clusters" class="summary-sig-name">clusters</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">STRONG</span>)</span><br />
      Calculates the (strong or weak) clusters for a given graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.clusters">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#cocitation" class="summary-sig-name">cocitation</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates cocitation scores for given vertices in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.cocitation">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#cohesive_blocks" class="summary-sig-name">cohesive_blocks</a>()</span><br />
      Calculates the cohesive block structure of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.cohesive_blocks">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_edge_betweenness" class="summary-sig-name">community_edge_betweenness</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Community structure detection based on the betweenness of the edges 
      in the network.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_edge_betweenness">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_fastgreedy" class="summary-sig-name">community_fastgreedy</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Finds the community structure of the graph according to the algorithm
      of Clauset et al based on the greedy optimization of modularity.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_fastgreedy">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_infomap" class="summary-sig-name">community_infomap</a>(<span class="summary-sig-arg">edge_weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">vertex_weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">trials</span>=<span class="summary-sig-default">10</span>)</span><br />
      Finds the community structure of the network according to the Infomap
      method of Martin Rosvall and Carl T.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_infomap">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_label_propagation" class="summary-sig-name">community_label_propagation</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">initial</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">fixed</span>=<span class="summary-sig-default">None</span>)</span><br />
      Finds the community structure of the graph according to the label 
      propagation method of Raghavan et al.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_label_propagation">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_leading_eigenvector" class="summary-sig-name">community_leading_eigenvector</a>(<span class="summary-sig-arg">n</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">arpack_options</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      A proper implementation of Newman's eigenvector community structure 
      detection.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_leading_eigenvector">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_multilevel" class="summary-sig-name">community_multilevel</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">return_levels</span>=<span class="summary-sig-default">True</span>)</span><br />
      Finds the community structure of the graph according to the 
      multilevel algorithm of Blondel et al.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_multilevel">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_optimal_modularity" class="summary-sig-name">community_optimal_modularity</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the optimal modularity score of the graph and the 
      corresponding community structure.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_optimal_modularity">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_spinglass" class="summary-sig-name">community_spinglass</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">spins</span>=<span class="summary-sig-default">25</span>,
        <span class="summary-sig-arg">parupdate</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">start_temp</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">stop_temp</span>=<span class="summary-sig-default">0.01</span>,
        <span class="summary-sig-arg">cool_fact</span>=<span class="summary-sig-default">0.99</span>,
        <span class="summary-sig-arg">update_rule</span>=<span class="summary-sig-default">&quot;config&quot;</span>,
        <span class="summary-sig-arg">gamma</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">implementation</span>=<span class="summary-sig-default">&quot;orig&quot;</span>,
        <span class="summary-sig-arg">lambda</span>=<span class="summary-sig-default">1</span>)</span><br />
      Finds the community structure of the graph according to the spinglass
      community detection method of Reichardt &amp; Bornholdt.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_spinglass">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#community_walktrap" class="summary-sig-name">community_walktrap</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">steps</span>=<span class="summary-sig-default">None</span>)</span><br />
      Finds the community structure of the graph according to the random 
      walk method of Latapy &amp; Pons.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_walktrap">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#complementer" class="summary-sig-name">complementer</a>(<span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>)</span><br />
      Returns the complementer of the graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.complementer">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="compose"></a><span class="summary-sig-name">compose</span>(<span class="summary-sig-arg">other</span>)</span><br />
      Returns the composition of two graphs.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.compose">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#constraint" class="summary-sig-name">constraint</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates Burt's constraint scores for given vertices in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.constraint">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#contract_vertices" class="summary-sig-name">contract_vertices</a>(<span class="summary-sig-arg">mapping</span>,
        <span class="summary-sig-arg">combine_attrs</span>=<span class="summary-sig-default">None</span>)</span><br />
      Contracts some vertices in the graph, i.e.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.contract_vertices">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="convergence_degree"></a><span class="summary-sig-name">convergence_degree</span>()</span><br />
      Undocumented (yet).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.convergence_degree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="convergence_field_size"></a><span class="summary-sig-name">convergence_field_size</span>()</span><br />
      Undocumented (yet).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.convergence_field_size">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="copy"></a><span class="summary-sig-name">copy</span>()</span><br />
      Creates an exact deep copy of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.copy">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#coreness" class="summary-sig-name">coreness</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>)</span><br />
      Finds the coreness (shell index) of the vertices of the network.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.shell_index">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#count_isomorphisms_vf2" class="summary-sig-name">count_isomorphisms_vf2</a>(<span class="summary-sig-arg">other</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">node_compat_fn</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_compat_fn</span>=<span class="summary-sig-default">None</span>)</span><br />
      Determines the number of isomorphisms between the graph and another 
      one</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.count_isomorphisms_vf2">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#count_multiple" class="summary-sig-name">count_multiple</a>(<span class="summary-sig-arg">edges</span>=<span class="summary-sig-default">None</span>)</span><br />
      Counts the multiplicities of the given edges.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.count_multiple">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#count_subisomorphisms_vf2" class="summary-sig-name">count_subisomorphisms_vf2</a>(<span class="summary-sig-arg">other</span>,
        <span class="summary-sig-arg">color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">node_compat_fn</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_compat_fn</span>=<span class="summary-sig-default">None</span>)</span><br />
      Determines the number of subisomorphisms between the graph and 
      another one</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.count_subisomorphisms_vf2">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#decompose" class="summary-sig-name">decompose</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">STRONG</span>,
        <span class="summary-sig-arg">maxcompno</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">minelements</span>=<span class="summary-sig-default">1</span>)</span><br />
      Decomposes the graph into subgraphs.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.decompose">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#degree" class="summary-sig-name">degree</a>(<span class="summary-sig-arg">vertices</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">True</span>)</span><br />
      Returns some vertex degrees from the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.degree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#delete_edges" class="summary-sig-name">delete_edges</a>(<span class="summary-sig-arg">es</span>)</span><br />
      Removes edges from the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.delete_edges">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#delete_vertices" class="summary-sig-name">delete_vertices</a>(<span class="summary-sig-arg">vs</span>)</span><br />
      Deletes vertices and all its edges from the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.delete_vertices">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#density" class="summary-sig-name">density</a>(<span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>)</span><br />
      Calculates the density of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.density">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#diameter" class="summary-sig-name">diameter</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">unconn</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the diameter of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.diameter">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="difference"></a><span class="summary-sig-name">difference</span>(<span class="summary-sig-arg">other</span>)</span><br />
      Subtracts the given graph from the original</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.difference">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#disjoint_union" class="summary-sig-name">disjoint_union</a>(<span class="summary-sig-arg">graphs</span>)</span><br />
      Creates the disjoint union of two (or more) graphs.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.disjoint_union">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#diversity" class="summary-sig-name">diversity</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the structural diversity index of the vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.diversity">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#dyad_census" class="summary-sig-name">dyad_census</a>()</span><br />
      Dyad census, as defined by Holland and Leinhardt</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.dyad_census">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#eccentricity" class="summary-sig-name">eccentricity</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>)</span><br />
      Calculates the eccentricities of given vertices in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.eccentricity">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">integer</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#ecount" class="summary-sig-name">ecount</a>()</span><br />
      Counts the number of edges.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.ecount">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#edge_attributes" class="summary-sig-name">edge_attributes</a>()</span><br />
      Returns:
      the attribute name list of the graph's edges</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.edge_attributes">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#edge_betweenness" class="summary-sig-name">edge_betweenness</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">cutoff</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates or estimates the edge betweennesses in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.edge_betweenness">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#edge_connectivity" class="summary-sig-name">edge_connectivity</a>(<span class="summary-sig-arg">source</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">target</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">checks</span>=<span class="summary-sig-default">True</span>)</span><br />
      Calculates the edge connectivity of the graph or between some 
      vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.edge_disjoint_paths">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#eigenvector_centrality" class="summary-sig-name">eigenvector_centrality</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">scale</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">return_eigenvalue</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">arpack_options</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the eigenvector centralities of the vertices in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.evcent">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#farthest_points" class="summary-sig-name">farthest_points</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">unconn</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns two vertex IDs whose distance equals the actual diameter of 
      the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.farthest_points">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#feedback_arc_set" class="summary-sig-name">feedback_arc_set</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">method</span>=<span class="summary-sig-default">&quot;eades&quot;</span>)</span><br />
      Calculates an approximately or exactly minimal feedback arc set.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.feedback_arc_set">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_adjacency" class="summary-sig-name">get_adjacency</a>(<span class="summary-sig-arg">type</span>=<span class="summary-sig-default">GET_ADJACENCY_BOTH</span>,
        <span class="summary-sig-arg">eids</span>=<span class="summary-sig-default">False</span>)</span><br />
      Returns the adjacency matrix of a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_adjacency">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_all_shortest_paths" class="summary-sig-name">get_all_shortest_paths</a>(<span class="summary-sig-arg">v</span>,
        <span class="summary-sig-arg">to</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>)</span><br />
      Calculates all of the shortest paths from/to a given node in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_all_shortest_paths">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_diameter" class="summary-sig-name">get_diameter</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">unconn</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns a path with the actual diameter of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_diameter">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="get_edgelist"></a><span class="summary-sig-name">get_edgelist</span>()</span><br />
      Returns the edge list of a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_edgelist">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_eid" class="summary-sig-name">get_eid</a>(<span class="summary-sig-arg">v1</span>,
        <span class="summary-sig-arg">v2</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">error</span>=<span class="summary-sig-default">True</span>)</span><br />
      Returns the edge ID of an arbitrary edge between vertices v1 and v2</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_eid">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_eids" class="summary-sig-name">get_eids</a>(<span class="summary-sig-arg">pairs</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">path</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">error</span>=<span class="summary-sig-default">True</span>)</span><br />
      Returns the edge IDs of some edges between some vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_eids">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_incidence" class="summary-sig-name">get_incidence</a>(<span class="summary-sig-arg">types</span>)</span><br />
      Internal function, undocumented.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_incidence">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_isomorphisms_vf2" class="summary-sig-name">get_isomorphisms_vf2</a>(<span class="summary-sig-arg">other</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">node_compat_fn</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_compat_fn</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns all isomorphisms between the graph and another one</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_isomorphisms_vf2">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_shortest_paths" class="summary-sig-name">get_shortest_paths</a>(<span class="summary-sig-arg">v</span>,
        <span class="summary-sig-arg">to</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>,
        <span class="summary-sig-arg">output</span>=<span class="summary-sig-default">&quot;vpath&quot;</span>)</span><br />
      Calculates the shortest paths from/to a given node in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_shortest_paths">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_subisomorphisms_lad" class="summary-sig-name">get_subisomorphisms_lad</a>(<span class="summary-sig-arg">other</span>,
        <span class="summary-sig-arg">domains</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">induced</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">time_limit</span>=<span class="summary-sig-default">0</span>)</span><br />
      Returns all subisomorphisms between the graph and another one using 
      the LAD algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_subisomorphisms_lad">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#get_subisomorphisms_vf2" class="summary-sig-name">get_subisomorphisms_vf2</a>(<span class="summary-sig-arg">other</span>,
        <span class="summary-sig-arg">color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">node_compat_fn</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_compat_fn</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns all subisomorphisms between the graph and another one</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_subisomorphisms_vf2">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#girth" class="summary-sig-name">girth</a>(<span class="summary-sig-arg">return_shortest_circle</span>=<span class="summary-sig-default">False</span>)</span><br />
      Returns the girth of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.girth">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#gomory_hu_tree" class="summary-sig-name">gomory_hu_tree</a>(<span class="summary-sig-arg">capacity</span>=<span class="summary-sig-default">None</span>)</span><br />
      Internal function, undocumented.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.gomory_hu_tree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">boolean</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#has_multiple" class="summary-sig-name">has_multiple</a>()</span><br />
      Checks whether the graph has multiple edges.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.has_multiple">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#hub_score" class="summary-sig-name">hub_score</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">scale</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">arpack_options</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">return_eigenvalue</span>=<span class="summary-sig-default">False</span>)</span><br />
      Calculates Kleinberg's hub score for the vertices of the graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.hub_score">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#incident" class="summary-sig-name">incident</a>(<span class="summary-sig-arg">vertex</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>)</span><br />
      Returns the edges a given vertex is incident on.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.incident">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#independence_number" class="summary-sig-name">independence_number</a>()</span><br />
      Returns the independence number of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.alpha">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#independent_vertex_sets" class="summary-sig-name">independent_vertex_sets</a>(<span class="summary-sig-arg">min</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">max</span>=<span class="summary-sig-default">0</span>)</span><br />
      Returns some or all independent vertex sets of the graph as a list of
      tuples.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.independent_vertex_sets">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#induced_subgraph" class="summary-sig-name">induced_subgraph</a>(<span class="summary-sig-arg">vertices</span>,
        <span class="summary-sig-arg">implementation</span>=<span class="summary-sig-default">&quot;auto&quot;</span>)</span><br />
      Returns a subgraph spanned by the given vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.subgraph">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#intersection" class="summary-sig-name">intersection</a>(<span class="summary-sig-arg">graphs</span>)</span><br />
      Creates the intersection of two (or more) graphs.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.intersection">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_bipartite" class="summary-sig-name">is_bipartite</a>(<span class="summary-sig-arg">return_types</span>=<span class="summary-sig-default">False</span>)</span><br />
      Decides whether the graph is bipartite or not.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_bipartite">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_connected" class="summary-sig-name">is_connected</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">STRONG</span>)</span><br />
      Decides whether the graph is connected.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_connected">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">boolean</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_dag" class="summary-sig-name">is_dag</a>()</span><br />
      Checks whether the graph is a DAG (directed acyclic graph).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_dag">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">boolean</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="is_directed"></a><span class="summary-sig-name">is_directed</span>()</span><br />
      Checks whether the graph is directed.@return: <code>True</code> if it
      is directed, <code>False</code> otherwise.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_directed">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_loop" class="summary-sig-name">is_loop</a>(<span class="summary-sig-arg">edges</span>=<span class="summary-sig-default">None</span>)</span><br />
      Checks whether a specific set of edges contain loop edges</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_loop">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_minimal_separator" class="summary-sig-name">is_minimal_separator</a>(<span class="summary-sig-arg">vertices</span>)</span><br />
      Decides whether the given vertex set is a minimal separator.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_minimal_separator">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_multiple" class="summary-sig-name">is_multiple</a>(<span class="summary-sig-arg">edges</span>=<span class="summary-sig-default">None</span>)</span><br />
      Checks whether an edge is a multiple edge.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_multiple">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_mutual" class="summary-sig-name">is_mutual</a>(<span class="summary-sig-arg">edges</span>=<span class="summary-sig-default">None</span>)</span><br />
      Checks whether an edge has an opposite pair.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_mutual">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_separator" class="summary-sig-name">is_separator</a>(<span class="summary-sig-arg">vertices</span>)</span><br />
      Decides whether the removal of the given vertices disconnects the 
      graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_separator">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">boolean</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#is_simple" class="summary-sig-name">is_simple</a>()</span><br />
      Checks whether the graph is simple (no loop or multiple edges).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_simple">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#isoclass" class="summary-sig-name">isoclass</a>(<span class="summary-sig-arg">vertices</span>)</span><br />
      Returns the isomorphy class of the graph or its subgraph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.isoclass">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#isomorphic" class="summary-sig-name">isomorphic</a>(<span class="summary-sig-arg">other</span>)</span><br />
      Checks whether the graph is isomorphic to another graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.isomorphic">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#isomorphic_bliss" class="summary-sig-name">isomorphic_bliss</a>(<span class="summary-sig-arg">other</span>,
        <span class="summary-sig-arg">return_mapping_12</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">return_mapping_21</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">sh1</span>=<span class="summary-sig-default">&quot;fm&quot;</span>,
        <span class="summary-sig-arg">sh2</span>=<span class="summary-sig-default">&quot;fm&quot;</span>)</span><br />
      Checks whether the graph is isomorphic to another graph, using the 
      BLISS isomorphism algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.isomorphic_bliss">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#isomorphic_vf2" class="summary-sig-name">isomorphic_vf2</a>(<span class="summary-sig-arg">other</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">return_mapping_12</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">return_mapping_21</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">node_compat_fn</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_compat_fn</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">callback</span>=<span class="summary-sig-default">None</span>)</span><br />
      Checks whether the graph is isomorphic to another graph, using the 
      VF2 isomorphism algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.isomorphic_vf2">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#knn" class="summary-sig-name">knn</a>(<span class="summary-sig-arg">vids</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the average degree of the neighbors for each vertex, and 
      the same quantity as the function of vertex degree.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.knn">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#laplacian" class="summary-sig-name">laplacian</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">normalized</span>=<span class="summary-sig-default">False</span>)</span><br />
      Returns the Laplacian matrix of a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.laplacian">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#largest_cliques" class="summary-sig-name">largest_cliques</a>()</span><br />
      Returns the largest cliques of the graph as a list of tuples.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.largest_cliques">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#largest_independent_vertex_sets" class="summary-sig-name">largest_independent_vertex_sets</a>()</span><br />
      Returns the largest independent vertex sets of the graph as a list of
      tuples.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.largest_independent_vertex_sets">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_bipartite" class="summary-sig-name">layout_bipartite</a>(<span class="summary-sig-arg">types</span>=<span class="summary-sig-default">&quot;type&quot;</span>,
        <span class="summary-sig-arg">hgap</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">vgap</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">maxiter</span>=<span class="summary-sig-default">100</span>)</span><br />
      Place the vertices of a bipartite graph in two layers.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_bipartite">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_circle" class="summary-sig-name">layout_circle</a>(<span class="summary-sig-arg">dim</span>=<span class="summary-sig-default">2</span>)</span><br />
      Places the vertices of the graph uniformly on a circle or a sphere.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_circle">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_drl" class="summary-sig-name">layout_drl</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">fixed</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">seed</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">options</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">dim</span>=<span class="summary-sig-default">2</span>)</span><br />
      Places the vertices on a 2D plane or in the 3D space ccording to the 
      DrL layout algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_drl">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_fruchterman_reingold" class="summary-sig-name">layout_fruchterman_reingold</a>(<span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxiter</span>=<span class="summary-sig-default">500</span>,
        <span class="summary-sig-arg">maxdelta</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">area</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">coolexp</span>=<span class="summary-sig-default">1.5</span>,
        <span class="summary-sig-arg">repulserad</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">seed</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">minx</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxx</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">miny</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxy</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">minz</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxz</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">dim</span>=<span class="summary-sig-default">2</span>)</span><br />
      Places the vertices on a 2D plane or in the 3D space according to the
      Fruchterman-Reingold algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_fruchterman_reingold">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_graphopt" class="summary-sig-name">layout_graphopt</a>(<span class="summary-sig-arg">niter</span>=<span class="summary-sig-default">500</span>,
        <span class="summary-sig-arg">node_charge</span>=<span class="summary-sig-default">0.001</span>,
        <span class="summary-sig-arg">node_mass</span>=<span class="summary-sig-default">30</span>,
        <span class="summary-sig-arg">spring_length</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">spring_constant</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">max_sa_movement</span>=<span class="summary-sig-default">5</span>,
        <span class="summary-sig-arg">seed</span>=<span class="summary-sig-default">None</span>)</span><br />
      This is a port of the graphopt layout algorithm by Michael Schmuhl.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_graphopt">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_grid" class="summary-sig-name">layout_grid</a>(<span class="summary-sig-arg">width</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">height</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">dim</span>=<span class="summary-sig-default">2</span>)</span><br />
      Places the vertices of a graph in a 2D or 3D grid.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_grid">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_grid_fruchterman_reingold" class="summary-sig-name">layout_grid_fruchterman_reingold</a>(<span class="summary-sig-arg">maxiter</span>=<span class="summary-sig-default">500</span>,
        <span class="summary-sig-arg">maxdelta</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">area</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">coolexp</span>=<span class="summary-sig-default">0.99</span>,
        <span class="summary-sig-arg">repulserad</span>=<span class="summary-sig-default">maxiter*maxdelta</span>,
        <span class="summary-sig-arg">cellsize</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">seed</span>=<span class="summary-sig-default">None</span>)</span><br />
      Places the vertices on a 2D plane according to the 
      Fruchterman-Reingold grid algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_grid_fruchterman_reingold">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_kamada_kawai" class="summary-sig-name">layout_kamada_kawai</a>(<span class="summary-sig-arg">maxiter</span>=<span class="summary-sig-default">1000</span>,
        <span class="summary-sig-arg">sigma</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">initemp</span>=<span class="summary-sig-default">10</span>,
        <span class="summary-sig-arg">coolexp</span>=<span class="summary-sig-default">0.99</span>,
        <span class="summary-sig-arg">kkconst</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">seed</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">minx</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxx</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">miny</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxy</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">minz</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">maxz</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">dim</span>=<span class="summary-sig-default">2</span>)</span><br />
      Places the vertices on a plane according to the Kamada-Kawai 
      algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_kamada_kawai">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_lgl" class="summary-sig-name">layout_lgl</a>(<span class="summary-sig-arg">maxiter</span>=<span class="summary-sig-default">150</span>,
        <span class="summary-sig-arg">maxdelta</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">area</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">coolexp</span>=<span class="summary-sig-default">1.5</span>,
        <span class="summary-sig-arg">repulserad</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">cellsize</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">root</span>=<span class="summary-sig-default">None</span>)</span><br />
      Places the vertices on a 2D plane according to the Large Graph 
      Layout.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_lgl">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_mds" class="summary-sig-name">layout_mds</a>(<span class="summary-sig-arg">dist</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">dim</span>=<span class="summary-sig-default">2</span>,
        <span class="summary-sig-arg">arpack_options</span>=<span class="summary-sig-default">None</span>)</span><br />
      Places the vertices in an Euclidean space with the given number of 
      dimensions using multidimensional scaling.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_mds">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_random" class="summary-sig-name">layout_random</a>(<span class="summary-sig-arg">dim</span>=<span class="summary-sig-default">2</span>)</span><br />
      Places the vertices of the graph randomly.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_random">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_reingold_tilford" class="summary-sig-name">layout_reingold_tilford</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;out&quot;</span>,
        <span class="summary-sig-arg">root</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">rootlevel</span>=<span class="summary-sig-default">None</span>)</span><br />
      Places the vertices on a 2D plane according to the Reingold-Tilford 
      layout algorithm.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_reingold_tilford">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_reingold_tilford_circular" class="summary-sig-name">layout_reingold_tilford_circular</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;out&quot;</span>,
        <span class="summary-sig-arg">root</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">rootlevel</span>=<span class="summary-sig-default">None</span>)</span><br />
      Circular Reingold-Tilford layout for trees.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_reingold_tilford_circular">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#layout_star" class="summary-sig-name">layout_star</a>(<span class="summary-sig-arg">center</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">order</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates a star-like layout for the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_star">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#linegraph" class="summary-sig-name">linegraph</a>()</span><br />
      Returns the line graph of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.linegraph">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#maxdegree" class="summary-sig-name">maxdegree</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>)</span><br />
      Returns the maximum degree of a vertex set in the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.maxdegree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#maxflow" class="summary-sig-name">maxflow</a>(<span class="summary-sig-arg">source</span>,
        <span class="summary-sig-arg">target</span>,
        <span class="summary-sig-arg">capacity</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns the maximum flow between the source and target vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.maxflow">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#maxflow_value" class="summary-sig-name">maxflow_value</a>(<span class="summary-sig-arg">source</span>,
        <span class="summary-sig-arg">target</span>,
        <span class="summary-sig-arg">capacity</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns the value of the maximum flow between the source and target 
      vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.maxflow_value">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#maximal_cliques" class="summary-sig-name">maximal_cliques</a>(<span class="summary-sig-arg">min</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">max</span>=<span class="summary-sig-default">0</span>)</span><br />
      Returns the maximal cliques of the graph as a list of tuples.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.maximal_cliques">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#maximal_independent_vertex_sets" class="summary-sig-name">maximal_independent_vertex_sets</a>()</span><br />
      Returns the maximal independent vertex sets of the graph as a list of
      tuples.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.maximal_independent_vertex_sets">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#mincut" class="summary-sig-name">mincut</a>(<span class="summary-sig-arg">source</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">target</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">capacity</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the minimum cut between the source and target vertices or 
      within the whole graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.mincut">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#mincut_value" class="summary-sig-name">mincut_value</a>(<span class="summary-sig-arg">source</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">target</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">capacity</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns the minimum cut between the source and target vertices or 
      within the whole graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.mincut_value">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#minimum_size_separators" class="summary-sig-name">minimum_size_separators</a>()</span><br />
      Returns a list containing all separator vertex sets of minimum size.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.minimum_size_separators">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#modularity" class="summary-sig-name">modularity</a>(<span class="summary-sig-arg">membership</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the modularity of the graph with respect to some vertex 
      types.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.modularity">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#motifs_randesu" class="summary-sig-name">motifs_randesu</a>(<span class="summary-sig-arg">size</span>=<span class="summary-sig-default">3</span>,
        <span class="summary-sig-arg">cut_prob</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">callback</span>=<span class="summary-sig-default">None</span>)</span><br />
      Counts the number of motifs in the graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.motifs_randesu">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#motifs_randesu_estimate" class="summary-sig-name">motifs_randesu_estimate</a>(<span class="summary-sig-arg">size</span>=<span class="summary-sig-default">3</span>,
        <span class="summary-sig-arg">cut_prob</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">sample</span>)</span><br />
      Counts the total number of motifs in the graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.motifs_randesu_estimate">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#motifs_randesu_no" class="summary-sig-name">motifs_randesu_no</a>(<span class="summary-sig-arg">size</span>=<span class="summary-sig-default">3</span>,
        <span class="summary-sig-arg">cut_prob</span>=<span class="summary-sig-default">None</span>)</span><br />
      Counts the total number of motifs in the graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.motifs_randesu_no">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#neighborhood" class="summary-sig-name">neighborhood</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">order</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>)</span><br />
      For each vertex specified by <i>vertices</i>, returns the vertices 
      reachable from that vertex in at most <i>order</i> steps.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.neighborhood">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#neighborhood_size" class="summary-sig-name">neighborhood_size</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">order</span>=<span class="summary-sig-default">1</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>)</span><br />
      For each vertex specified by <i>vertices</i>, returns the number of 
      vertices reachable from that vertex in at most <i>order</i> steps.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.neighborhood_size">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#neighbors" class="summary-sig-name">neighbors</a>(<span class="summary-sig-arg">vertex</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>)</span><br />
      Returns adjacent vertices to a given vertex.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.neighbors">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#path_length_hist" class="summary-sig-name">path_length_hist</a>(<span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>)</span><br />
      Calculates the path length histogram of the graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.path_length_hist">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#permute_vertices" class="summary-sig-name">permute_vertices</a>(<span class="summary-sig-arg">permutation</span>)</span><br />
      Permutes the vertices of the graph according to the given permutation
      and returns the new graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.permute_vertices">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#personalized_pagerank" class="summary-sig-name">personalized_pagerank</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">directed</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">damping</span>=<span class="summary-sig-default">0.85</span>,
        <span class="summary-sig-arg">reset</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">reset_vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">arpack_options</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">implementation</span>=<span class="summary-sig-default">&quot;prpack&quot;</span>,
        <span class="summary-sig-arg">niter</span>=<span class="summary-sig-default">1000</span>,
        <span class="summary-sig-arg">eps</span>=<span class="summary-sig-default">0.001</span>)</span><br />
      Calculates the personalized PageRank values of a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.personalized_pagerank">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#predecessors" class="summary-sig-name">predecessors</a>(<span class="summary-sig-arg">vertex</span>)</span><br />
      Returns the predecessors of a given vertex.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.predecessors">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#radius" class="summary-sig-name">radius</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>)</span><br />
      Calculates the radius of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.radius">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#reciprocity" class="summary-sig-name">reciprocity</a>(<span class="summary-sig-arg">ignore_loops</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;default&quot;</span>)</span><br />
      Reciprocity defines the proportion of mutual connections in a 
      directed graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.reciprocity">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#rewire" class="summary-sig-name">rewire</a>(<span class="summary-sig-arg">n</span>=<span class="summary-sig-default">1000</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;simple&quot;</span>)</span><br />
      Randomly rewires the graph while preserving the degree distribution.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.rewire">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#rewire_edges" class="summary-sig-name">rewire_edges</a>(<span class="summary-sig-arg">prob</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">multiple</span>=<span class="summary-sig-default">False</span>)</span><br />
      Rewires the edges of a graph with constant probability.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.rewire_edges">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#shortest_paths" class="summary-sig-name">shortest_paths</a>(<span class="summary-sig-arg">source</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">target</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>)</span><br />
      Calculates shortest path lengths for given vertices in a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.shortest_paths_dijkstra">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#similarity_dice" class="summary-sig-name">similarity_dice</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">pairs</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">IGRAPH_ALL</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">True</span>)</span><br />
      Dice similarity coefficient of vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.similarity_dice">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#similarity_inverse_log_weighted" class="summary-sig-name">similarity_inverse_log_weighted</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">IGRAPH_ALL</span>)</span><br />
      Inverse log-weighted similarity coefficient of vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.similarity_inverse_log_weighted">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#similarity_jaccard" class="summary-sig-name">similarity_jaccard</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">pairs</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">IGRAPH_ALL</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">True</span>)</span><br />
      Jaccard similarity coefficient of vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.similarity_jaccard">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#simplify" class="summary-sig-name">simplify</a>(<span class="summary-sig-arg">multiple</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">combine_edges</span>=<span class="summary-sig-default">None</span>)</span><br />
      Simplifies a graph by removing self-loops and/or multiple edges.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.simplify">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#st_mincut" class="summary-sig-name">st_mincut</a>(<span class="summary-sig-arg">source</span>,
        <span class="summary-sig-arg">target</span>,
        <span class="summary-sig-arg">capacity</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the minimum cut between the source and target vertices in 
      a graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.st_mincut">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#strength" class="summary-sig-name">strength</a>(<span class="summary-sig-arg">vertices</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>,
        <span class="summary-sig-arg">loops</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Returns the strength (weighted degree) of some vertices from the 
      graph</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.strength">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#subcomponent" class="summary-sig-name">subcomponent</a>(<span class="summary-sig-arg">v</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">ALL</span>)</span><br />
      Determines the indices of vertices which are in the same component as
      a given vertex.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.subcomponent">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#subgraph_edges" class="summary-sig-name">subgraph_edges</a>(<span class="summary-sig-arg">edges</span>,
        <span class="summary-sig-arg">delete_vertices</span>=<span class="summary-sig-default">True</span>)</span><br />
      Returns a subgraph spanned by the given edges.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.subgraph_edges">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#subisomorphic_lad" class="summary-sig-name">subisomorphic_lad</a>(<span class="summary-sig-arg">other</span>,
        <span class="summary-sig-arg">domains</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">induced</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">time_limit</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">return_mapping</span>=<span class="summary-sig-default">False</span>)</span><br />
      Checks whether a subgraph of the graph is isomorphic to another 
      graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.subisomorphic_lad">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#subisomorphic_vf2" class="summary-sig-name">subisomorphic_vf2</a>(<span class="summary-sig-arg">other</span>,
        <span class="summary-sig-arg">color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color1</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_color2</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">return_mapping_12</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">return_mapping_21</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">callback</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">node_compat_fn</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">edge_compat_fn</span>=<span class="summary-sig-default">None</span>)</span><br />
      Checks whether a subgraph of the graph is isomorphic to another 
      graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.subisomorphic_vf2">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#successors" class="summary-sig-name">successors</a>(<span class="summary-sig-arg">vertex</span>)</span><br />
      Returns the successors of a given vertex.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.successors">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#to_directed" class="summary-sig-name">to_directed</a>(<span class="summary-sig-arg">mutual</span>=<span class="summary-sig-default">True</span>)</span><br />
      Converts an undirected graph to directed.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.to_directed">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#to_undirected" class="summary-sig-name">to_undirected</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;collapse&quot;</span>,
        <span class="summary-sig-arg">combine_edges</span>=<span class="summary-sig-default">None</span>)</span><br />
      Converts a directed graph to undirected.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.to_undirected">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#topological_sorting" class="summary-sig-name">topological_sorting</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>)</span><br />
      Calculates a possible topological sorting of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.topological_sorting">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#transitivity_avglocal_undirected" class="summary-sig-name">transitivity_avglocal_undirected</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;nan&quot;</span>)</span><br />
      Calculates the average of the vertex transitivities of the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.transitivity_avglocal_undirected">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#transitivity_local_undirected" class="summary-sig-name">transitivity_local_undirected</a>(<span class="summary-sig-arg">vertices</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;nan&quot;</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">None</span>)</span><br />
      Calculates the local transitivity (clustering coefficient) of the 
      given vertices in the graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.transitivity_local_undirected">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#transitivity_undirected" class="summary-sig-name">transitivity_undirected</a>(<span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">&quot;nan&quot;</span>)</span><br />
      Calculates the global transitivity (clustering coefficient) of the 
      graph.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.transitivity_undirected">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#triad_census" class="summary-sig-name">triad_census</a>()</span><br />
      Triad census, as defined by Davis and Leinhardt</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.triad_census">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#unfold_tree" class="summary-sig-name">unfold_tree</a>(<span class="summary-sig-arg">sources</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mode</span>=<span class="summary-sig-default">OUT</span>)</span><br />
      Unfolds the graph using a BFS to a tree by duplicating vertices as 
      necessary.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.unfold_tree">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#union" class="summary-sig-name">union</a>(<span class="summary-sig-arg">graphs</span>)</span><br />
      Creates the union of two (or more) graphs.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.union">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">integer</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#vcount" class="summary-sig-name">vcount</a>()</span><br />
      Counts the number of vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.vcount">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#vertex_attributes" class="summary-sig-name">vertex_attributes</a>()</span><br />
      Returns:
      the attribute name list of the graph's vertices</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.vertex_attributes">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#vertex_connectivity" class="summary-sig-name">vertex_connectivity</a>(<span class="summary-sig-arg">source</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">target</span>=<span class="summary-sig-default">-1</span>,
        <span class="summary-sig-arg">checks</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">neighbors</span>=<span class="summary-sig-default">&quot;error&quot;</span>)</span><br />
      Calculates the vertex connectivity of the graph or between some 
      vertices.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#Graph.cohesion">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_dimacs" class="summary-sig-name">write_dimacs</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">source</span>,
        <span class="summary-sig-arg">target</span>,
        <span class="summary-sig-arg">capacity</span>=<span class="summary-sig-default">None</span>)</span><br />
      Writes the graph in DIMACS format to the given file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_dimacs">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_dot" class="summary-sig-name">write_dot</a>(<span class="summary-sig-arg">f</span>)</span><br />
      Writes the graph in DOT format to the given file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_dot">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_edgelist" class="summary-sig-name">write_edgelist</a>(<span class="summary-sig-arg">f</span>)</span><br />
      Writes the edge list of a graph to a file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_edgelist">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_gml" class="summary-sig-name">write_gml</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">creator</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">ids</span>=<span class="summary-sig-default">None</span>)</span><br />
      Writes the graph in GML format to the given file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_gml">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_graphml" class="summary-sig-name">write_graphml</a>(<span class="summary-sig-arg">f</span>)</span><br />
      Writes the graph to a GraphML file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_graphml">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_leda" class="summary-sig-name">write_leda</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">names</span>=<span class="summary-sig-default">&quot;name&quot;</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">&quot;weights&quot;</span>)</span><br />
      Writes the graph to a file in LEDA native format.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_leda">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_lgl" class="summary-sig-name">write_lgl</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">names</span>=<span class="summary-sig-default">&quot;name&quot;</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">&quot;weights&quot;</span>,
        <span class="summary-sig-arg">isolates</span>=<span class="summary-sig-default">True</span>)</span><br />
      Writes the edge list of a graph to a file in .lgl format.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_lgl">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_ncol" class="summary-sig-name">write_ncol</a>(<span class="summary-sig-arg">f</span>,
        <span class="summary-sig-arg">names</span>=<span class="summary-sig-default">&quot;name&quot;</span>,
        <span class="summary-sig-arg">weights</span>=<span class="summary-sig-default">&quot;weights&quot;</span>)</span><br />
      Writes the edge list of a graph to a file in .ncol format.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_ncol">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="igraph.GraphBase-class.html#write_pajek" class="summary-sig-name">write_pajek</a>(<span class="summary-sig-arg">f</span>)</span><br />
      Writes the graph in Pajek format to the given file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_pajek">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__repr__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="Adjacency"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Adjacency</span>(<span class="sig-arg">matrix</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ADJ_DIRECTED</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Adjacency">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph from its adjacency matrix.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>matrix</code></strong> - the adjacency matrix</li>
        <li><strong class="pname"><code>mode</code></strong> - the mode to be used. Possible values are:
          <ul>
            <li>
              <code>ADJ_DIRECTED</code> - the graph will be directed and a 
              matrix element gives the number of edges between two vertex.
            </li>
            <li>
              <code>ADJ_UNDIRECTED</code> - alias to <code>ADJ_MAX</code> 
              for convenience.
            </li>
            <li>
              <code>ADJ_MAX</code>   - undirected graph will be created and
              the number of edges between vertex <i class="math">i</i> and 
              <i class="math">j</i> is <i class="math">max(A(i,j), 
              A(j,i))</i>
            </li>
            <li>
              <code>ADJ_MIN</code>   - like <code>ADJ_MAX</code>, but with 
              <i class="math">min(A(i,j), A(j,i))</i>
            </li>
            <li>
              <code>ADJ_PLUS</code>  - like <code>ADJ_MAX</code>, but with 
              <i class="math">A(i,j) + A(j,i)</i>
            </li>
            <li>
              <code>ADJ_UPPER</code> - undirected graph with the upper 
              right triangle of the matrix (including the diagonal)
            </li>
            <li>
              <code>ADJ_LOWER</code> - undirected graph with the lower left
              triangle of the matrix (including the diagonal)
            </li>
          </ul>
          <p>These values can also be given as strings without the 
          <code>ADJ</code> prefix.</p></li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Asymmetric_Preference"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Asymmetric_Preference</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">type_dist_matrix</span>,
        <span class="sig-arg">pref_matrix</span>,
        <span class="sig-arg">attribute</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Asymmetric_Preference">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph based on asymmetric vertex types and connection 
  probabilities.</p>
  <p>This is the asymmetric variant of <a 
  href="igraph.GraphBase-class.html#Preference" 
  class="link">Graph.Preference</a>. A given number of vertices are 
  generated. Every vertex is assigned to an &quot;incoming&quot; and an 
  &quot;outgoing&quot; vertex type according to the given joint type 
  probabilities. Finally, every vertex pair is evaluated and a directed 
  edge is created between them with a probability depending on the 
  &quot;outgoing&quot; type of the source vertex and the 
  &quot;incoming&quot; type of the target vertex.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph</li>
        <li><strong class="pname"><code>type_dist_matrix</code></strong> - matrix giving the joint distribution of vertex types</li>
        <li><strong class="pname"><code>pref_matrix</code></strong> - matrix giving the connection probabilities for different vertex 
          types.</li>
        <li><strong class="pname"><code>attribute</code></strong> - the vertex attribute name used to store the vertex types. If 
          <code>None</code>, vertex types are not stored.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether loop edges are allowed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Atlas"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Atlas</span>(<span class="sig-arg">idx</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Atlas">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph from the Graph Atlas.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>idx</code></strong> - The index of the graph to be generated. Indices start from zero, 
          graphs are listed:
          <ol start="1">
            <li>
              in increasing order of number of vertices;
            </li>
            <li>
              for a fixed number of vertices, in increasing order of the 
              number of edges;
            </li>
            <li>
              for fixed numbers of vertices and edges, in increasing order 
              of the degree sequence, for example 111223 &lt; 112222;
            </li>
            <li>
              for fixed degree sequence, in increasing number of 
              automorphisms.
            </li>
          </ol></li>
    </ul></dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        <i>An Atlas of Graphs</i> by Ronald C. Read and Robin J. Wilson, 
        Oxford University Press, 1998.
      </p>
</div></td></tr></table>
</div>
<a name="Barabasi"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Barabasi</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">m</span>,
        <span class="sig-arg">outpref</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">power</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">zero_appeal</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">implementation</span>=<span class="sig-default">&quot;psumtree&quot;</span>,
        <span class="sig-arg">start_from</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Barabasi">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph based on the Barabasi-Albert model.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices</li>
        <li><strong class="pname"><code>m</code></strong> - either the number of outgoing edges generated for each vertex or 
          a list containing the number of outgoing edges for each vertex 
          explicitly.</li>
        <li><strong class="pname"><code>outpref</code></strong> - <code>True</code> if the out-degree of a given vertex should also
          increase its citation probability (as well as its in-degree), but
          it defaults to <code>False</code>.</li>
        <li><strong class="pname"><code>directed</code></strong> - <code>True</code> if the generated graph should be directed 
          (default: <code>False</code>).</li>
        <li><strong class="pname"><code>power</code></strong> - the power constant of the nonlinear model. It can be omitted, and
          in this case the usual linear model will be used.</li>
        <li><strong class="pname"><code>zero_appeal</code></strong> - the attractivity of vertices with degree zero.</li>
        <li><strong class="pname"><code>implementation</code></strong> - the algorithm to use to generate the network. Possible values 
          are:
          <ul>
            <li>
              <code>&quot;bag&quot;</code>: the algorithm that was the 
              default in igraph before 0.6. It works by putting the ids of 
              the vertices into a bag (multiset) exactly as many times as 
              their in-degree, plus once more. The required number of cited
              vertices are then drawn from the bag with replacement. It 
              works only for <i>power</i>=1 and <i>zero_appeal</i>=1.
            </li>
            <li>
              <code>&quot;psumtree&quot;</code>: this algorithm uses a 
              partial prefix-sum tree to generate the graph. It does not 
              generate multiple edges and it works for any values of 
              <i>power</i> and <i>zero_appeal</i>.
            </li>
            <li>
              <code>&quot;psumtree_multiple&quot;</code>: similar to 
              <code>&quot;psumtree&quot;</code>, but it will generate 
              multiple edges as well. igraph before 0.6 used this algorithm
              for <i>power</i>s other than 1.
            </li>
          </ul></li>
        <li><strong class="pname"><code>start_from</code></strong> - if given and not <code>None</code>, this must be another <a 
          href="igraph.Graph-class.html" class="link">Graph</a> object. 
          igraph will use this graph as a starting point for the 
          preferential attachment model.</li>
    </ul></dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Barabasi, A-L and Albert, R. 1999. Emergence of scaling in random 
        networks. Science, 286 509-512.
      </p>
</div></td></tr></table>
</div>
<a name="De_Bruijn"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">De_Bruijn</span>(<span class="sig-arg">m</span>,
        <span class="sig-arg">n</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.De_Bruijn">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a de Bruijn graph with parameters (m, n)</p>
  <p>A de Bruijn graph represents relationships between strings. An 
  alphabet of <i class="math">m</i> letters are used and strings of length 
  <i class="math">n</i> are considered. A vertex corresponds to every 
  possible string and there is a directed edge from vertex <i 
  class="math">v</i> to vertex <i class="math">w</i> if the string of <i 
  class="math">v</i> can be transformed into the string of <i 
  class="math">w</i> by removing its first letter and appending a letter to
  it.</p>
  <p>Please note that the graph will have <i class="math">m^n</i> vertices 
  and even more edges, so probably you don't want to supply too big numbers
  for <i class="math">m</i> and <i class="math">n</i>.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>m</code></strong> - the size of the alphabet</li>
        <li><strong class="pname"><code>n</code></strong> - the length of the strings</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Degree_Sequence"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Degree_Sequence</span>(<span class="sig-arg">out</span>,
        <span class="sig-arg">in</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">method</span>=<span class="sig-default">&quot;simple&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Degree_Sequence">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph with a given degree sequence.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>out</code></strong> - the out-degree sequence for a directed graph. If the in-degree 
          sequence is omitted, the generated graph will be undirected, so 
          this will be the in-degree sequence as well</li>
        <li><strong class="pname"><code>in</code></strong> - the in-degree sequence for a directed graph. If omitted, the 
          generated graph will be undirected.</li>
        <li><strong class="pname"><code>method</code></strong> - the generation method to be used. One of the following:
          <ul>
            <li>
              <code>&quot;simple&quot;</code> -- simple generator that 
              sometimes generates loop edges and multiple edges. The 
              generated graph is not guaranteed to be connected.
            </li>
            <li>
              <code>&quot;no_multiple&quot;</code> -- similar to 
              <code>&quot;simple&quot;</code> but avoids the generation of 
              multiple and loop edges at the expense of increased time 
              complexity. The method will re-start the generation every 
              time it gets stuck in a configuration where it is not 
              possible to insert any more edges without creating loops or 
              multiple edges, and there is no upper bound on the number of 
              iterations, but it will succeed eventually if the input 
              degree sequence is graphical and throw an exception if the 
              input degree sequence is not graphical.
            </li>
            <li>
              <code>&quot;vl&quot;</code> -- a more sophisticated generator
              that can sample undirected, connected simple graphs 
              uniformly. It uses Monte-Carlo methods to randomize the 
              graphs. This generator should be favoured if undirected and 
              connected graphs are to be generated and execution time is 
              not a concern. igraph uses the original implementation of 
              Fabien Viger; see the following URL and the paper cited on it
              for the details of the algorithm: <a 
              href="http://www-rp.lip6.fr/~latapy/FV/generation.html" 
              target="_top">http://www-rp.lip6.fr/~latapy/FV/generation.html</a>.
            </li>
          </ul></li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Erdos_Renyi"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Erdos_Renyi</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">p</span>,
        <span class="sig-arg">m</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Erdos_Renyi">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph based on the Erdos-Renyi model.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices.</li>
        <li><strong class="pname"><code>p</code></strong> - the probability of edges. If given, <code>m</code> must be 
          missing.</li>
        <li><strong class="pname"><code>m</code></strong> - the number of edges. If given, <code>p</code> must be missing.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to generate a directed graph.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether self-loops are allowed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Establishment"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Establishment</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">k</span>,
        <span class="sig-arg">type_dist</span>,
        <span class="sig-arg">pref_matrix</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Establishment">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph based on a simple growing model with vertex 
  types.</p>
  <p>A single vertex is added at each time step. This new vertex tries to 
  connect to k vertices in the graph. The probability that such a 
  connection is realized depends on the types of the vertices involved.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph</li>
        <li><strong class="pname"><code>k</code></strong> - the number of connections tried in each step</li>
        <li><strong class="pname"><code>type_dist</code></strong> - list giving the distribution of vertex types</li>
        <li><strong class="pname"><code>pref_matrix</code></strong> - matrix (list of lists) giving the connection probabilities for 
          different vertex types</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to generate a directed graph.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Famous"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Famous</span>(<span class="sig-arg">name</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Famous">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a famous graph based on its name.</p>
  <p>Several famous graphs are known to <code>igraph</code> including (but 
  not limited to) the Chvatal graph, the Petersen graph or the Tutte graph.
  This method generates one of them based on its name (case insensitive). 
  See the documentation of the C interface of <code>igraph</code> for the 
  names available: <a href="http://igraph.org/doc/c" 
  target="_top">http://igraph.org/doc/c</a>.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>name</code></strong> - the name of the graph to be generated.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Forest_Fire"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Forest_Fire</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">fw_prob</span>,
        <span class="sig-arg">bw_factor</span>=<span class="sig-default">0.0</span>,
        <span class="sig-arg">ambs</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Forest_Fire">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph based on the forest fire model</p>
  <p>The forest fire model is a growin graph model. In every time step, a 
  new vertex is added to the graph. The new vertex chooses an ambassador 
  (or more than one if <i class="math">ambs&gt;1</i>) and starts a 
  simulated forest fire at its ambassador(s). The fire spreads through the 
  edges. The spreading probability along an edge is given by <i 
  class="math">fw_prob</i>. The fire may also spread backwards on an edge 
  by probability <i class="math">fw_prob * bw_factor</i>. When the fire 
  ended, the newly added vertex connects to the vertices ``burned'' in the 
  previous fire.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph</li>
        <li><strong class="pname"><code>fw_prob</code></strong> - forward burning probability</li>
        <li><strong class="pname"><code>bw_factor</code></strong> - ratio of backward and forward burning probability</li>
        <li><strong class="pname"><code>ambs</code></strong> - number of ambassadors chosen in each step</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the graph will be directed</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Full"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Full</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Full">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a full graph (directed or undirected, with or without 
  loops).</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to generate a directed graph.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether self-loops are allowed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Full_Citation"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Full_Citation</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Full_Citation">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a full citation graph</p>
  <p>A full citation graph is a graph where the vertices are indexed from 0
  to <i class="math">n-1</i> and vertex <i class="math">i</i> has a 
  directed edge towards all vertices with an index less than <i 
  class="math">i</i>.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to generate a directed graph.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Growing_Random"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Growing_Random</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">m</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">citation</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Growing_Random">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a growing random graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - The number of vertices in the graph</li>
        <li><strong class="pname"><code>m</code></strong> - The number of edges to add in each step (after adding a new 
          vertex)</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the graph should be directed.</li>
        <li><strong class="pname"><code>citation</code></strong> - whether the new edges should originate from the most recently 
          added vertex.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Isoclass"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Isoclass</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">class</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Isoclass">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph with a given isomorphy class.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph (3 or 4)</li>
        <li><strong class="pname"><code>class</code></strong> - the isomorphy class</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the graph should be directed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="K_Regular"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">K_Regular</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">k</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">multiple</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.K_Regular">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a k-regular random graph</p>
  <p>A k-regular random graph is a random graph where each vertex has 
  degree k. If the graph is directed, both the in-degree and the out-degree
  of each vertex will be k.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - The number of vertices in the graph</li>
        <li><strong class="pname"><code>k</code></strong> - The degree of each vertex if the graph is undirected, or the 
          in-degree and out-degree of each vertex if the graph is directed</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the graph should be directed.</li>
        <li><strong class="pname"><code>multiple</code></strong> - whether it is allowed to create multiple edges.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Kautz"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Kautz</span>(<span class="sig-arg">m</span>,
        <span class="sig-arg">n</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Kautz">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a Kautz graph with parameters (m, n)</p>
  <p>A Kautz graph is a labeled graph, vertices are labeled by strings of 
  length <i class="math">n+1</i> above an alphabet with <i 
  class="math">m+1</i> letters, with the restriction that every two 
  consecutive letters in the string must be different. There is a directed 
  edge from a vertex <i class="math">v</i> to another vertex <i 
  class="math">w</i> if it is possible to transform the string of <i 
  class="math">v</i> into the string of <i class="math">w</i> by removing 
  the first letter and appending a letter to it.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>m</code></strong> - the size of the alphabet minus one</li>
        <li><strong class="pname"><code>n</code></strong> - the length of the strings minus one</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="LCF"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">LCF</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">shifts</span>,
        <span class="sig-arg">repeats</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.LCF">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph from LCF notation.</p>
  <p>LCF is short for Lederberg-Coxeter-Frucht, it is a concise notation 
  for 3-regular Hamiltonian graphs. It consists of three parameters, the 
  number of vertices in the graph, a list of shifts giving additional edges
  to a cycle backbone and another integer giving how many times the shifts 
  should be performed. See <a 
  href="http://mathworld.wolfram.com/LCFNotation.html" 
  target="_top">http://mathworld.wolfram.com/LCFNotation.html</a> for 
  details.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices</li>
        <li><strong class="pname"><code>shifts</code></strong> - the shifts in a list or tuple</li>
        <li><strong class="pname"><code>repeats</code></strong> - the number of repeats</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Lattice"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Lattice</span>(<span class="sig-arg">dim</span>,
        <span class="sig-arg">nei</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">mutual</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">circular</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Lattice">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a regular lattice.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>dim</code></strong> - list with the dimensions of the lattice</li>
        <li><strong class="pname"><code>nei</code></strong> - value giving the distance (number of steps) within which two 
          vertices will be connected.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to create a directed graph.</li>
        <li><strong class="pname"><code>mutual</code></strong> - whether to create all connections as mutual in case of a directed
          graph.</li>
        <li><strong class="pname"><code>circular</code></strong> - whether the generated lattice is periodic.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Preference"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Preference</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">type_dist</span>,
        <span class="sig-arg">pref_matrix</span>,
        <span class="sig-arg">attribute</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Preference">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph based on vertex types and connection 
  probabilities.</p>
  <p>This is practically the nongrowing variant of <a 
  href="igraph.GraphBase-class.html#Establishment" 
  class="link">Graph.Establishment</a>. A given number of vertices are 
  generated. Every vertex is assigned to a vertex type according to the 
  given type probabilities. Finally, every vertex pair is evaluated and an 
  edge is created between them with a probability depending on the types of
  the vertices involved.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph</li>
        <li><strong class="pname"><code>type_dist</code></strong> - list giving the distribution of vertex types</li>
        <li><strong class="pname"><code>pref_matrix</code></strong> - matrix giving the connection probabilities for different vertex 
          types.</li>
        <li><strong class="pname"><code>attribute</code></strong> - the vertex attribute name used to store the vertex types. If 
          <code>None</code>, vertex types are not stored.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to generate a directed graph.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether loop edges are allowed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_DIMACS"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_DIMACS</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_DIMACS">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads a graph from a file conforming to the DIMACS minimum-cost flow 
  file format.</p>
  <p>For the exact description of the format, see <a 
  href="http://lpsolve.sourceforge.net/5.5/DIMACS.htm" 
  target="_top">http://lpsolve.sourceforge.net/5.5/DIMACS.htm</a></p>
  <p>Restrictions compared to the official description of the format:</p>
  <ul>
    <li>
      igraph's DIMACS reader requires only three fields in an arc 
      definition, describing the edge's source and target node and its 
      capacity.
    </li>
    <li>
      Source vertices are identified by 's' in the FLOW field, target 
      vertices are identified by 't'.
    </li>
    <li>
      Node indices start from 1. Only a single source and target node is 
      allowed.
    </li>
  </ul>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the generated graph should be directed.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the generated graph, the source and the target of the flow and 
          the edge capacities in a tuple</dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_DL"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_DL</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_DL">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads an UCINET DL file and creates a graph based on it.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the generated graph should be directed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_Edgelist"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_Edgelist</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_Edgelist">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads an edge list from a file and creates a graph based on it.</p>
  <p>Please note that the vertex indices are zero-based.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the generated graph should be directed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_GML"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_GML</span>(<span class="sig-arg">f</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_GML">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads a GML file and creates a graph based on it.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_GraphDB"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_GraphDB</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_GraphDB">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads a GraphDB format file and creates a graph based on it.</p>
  <p>GraphDB is a binary format, used in the graph database for isomorphism
  testing (see <a href="http://amalfi.dis.unina.it/graph/" 
  target="_top">http://amalfi.dis.unina.it/graph/</a>).</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the generated graph should be directed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_GraphML"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_GraphML</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">index</span>=<span class="sig-default">0</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_GraphML">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads a GraphML format file and creates a graph based on it.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
        <li><strong class="pname"><code>index</code></strong> - if the GraphML file contains multiple graphs, specifies the one 
          that should be loaded. Graph indices start from zero, so if you 
          want to load the first graph, specify 0 here.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_Lgl"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_Lgl</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">names</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">&quot;if_present&quot;</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_Lgl">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads an .lgl file used by LGL.</p>
  <p>It is also useful for creating graphs from &quot;named&quot; (and 
  optionally weighted) edge lists.</p>
  <p>This format is used by the Large Graph Layout program. See the <a 
  href="http://bioinformatics.icmb.utexas.edu/lgl/" 
  target="_top">documentation of LGL</a> regarding the exact format 
  description.</p>
  <p>LGL originally cannot deal with graphs containing multiple or loop 
  edges, but this condition is not checked here, as igraph is happy with 
  these.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
        <li><strong class="pname"><code>names</code></strong> - If <code>True</code>, the vertex names are added as a vertex 
          attribute called 'name'.</li>
        <li><strong class="pname"><code>weights</code></strong> - If True, the edge weights are added as an edge attribute called 
          'weight', even if there are no weights in the file. If False, the
          edge weights are never added, even if they are present. 
          <code>&quot;auto&quot;</code> or 
          <code>&quot;if_present&quot;</code> means that weights are added 
          if there is at least one weighted edge in the input file, but 
          they are not added otherwise.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the graph being created should be directed</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_Ncol"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_Ncol</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">names</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">&quot;if_present&quot;</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_Ncol">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads an .ncol file used by LGL.</p>
  <p>It is also useful for creating graphs from &quot;named&quot; (and 
  optionally weighted) edge lists.</p>
  <p>This format is used by the Large Graph Layout program. See the <a 
  href="http://bioinformatics.icmb.utexas.edu/lgl/" 
  target="_top">documentation of LGL</a> regarding the exact format 
  description.</p>
  <p>LGL originally cannot deal with graphs containing multiple or loop 
  edges, but this condition is not checked here, as igraph is happy with 
  these.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
        <li><strong class="pname"><code>names</code></strong> - If <code>True</code>, the vertex names are added as a vertex 
          attribute called 'name'.</li>
        <li><strong class="pname"><code>weights</code></strong> - If True, the edge weights are added as an edge attribute called 
          'weight', even if there are no weights in the file. If False, the
          edge weights are never added, even if they are present. 
          <code>&quot;auto&quot;</code> or 
          <code>&quot;if_present&quot;</code> means that weights are added 
          if there is at least one weighted edge in the input file, but 
          they are not added otherwise.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether the graph being created should be directed</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Read_Pajek"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Read_Pajek</span>(<span class="sig-arg">f</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Read_Pajek">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reads a Pajek format file and creates a graph based on it.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file or a Python file handle</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Recent_Degree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Recent_Degree</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">m</span>,
        <span class="sig-arg">window</span>,
        <span class="sig-arg">outpref</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">power</span>=<span class="sig-default">1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Recent_Degree">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph based on a stochastic model where the probability of
  an edge gaining a new node is proportional to the edges gained in a given
  time window.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices</li>
        <li><strong class="pname"><code>m</code></strong> - either the number of outgoing edges generated for each vertex or 
          a list containing the number of outgoing edges for each vertex 
          explicitly.</li>
        <li><strong class="pname"><code>window</code></strong> - size of the window in time steps</li>
        <li><strong class="pname"><code>outpref</code></strong> - <code>True</code> if the out-degree of a given vertex should also
          increase its citation probability (as well as its in-degree), but
          it defaults to <code>False</code>.</li>
        <li><strong class="pname"><code>directed</code></strong> - <code>True</code> if the generated graph should be directed 
          (default: <code>False</code>).</li>
        <li><strong class="pname"><code>power</code></strong> - the power constant of the nonlinear model. It can be omitted, and
          in this case the usual linear model will be used.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Ring"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Ring</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">mutual</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">circular</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Ring">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a ring graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the ring</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to create a directed ring.</li>
        <li><strong class="pname"><code>mutual</code></strong> - whether to create mutual edges in a directed ring.</li>
        <li><strong class="pname"><code>circular</code></strong> - whether to create a closed ring.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="SBM"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">SBM</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">pref_matrix</span>,
        <span class="sig-arg">block_sizes</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.SBM">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph based on a stochastic blockmodel.</p>
  <p>A given number of vertices are generated. Every vertex is assigned to 
  a vertex type according to the given block sizes. Vertices of the same 
  type will be assigned consecutive vertex IDs. Finally, every vertex pair 
  is evaluated and an edge is created between them with a probability 
  depending on the types of the vertices involved. The probabilities are 
  taken from the preference matrix.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph</li>
        <li><strong class="pname"><code>pref_matrix</code></strong> - matrix giving the connection probabilities for different vertex 
          types.</li>
        <li><strong class="pname"><code>block_sizes</code></strong> - list giving the number of vertices in each block; must sum up to 
          <i>n</i>.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to generate a directed graph.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether loop edges are allowed.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Star"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Star</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">&quot;undirected&quot;</span>,
        <span class="sig-arg">center</span>=<span class="sig-default">0</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Star">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a star graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph</li>
        <li><strong class="pname"><code>mode</code></strong> - Gives the type of the star graph to create. Should be either 
          &quot;in&quot;, &quot;out&quot;, &quot;mutual&quot; or 
          &quot;undirected&quot;</li>
        <li><strong class="pname"><code>center</code></strong> - Vertex ID for the central vertex in the star.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Static_Fitness"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Static_Fitness</span>(<span class="sig-arg">m</span>,
        <span class="sig-arg">fitness_out</span>,
        <span class="sig-arg">fitness_in</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">multiple</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Static_Fitness">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a non-growing graph with edge probabilities proportional to 
  node fitnesses.</p>
  <p>The algorithm randomly selects vertex pairs and connects them until 
  the given number of edges are created. Each vertex is selected with a 
  probability proportional to its fitness; for directed graphs, a vertex is
  selected as a source proportional to its out-fitness and as a target 
  proportional to its in-fitness.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>m</code></strong> - the number of edges in the graph</li>
        <li><strong class="pname"><code>fitness_out</code></strong> - a numeric vector with non-negative entries, one for each vertex. 
          These values represent the fitness scores (out-fitness scores for
          directed graphs). <i>fitness</i> is an alias of this keyword 
          argument.</li>
        <li><strong class="pname"><code>fitness_in</code></strong> - a numeric vector with non-negative entries, one for each vertex. 
          These values represent the in-fitness scores for directed graphs.
          For undirected graphs, this argument must be <code>None</code>.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether loop edges are allowed.</li>
        <li><strong class="pname"><code>multiple</code></strong> - whether multiple edges are allowed.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a directed or undirected graph with the prescribed power-law 
          degree distributions.</dd>
  </dl>
</td></tr></table>
</div>
<a name="Static_Power_Law"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Static_Power_Law</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">m</span>,
        <span class="sig-arg">exponent_out</span>,
        <span class="sig-arg">exponent_in</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">multiple</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">finite_size_correction</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Static_Power_Law">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a non-growing graph with prescribed power-law degree 
  distributions.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph</li>
        <li><strong class="pname"><code>m</code></strong> - the number of edges in the graph</li>
        <li><strong class="pname"><code>exponent_out</code></strong> - the exponent of the out-degree distribution, which must be 
          between 2 and infinity (inclusive). When <i>exponent_in</i> is 
          not given or negative, the graph will be undirected and this 
          parameter specifies the degree distribution. <i>exponent</i> is 
          an alias to this keyword argument.</li>
        <li><strong class="pname"><code>exponent_in</code></strong> - the exponent of the in-degree distribution, which must be between
          2 and infinity (inclusive) It can also be negative, in which case
          an undirected graph will be generated.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether loop edges are allowed.</li>
        <li><strong class="pname"><code>multiple</code></strong> - whether multiple edges are allowed.</li>
        <li><strong class="pname"><code>finite_size_correction</code></strong> - whether to apply a finite-size correction to the generated 
          fitness values for exponents less than 3. See the paper of Cho et
          al for more details.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a directed or undirected graph with the prescribed power-law 
          degree distributions.</dd>
  </dl>
<div class="fields">      <strong>Reference:</strong>
      <ul class="nomargin-top">
        <li>
        Goh K-I, Kahng B, Kim D: Universal behaviour of load distribution 
        in scale-free networks. Phys Rev Lett 87(27):278701, 2001.
        </li>
        <li>
        Cho YS, Kim JS, Park J, Kahng B, Kim D: Percolation transitions in 
        scale-free networks under the Achlioptas process. Phys Rev Lett 
        103:135702, 2009.
        </li>
      </ul>
</div></td></tr></table>
</div>
<a name="Tree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Tree</span>(<span class="sig-arg">n</span>,
        <span class="sig-arg">children</span>,
        <span class="sig-arg">type</span>=<span class="sig-default">TREE_UNDIRECTED</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Tree">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a tree in which almost all vertices have the same number of 
  children.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices in the graph</li>
        <li><strong class="pname"><code>children</code></strong> - the number of children of a vertex in the graph</li>
        <li><strong class="pname"><code>type</code></strong> - determines whether the tree should be directed, and if this is 
          the case, also its orientation. Must be one of 
          <code>TREE_IN</code>, <code>TREE_OUT</code> and 
          <code>TREE_UNDIRECTED</code>.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="Watts_Strogatz"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Watts_Strogatz</span>(<span class="sig-arg">dim</span>,
        <span class="sig-arg">size</span>,
        <span class="sig-arg">nei</span>,
        <span class="sig-arg">p</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">multiple</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Watts_Strogatz">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>dim</code></strong> - the dimension of the lattice</li>
        <li><strong class="pname"><code>size</code></strong> - the size of the lattice along all dimensions</li>
        <li><strong class="pname"><code>nei</code></strong> - value giving the distance (number of steps) within which two 
          vertices will be connected.</li>
        <li><strong class="pname"><code>p</code></strong> - rewiring probability</li>
        <li><strong class="pname"><code>loops</code></strong> - specifies whether loop edges are allowed</li>
        <li><strong class="pname"><code>multiple</code></strong> - specifies whether multiple edges are allowed</li>
    </ul></dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.GraphBase-class.html#Lattice" 
        class="link">Lattice()</a>, <a 
        href="igraph.GraphBase-class.html#rewire" 
        class="link">rewire()</a>, <a 
        href="igraph.GraphBase-class.html#rewire_edges" 
        class="link">rewire_edges()</a> if more flexibility is needed
      </p>
      <p><strong>Reference:</strong>
        Duncan J Watts and Steven H Strogatz: <i>Collective dynamics of 
        small world networks</i>, Nature 393, 440-442, 1998
      </p>
</div></td></tr></table>
</div>
<a name="Weighted_Adjacency"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">Weighted_Adjacency</span>(<span class="sig-arg">matrix</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ADJ_DIRECTED</span>,
        <span class="sig-arg">attr</span>=<span class="sig-default">&quot;weight&quot;</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.Weighted_Adjacency">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generates a graph from its adjacency matrix.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>matrix</code></strong> - the adjacency matrix</li>
        <li><strong class="pname"><code>mode</code></strong> - the mode to be used. Possible values are:
          <ul>
            <li>
              <code>ADJ_DIRECTED</code> - the graph will be directed and a 
              matrix element gives the number of edges between two vertex.
            </li>
            <li>
              <code>ADJ_UNDIRECTED</code> - alias to <code>ADJ_MAX</code> 
              for convenience.
            </li>
            <li>
              <code>ADJ_MAX</code>   - undirected graph will be created and
              the number of edges between vertex <i class="math">i</i> and 
              <i class="math">j</i> is <i class="math">max(A(i,j), 
              A(j,i))</i>
            </li>
            <li>
              <code>ADJ_MIN</code>   - like <code>ADJ_MAX</code>, but with 
              <i class="math">min(A(i,j), A(j,i))</i>
            </li>
            <li>
              <code>ADJ_PLUS</code>  - like <code>ADJ_MAX</code>, but with 
              <i class="math">A(i,j) + A(j,i)</i>
            </li>
            <li>
              <code>ADJ_UPPER</code> - undirected graph with the upper 
              right triangle of the matrix (including the diagonal)
            </li>
            <li>
              <code>ADJ_LOWER</code> - undirected graph with the lower left
              triangle of the matrix (including the diagonal)
            </li>
          </ul>
          <p>These values can also be given as strings without the 
          <code>ADJ</code> prefix.</p></li>
        <li><strong class="pname"><code>attr</code></strong> - the name of the edge attribute that stores the edge weights.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether to include loop edges. When <code>False</code>, the 
          diagonal of the adjacency matrix will be ignored.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="_Random_Bipartite"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_Random_Bipartite</span>(<span class="sig-arg">n1</span>,
        <span class="sig-arg">n2</span>,
        <span class="sig-arg">p</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">m</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">neimode</span>=<span class="sig-default">&quot;all&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase._Random_Bipartite">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Internal function, undocumented.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        Graph.Random_Bipartite()
      </p>
</div></td></tr></table>
</div>
<a name="__graph_as_cobject"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__graph_as_cobject</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.__graph_as_cobject">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the igraph graph encapsulated by the Python object as a 
  PyCObject</p>
  <p>.A PyCObject is practically a regular C pointer, wrapped in a Python 
  object. This function should not be used directly by igraph users, it is 
  useful only in the case when the underlying igraph object must be passed 
  to other C code through Python.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">...</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>x.__init__(...) initializes x; see help(type(x)) for signature</p>
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__new__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__new__</span>(<span class="sig-arg">T</span>,
        <span class="sig-arg">S</span>,
        <span class="sig-arg">...</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.__new__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  
  <dl class="fields">
    <dt>Returns: a new object with type S, a subtype of T</dt>
    <dt>Overrides:
        object.__new__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__register_destructor"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__register_destructor</span>(<span class="sig-arg">destructor</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.__register_destructor">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Registers a destructor to be called when the object is freed by 
  Python. This function should not be used directly by igraph users.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="__str__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__str__</span>(<span class="sig-arg">x</span>)</span>
    <br /><em class="fname">(Informal representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.__str__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>str(x)</p>
  <dl class="fields">
    <dt>Overrides:
        object.__str__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="add_edges"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">add_edges</span>(<span class="sig-arg">es</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.add_edges">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Adds edges to the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>es</code></strong> - the list of edges to be added. Every edge is represented with a 
          tuple, containing the vertex IDs of the two endpoints. Vertices 
          are enumerated from zero.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="add_vertices"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">add_vertices</span>(<span class="sig-arg">n</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.add_vertices">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Adds vertices to the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of vertices to be added</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="all_minimal_st_separators"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">all_minimal_st_separators</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.all_minimal_st_separators">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a list containing all the minimal s-t separators of a 
  graph.</p>
  <p>A minimal separator is a set of vertices whose removal disconnects the
  graph, while the removal of any subset of the set keeps the graph 
  connected.</p>
  <dl class="fields">
    <dt>Returns:</dt>
        <dd>a list where each item lists the vertex indices of a given 
          minimal s-t separator.</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Anne Berry, Jean-Paul Bordat and Olivier Cogis: Generating all the 
        minimal separators of a graph. In: Peter Widmayer, Gabriele Neyer 
        and Stephan Eidenbenz (eds.): Graph-theoretic concepts in computer 
        science, 1665, 167--172, 1999. Springer.
      </p>
</div></td></tr></table>
</div>
<a name="all_st_cuts"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">all_st_cuts</span>(<span class="sig-arg">source</span>,
        <span class="sig-arg">target</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.all_st_cuts">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns all the cuts between the source and target vertices in a 
  directed graph.</p>
  <p>This function lists all edge-cuts between a source and a target 
  vertex. Every cut is listed exactly once.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex ID</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex ID</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a tuple where the first element is a list of lists of edge IDs 
          representing a cut and the second element is a list of lists of 
          vertex IDs representing the sets of vertices that were separated 
          by the cuts.</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a list of <a href="igraph.cut.Cut-class.html" 
        class="link">Cut</a> objects. It is advised to use that.
      </p>
</div></td></tr></table>
</div>
<a name="all_st_mincuts"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">all_st_mincuts</span>(<span class="sig-arg">source</span>,
        <span class="sig-arg">target</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.all_st_mincuts">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns all minimum cuts between the source and target vertices in a 
  directed graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex ID</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex ID</li>
    </ul></dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a list of <a href="igraph.cut.Cut-class.html" 
        class="link">Cut</a> objects. It is advised to use that.
      </p>
</div></td></tr></table>
</div>
<a name="are_connected"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">are_connected</span>(<span class="sig-arg">v1</span>,
        <span class="sig-arg">v2</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.are_connected">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Decides whether two given vertices are directly connected.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>v1</code></strong> - the first vertex</li>
        <li><strong class="pname"><code>v2</code></strong> - the second vertex</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd><code>True</code> if there exists an edge from v1 to v2, 
          <code>False</code> otherwise.</dd>
  </dl>
</td></tr></table>
</div>
<a name="articulation_points"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">articulation_points</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.cut_vertices">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the list of articulation points in the graph.</p>
  <p>A vertex is an articulation point if its removal increases the number 
  of connected components in the graph.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="assortativity"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">assortativity</span>(<span class="sig-arg">types1</span>,
        <span class="sig-arg">types2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.assortativity">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the assortativity of the graph based on numeric properties of 
  the vertices.</p>
  <p>This coefficient is basically the correlation between the actual 
  connectivity patterns of the vertices and the pattern expected from the 
  disribution of the vertex types.</p>
  <p>See equation (21) in Newman MEJ: Mixing patterns in networks, Phys Rev
  E 67:026126 (2003) for the proper definition. The actual calculation is 
  performed using equation (26) in the same paper for directed graphs, and 
  equation (4) in Newman MEJ: Assortative mixing in networks, Phys Rev Lett
  89:208701 (2002) for undirected graphs.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>types1</code></strong> - vertex types in a list or the name of a vertex attribute holding 
          vertex types. Types are ideally denoted by numeric values.</li>
        <li><strong class="pname"><code>types2</code></strong> - in directed assortativity calculations, each vertex can have an 
          out-type and an in-type. In this case, <i>types1</i> contains the
          out-types and this parameter contains the in-types in a list or 
          the name of a vertex attribute. If <code>None</code>, it is 
          assumed to be equal to <i>types1</i>.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to consider edge directions or not.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the assortativity coefficient</dd>
  </dl>
<div class="fields">      <strong>Reference:</strong>
      <ul class="nomargin-top">
        <li>
        Newman MEJ: Mixing patterns in networks, Phys Rev E 67:026126, 
        2003.
        </li>
        <li>
        Newman MEJ: Assortative mixing in networks, Phys Rev Lett 
        89:208701, 2002.@see: <a 
        href="igraph.GraphBase-class.html#assortativity_degree" 
        class="link">assortativity_degree()</a> when the types are the 
        vertex degrees
        </li>
      </ul>
</div></td></tr></table>
</div>
<a name="assortativity_degree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">assortativity_degree</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.assortativity_degree">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the assortativity of a graph based on vertex degrees.</p>
  <p>See <a href="igraph.GraphBase-class.html#assortativity" 
  class="link">assortativity()</a> for the details. <a 
  href="igraph.GraphBase-class.html#assortativity_degree" 
  class="link">assortativity_degree()</a> simply calls <a 
  href="igraph.GraphBase-class.html#assortativity" 
  class="link">assortativity()</a> with the vertex degrees as types.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to consider edge directions for directed graphs or not. 
          This argument is ignored for undirected graphs.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the assortativity coefficient</dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.GraphBase-class.html#assortativity" 
        class="link">assortativity()</a>
      </p>
</div></td></tr></table>
</div>
<a name="assortativity_nominal"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">assortativity_nominal</span>(<span class="sig-arg">types</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.assortativity_nominal">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the assortativity of the graph based on vertex categories.</p>
  <p>Assuming that the vertices belong to different categories, this 
  function calculates the assortativity coefficient, which specifies the 
  extent to which the connections stay within categories. The assortativity
  coefficient is one if all the connections stay within categories and 
  minus one if all the connections join vertices of different categories. 
  For a randomly connected network, it is asymptotically zero.</p>
  <p>See equation (2) in Newman MEJ: Mixing patterns in networks, Phys Rev 
  E 67:026126 (2003) for the proper definition.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>types</code></strong> - vertex types in a list or the name of a vertex attribute holding 
          vertex types. Types should be denoted by numeric values.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to consider edge directions or not.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the assortativity coefficient</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Newman MEJ: Mixing patterns in networks, Phys Rev E 67:026126, 
        2003.
      </p>
</div></td></tr></table>
</div>
<a name="attributes"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">attributes</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.attributes">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  
  <dl class="fields">
    <dt>Returns:</dt>
        <dd>the attribute name list of the graph</dd>
  </dl>
</td></tr></table>
</div>
<a name="authority_score"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">authority_score</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">scale</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">arpack_options</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">return_eigenvalue</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.authority_score">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates Kleinberg's authority score for the vertices of the 
  graph</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
        <li><strong class="pname"><code>scale</code></strong> - whether to normalize the scores so that the largest one is 1.</li>
        <li><strong class="pname"><code>arpack_options</code></strong> - an <a href="igraph.ARPACKOptions-class.html" 
          class="link">ARPACKOptions</a> object used to fine-tune the 
          ARPACK eigenvector calculation. If omitted, the module-level 
          variable called <code>arpack_options</code> is used.</li>
        <li><strong class="pname"><code>return_eigenvalue</code></strong> - whether to return the largest eigenvalue</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the authority scores in a list and optionally the largest 
          eigenvalue as a second member of a tuple</dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        hub_score()
      </p>
</div></td></tr></table>
</div>
<a name="average_path_length"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">average_path_length</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">unconn</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.average_path_length">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the average path length in a graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to consider directed paths in case of a directed graph. 
          Ignored for undirected graphs.</li>
        <li><strong class="pname"><code>unconn</code></strong> - what to do when the graph is unconnected. If <code>True</code>, 
          the average of the geodesic lengths in the components is 
          calculated. Otherwise for all unconnected vertex pairs, a path 
          length equal to the number of vertices is used.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the average path length in the graph</dd>
  </dl>
</td></tr></table>
</div>
<a name="betweenness"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">betweenness</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">cutoff</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">nobigint</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.betweenness">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates or estimates the betweenness of vertices in a graph.</p>
  <p>Keyword arguments:</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices for which the betweennesses must be returned. If 
          <code>None</code>, assumes all of the vertices in the graph.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to consider directed paths.</li>
        <li><strong class="pname"><code>cutoff</code></strong> - if it is an integer, only paths less than or equal to this length
          are considered, effectively resulting in an estimation of the 
          betweenness for the given vertices. If <code>None</code>, the 
          exact betweenness is returned.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
        <li><strong class="pname"><code>nobigint</code></strong> - if <code>True</code>, igraph uses the longest available integer 
          type on the current platform to count shortest paths. For some 
          large networks that have a specific structure, the counters may 
          overflow. To prevent this, use <code>nobigint=False</code>, which
          forces igraph to use arbitrary precision integers at the expense 
          of increased computation time.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the (possibly estimated) betweenness of the given vertices in a 
          list</dd>
  </dl>
</td></tr></table>
</div>
<a name="bfs"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">bfs</span>(<span class="sig-arg">vid</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.bfs">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Conducts a breadth first search (BFS) on the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vid</code></strong> - the root vertex ID</li>
        <li><strong class="pname"><code>mode</code></strong> - either <a href="igraph-module.html#IN" class="link">IN</a> or <a 
          href="igraph-module.html#OUT" class="link">OUT</a> or <a 
          href="igraph-module.html#ALL" class="link">ALL</a>, ignored for 
          undirected graphs.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a tuple with the following items:
          <ul>
            <li>
              The vertex IDs visited (in order)
            </li>
            <li>
              The start indices of the layers in the vertex list
            </li>
            <li>
              The parent of every vertex in the BFS
            </li>
          </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="bfsiter"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">bfsiter</span>(<span class="sig-arg">vid</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>,
        <span class="sig-arg">advanced</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.bfsiter">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Constructs a breadth first search (BFS) iterator of the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vid</code></strong> - the root vertex ID</li>
        <li><strong class="pname"><code>mode</code></strong> - either <a href="igraph-module.html#IN" class="link">IN</a> or <a 
          href="igraph-module.html#OUT" class="link">OUT</a> or <a 
          href="igraph-module.html#ALL" class="link">ALL</a>.</li>
        <li><strong class="pname"><code>advanced</code></strong> - if <code>False</code>, the iterator returns the next vertex in 
          BFS order in every step. If <code>True</code>, the iterator 
          returns the distance of the vertex from the root and the parent 
          of the vertex in the BFS tree as well.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the BFS iterator as an <a href="igraph.BFSIter-class.html" 
          class="link">igraph.BFSIter</a> object.</dd>
  </dl>
</td></tr></table>
</div>
<a name="bibcoupling"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">bibcoupling</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.bibcoupling">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates bibliographic coupling scores for given vertices in a 
  graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices to be analysed. If <code>None</code>, all vertices 
          will be considered.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>bibliographic coupling scores for all given vertices in a matrix.</dd>
  </dl>
</td></tr></table>
</div>
<a name="biconnected_components"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">biconnected_components</span>(<span class="sig-arg">return_articulation_points</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.biconnected_components">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the biconnected components of the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>return_articulation_points</code></strong> - whether to return the articulation points as well</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list of lists containing edge indices making up spanning trees 
          of the biconnected components (one spanning tree for each 
          component) and optionally the list of articulation points</dd>
  </dl>
</td></tr></table>
</div>
<a name="bipartite_projection"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">bipartite_projection</span>(<span class="sig-arg">types</span>,
        <span class="sig-arg">multiplicity</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">probe1</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">which</span>=<span class="sig-default">-1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.bipartite_projection">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Internal function, undocumented.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        Graph.bipartite_projection()
      </p>
</div></td></tr></table>
</div>
<a name="bipartite_projection_size"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">bipartite_projection_size</span>(<span class="sig-arg">types</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.bipartite_projection_size">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Internal function, undocumented.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        Graph.bipartite_projection_size()
      </p>
</div></td></tr></table>
</div>
<a name="canonical_permutation"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">canonical_permutation</span>(<span class="sig-arg">sh</span>=<span class="sig-default">&quot;fm&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.canonical_permutation">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the canonical permutation of a graph using the BLISS 
  isomorphism algorithm.</p>
  <p>Passing the permutation returned here to <a 
  href="igraph.GraphBase-class.html#permute_vertices" 
  class="link">Graph.permute_vertices()</a> will transform the graph into 
  its canonical form.</p>
  <p>See <a href="http://www.tcs.hut.fi/Software/bliss/index.html" 
  target="_top">http://www.tcs.hut.fi/Software/bliss/index.html</a> for 
  more information about the BLISS algorithm and canonical 
  permutations.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>sh</code></strong> - splitting heuristics for graph as a case-insensitive string, with
          the following possible values:
          <ul>
            <li>
              <code>&quot;f&quot;</code>: first non-singleton cell
            </li>
            <li>
              <code>&quot;fl&quot;</code>: first largest non-singleton cell
            </li>
            <li>
              <code>&quot;fs&quot;</code>: first smallest non-singleton 
              cell
            </li>
            <li>
              <code>&quot;fm&quot;</code>: first maximally non-trivially 
              connected non-singleton cell
            </li>
            <li>
              <code>&quot;flm&quot;</code>: largest maximally non-trivially
              connected non-singleton cell
            </li>
            <li>
              <code>&quot;fsm&quot;</code>: smallest maximally 
              non-trivially connected non-singleton cell
            </li>
          </ul></li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a permutation vector containing vertex IDs. Vertex 0 in the 
          original graph will be mapped to an ID contained in the first 
          element of this vector; vertex 1 will be mapped to the second and
          so on.</dd>
  </dl>
</td></tr></table>
</div>
<a name="clique_number"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">clique_number</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.omega">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the clique number of the graph.</p>
  <p>The clique number of the graph is the size of the largest clique.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.GraphBase-class.html#largest_cliques" 
        class="link">largest_cliques()</a> for the largest cliques.
      </p>
</div></td></tr></table>
</div>
<a name="cliques"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">cliques</span>(<span class="sig-arg">min</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">max</span>=<span class="sig-default">0</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.cliques">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns some or all cliques of the graph as a list of tuples.</p>
  <p>A clique is a complete subgraph -- a set of vertices where an edge is 
  present between any two of them (excluding loops)</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>min</code></strong> - the minimum size of cliques to be returned. If zero or negative, 
          no lower bound will be used.</li>
        <li><strong class="pname"><code>max</code></strong> - the maximum size of cliques to be returned. If zero or negative, 
          no upper bound will be used.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="closeness"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">closeness</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>,
        <span class="sig-arg">cutoff</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">normalized</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.closeness">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the closeness centralities of given vertices in a 
  graph.</p>
  <p>The closeness centerality of a vertex measures how easily other 
  vertices can be reached from it (or the other way: how easily it can be 
  reached from the other vertices). It is defined as the number of the 
  number of vertices minus one divided by the sum of the lengths of all 
  geodesics from/to the given vertex.</p>
  <p>If the graph is not connected, and there is no path between two 
  vertices, the number of vertices is used instead the length of the 
  geodesic. This is always longer than the longest possible geodesic.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices for which the closenesses must be returned. If 
          <code>None</code>, uses all of the vertices in the graph.</li>
        <li><strong class="pname"><code>mode</code></strong> - must be one of <a href="igraph-module.html#IN" 
          class="link">IN</a>, <a href="igraph-module.html#OUT" 
          class="link">OUT</a> and <a href="igraph-module.html#ALL" 
          class="link">ALL</a>. <a href="igraph-module.html#IN" 
          class="link">IN</a> means that the length of the incoming paths, 
          <a href="igraph-module.html#OUT" class="link">OUT</a> means that 
          the length of the outgoing paths must be calculated. <a 
          href="igraph-module.html#ALL" class="link">ALL</a> means that 
          both of them must be calculated.</li>
        <li><strong class="pname"><code>cutoff</code></strong> - if it is an integer, only paths less than or equal to this length
          are considered, effectively resulting in an estimation of the 
          closeness for the given vertices (which is always an 
          underestimation of the real closeness, since some vertex pairs 
          will appear as disconnected even though they are connected).. If 
          <code>None</code>, the exact closeness is returned.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
        <li><strong class="pname"><code>normalized</code></strong> - Whether to normalize the raw closeness scores by multiplying by 
          the number of vertices minus one.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated closenesses in a list</dd>
  </dl>
</td></tr></table>
</div>
<a name="clusters"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">clusters</span>(<span class="sig-arg">mode</span>=<span class="sig-default">STRONG</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.clusters">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the (strong or weak) clusters for a given graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - must be either <code>STRONG</code> or <code>WEAK</code>, 
          depending on the clusters being sought. Optional, defaults to 
          <code>STRONG</code>.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the component index for every node in the graph.</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a <a 
        href="igraph.clustering.VertexClustering-class.html" 
        class="link">VertexClustering</a> object. It is advised to use 
        that.
      </p>
</div></td></tr></table>
</div>
<a name="cocitation"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">cocitation</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.cocitation">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates cocitation scores for given vertices in a graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices to be analysed. If <code>None</code>, all vertices 
          will be considered.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>cocitation scores for all given vertices in a matrix.</dd>
  </dl>
</td></tr></table>
</div>
<a name="cohesive_blocks"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">cohesive_blocks</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.cohesive_blocks">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the cohesive block structure of the graph.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a <a 
        href="igraph.clustering.CohesiveBlocks-class.html" 
        class="link">CohesiveBlocks</a> object. It is advised to use that.
      </p>
</div></td></tr></table>
</div>
<a name="community_edge_betweenness"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_edge_betweenness</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_edge_betweenness">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Community structure detection based on the betweenness of the edges in
  the network. This algorithm was invented by M Girvan and MEJ Newman, see:
  M Girvan and MEJ Newman: Community structure in social and biological 
  networks, Proc. Nat. Acad. Sci. USA 99, 7821-7826 (2002).</p>
  <p>The idea is that the betweenness of the edges connecting two 
  communities is typically high. So we gradually remove the edge with the 
  highest betweenness from the network and recalculate edge betweenness 
  after every removal, as long as all edges are removed.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to take into account the directedness of the edges when 
          we calculate the betweenness values.</li>
        <li><strong class="pname"><code>weights</code></strong> - name of an edge attribute or a list containing edge weights.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a tuple with the merge matrix that describes the dendrogram and 
          the modularity scores before each merge. The modularity scores 
          use the weights if the original graph was weighted.</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function is wrapped in a more convenient syntax in the derived
        class <a href="igraph.Graph-class.html" class="link">Graph</a>. It 
        is advised to use that instead of this version.
      </p>
</div></td></tr></table>
</div>
<a name="community_fastgreedy"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_fastgreedy</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_fastgreedy">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Finds the community structure of the graph according to the algorithm 
  of Clauset et al based on the greedy optimization of modularity.</p>
  <p>This is a bottom-up algorithm: initially every vertex belongs to a 
  separate community, and communities are merged one by one. In every step,
  the two communities being merged are the ones which result in the maximal
  increase in modularity.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - name of an edge attribute or a list containing edge weights</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a tuple with the following elements:
          <ol start="1">
            <li>
              The list of merges
            </li>
            <li>
              The modularity scores before each merge
            </li>
          </ol></dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function is wrapped in a more convenient syntax in the derived
        class <a href="igraph.Graph-class.html" class="link">Graph</a>. It 
        is advised to use that instead of this version.
      </p>
      <p><strong>Reference:</strong>
        A. Clauset, M. E. J. Newman and C. Moore: <i>Finding community 
        structure in very large networks.</i> Phys Rev E 70, 066111 (2004).
      </p>
      <p><strong>See Also:</strong>
        modularity()
      </p>
</div></td></tr></table>
</div>
<a name="community_infomap"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_infomap</span>(<span class="sig-arg">edge_weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">vertex_weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">trials</span>=<span class="sig-default">10</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_infomap">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Finds the community structure of the network according to the Infomap 
  method of Martin Rosvall and Carl T. Bergstrom.</p>
  <p>See <a href="http://www.mapequation.org" 
  target="_top">http://www.mapequation.org</a> for a visualization of the 
  algorithm or one of the references provided below.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>edge_weights</code></strong> - name of an edge attribute or a list containing edge weights.</li>
        <li><strong class="pname"><code>vertex_weights</code></strong> - name of an vertex attribute or a list containing vertex weights.</li>
        <li><strong class="pname"><code>trials</code></strong> - the number of attempts to partition the network.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated membership vector and the corresponding codelength
          in a tuple.</dd>
  </dl>
<div class="fields">      <strong>Reference:</strong>
      <ul class="nomargin-top">
        <li>
        M. Rosvall and C. T. Bergstrom: <i>Maps of information flow reveal 
        community structure in complex networks</i>. PNAS 105, 1118 (2008).
        <a href="http://arxiv.org/abs/0707.0609" 
        target="_top">http://arxiv.org/abs/0707.0609</a>
        </li>
        <li>
        M. Rosvall, D. Axelsson and C. T. Bergstrom: <i>The map 
        equation</i>. Eur Phys J Special Topics 178, 13 (2009). <a 
        href="http://arxiv.org/abs/0906.1405" 
        target="_top">http://arxiv.org/abs/0906.1405</a>
        </li>
      </ul>
</div></td></tr></table>
</div>
<a name="community_label_propagation"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_label_propagation</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">initial</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">fixed</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_label_propagation">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Finds the community structure of the graph according to the label 
  propagation method of Raghavan et al.</p>
  <p>Initially, each vertex is assigned a different label. After that, each
  vertex chooses the dominant label in its neighbourhood in each iteration.
  Ties are broken randomly and the order in which the vertices are updated 
  is randomized before every iteration. The algorithm ends when vertices 
  reach a consensus.</p>
  <p>Note that since ties are broken randomly, there is no guarantee that 
  the algorithm returns the same community structure after each run. In 
  fact, they frequently differ. See the paper of Raghavan et al on how to 
  come up with an aggregated community structure.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - name of an edge attribute or a list containing edge weights</li>
        <li><strong class="pname"><code>initial</code></strong> - name of a vertex attribute or a list containing the initial 
          vertex labels. Labels are identified by integers from zero to <i 
          class="math">n-1</i> where <i class="math">n</i> is the number of
          vertices. Negative numbers may also be present in this vector, 
          they represent unlabeled vertices.</li>
        <li><strong class="pname"><code>fixed</code></strong> - a list of booleans for each vertex. <code>True</code> corresponds
          to vertices whose labeling should not change during the 
          algorithm. It only makes sense if initial labels are also given. 
          Unlabeled vertices cannot be fixed. Note that vertex attribute 
          names are not accepted here.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the resulting membership vector</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Raghavan, U.N. and Albert, R. and Kumara, S. Near linear time 
        algorithm to detect community structures in large-scale networks. 
        Phys Rev E 76:036106, 2007. <a 
        href="http://arxiv.org/abs/0709.2938" 
        target="_top">http://arxiv.org/abs/0709.2938</a>.
      </p>
</div></td></tr></table>
</div>
<a name="community_leading_eigenvector"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_leading_eigenvector</span>(<span class="sig-arg">n</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">arpack_options</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_leading_eigenvector">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>A proper implementation of Newman's eigenvector community structure 
  detection. Each split is done by maximizing the modularity regarding the 
  original network. See the reference for details.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the desired number of communities. If negative, the algorithm 
          tries to do as many splits as possible. Note that the algorithm 
          won't split a community further if the signs of the leading 
          eigenvector are all the same.</li>
        <li><strong class="pname"><code>arpack_options</code></strong> - an <a href="igraph.ARPACKOptions-class.html" 
          class="link">ARPACKOptions</a> object used to fine-tune the 
          ARPACK eigenvector calculation. If omitted, the module-level 
          variable called <code>arpack_options</code> is used.</li>
        <li><strong class="pname"><code>weights</code></strong> - name of an edge attribute or a list containing edge weights</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a tuple where the first element is the membership vector of the 
          clustering and the second element is the merge matrix.</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function is wrapped in a more convenient syntax in the derived
        class <a href="igraph.Graph-class.html" class="link">Graph</a>. It 
        is advised to use that instead of this version.
      </p>
      <p><strong>Reference:</strong>
        MEJ Newman: Finding community structure in networks using the 
        eigenvectors of matrices, arXiv:physics/0605087
      </p>
</div></td></tr></table>
</div>
<a name="community_multilevel"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_multilevel</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">return_levels</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_multilevel">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Finds the community structure of the graph according to the multilevel
  algorithm of Blondel et al. This is a bottom-up algorithm: initially 
  every vertex belongs to a separate community, and vertices are moved 
  between communities iteratively in a way that maximizes the vertices' 
  local contribution to the overall modularity score. When a consensus is 
  reached (i.e. no single move would increase the modularity score), every 
  community in the original graph is shrank to a single vertex (while 
  keeping the total weight of the incident edges) and the process continues
  on the next level. The algorithm stops when it is not possible to 
  increase the modularity any more after shrinking the communities to 
  vertices.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - name of an edge attribute or a list containing edge weights</li>
        <li><strong class="pname"><code>return_levels</code></strong> - if <code>True</code>, returns the multilevel result. If 
          <code>False</code>, only the best level (corresponding to the 
          best modularity) is returned.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>either a single list describing the community membership of each 
          vertex (if <code>return_levels</code> is <code>False</code>), or 
          a list of community membership vectors, one corresponding to each
          level and a list of corresponding modularities (if 
          <code>return_levels</code> is <code>True</code>).</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function is wrapped in a more convenient syntax in the derived
        class <a href="igraph.Graph-class.html" class="link">Graph</a>. It 
        is advised to use that instead of this version.
      </p>
      <p><strong>Reference:</strong>
        VD Blondel, J-L Guillaume, R Lambiotte and E Lefebvre: Fast 
        unfolding of community hierarchies in large networks. J Stat Mech 
        P10008 (2008), http://arxiv.org/abs/0803.0476
      </p>
      <p><strong>See Also:</strong>
        modularity()
      </p>
</div></td></tr></table>
</div>
<a name="community_optimal_modularity"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_optimal_modularity</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_optimal_modularity">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the optimal modularity score of the graph and the 
  corresponding community structure.</p>
  <p>This function uses the GNU Linear Programming Kit to solve a large 
  integer optimization problem in order to find the optimal modularity 
  score and the corresponding community structure, therefore it is unlikely
  to work for graphs larger than a few (less than a hundred) vertices. 
  Consider using one of the heuristic approaches instead if you have such a
  large graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - name of an edge attribute or a list containing edge weights.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated membership vector and the corresponding modularity
          in a tuple.</dd>
  </dl>
</td></tr></table>
</div>
<a name="community_spinglass"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_spinglass</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">spins</span>=<span class="sig-default">25</span>,
        <span class="sig-arg">parupdate</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">start_temp</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">stop_temp</span>=<span class="sig-default">0.01</span>,
        <span class="sig-arg">cool_fact</span>=<span class="sig-default">0.99</span>,
        <span class="sig-arg">update_rule</span>=<span class="sig-default">&quot;config&quot;</span>,
        <span class="sig-arg">gamma</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">implementation</span>=<span class="sig-default">&quot;orig&quot;</span>,
        <span class="sig-arg">lambda</span>=<span class="sig-default">1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_spinglass">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Finds the community structure of the graph according to the spinglass 
  community detection method of Reichardt &amp; Bornholdt.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
        <li><strong class="pname"><code>spins</code></strong> - integer, the number of spins to use. This is the upper limit for 
          the number of communities. It is not a problem to supply a 
          (reasonably) big number here, in which case some spin states will
          be unpopulated.</li>
        <li><strong class="pname"><code>parupdate</code></strong> - whether to update the spins of the vertices in parallel 
          (synchronously) or not</li>
        <li><strong class="pname"><code>start_temp</code></strong> - the starting temperature</li>
        <li><strong class="pname"><code>stop_temp</code></strong> - the stop temperature</li>
        <li><strong class="pname"><code>cool_fact</code></strong> - cooling factor for the simulated annealing</li>
        <li><strong class="pname"><code>update_rule</code></strong> - specifies the null model of the simulation. Possible values are 
          <code>&quot;config&quot;</code> (a random graph with the same 
          vertex degrees as the input graph) or 
          <code>&quot;simple&quot;</code> (a random graph with the same 
          number of edges)</li>
        <li><strong class="pname"><code>gamma</code></strong> - the gamma argument of the algorithm, specifying the balance 
          between the importance of present and missing edges within a 
          community. The default value of 1.0 assigns equal importance to 
          both of them.</li>
        <li><strong class="pname"><code>implementation</code></strong> - currently igraph contains two implementations for the spinglass 
          community detection algorithm. The faster original implementation
          is the default. The other implementation is able to take into 
          account negative weights, this can be chosen by setting 
          <code>implementation</code> to <code>&quot;neg&quot;</code>.</li>
        <li><strong class="pname"><code>lambda</code></strong> - the lambda argument of the algorithm, which specifies the balance
          between the importance of present and missing negatively weighted
          edges within a community. Smaller values of lambda lead to 
          communities with less negative intra-connectivity. If the 
          argument is zero, the algorithm reduces to a graph coloring 
          algorithm, using the number of spins as colors. This argument is 
          ignored if the original implementation is used.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the community membership vector.</dd>
  </dl>
</td></tr></table>
</div>
<a name="community_walktrap"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">community_walktrap</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">steps</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.community_walktrap">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Finds the community structure of the graph according to the random 
  walk method of Latapy &amp; Pons.</p>
  <p>The basic idea of the algorithm is that short random walks tend to 
  stay in the same community. The method provides a dendrogram.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - name of an edge attribute or a list containing edge weights</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a tuple with the list of merges and the modularity scores 
          corresponding to each merge</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function is wrapped in a more convenient syntax in the derived
        class <a href="igraph.Graph-class.html" class="link">Graph</a>. It 
        is advised to use that instead of this version.
      </p>
      <p><strong>Reference:</strong>
        Pascal Pons, Matthieu Latapy: Computing communities in large 
        networks using random walks, <a 
        href="http://arxiv.org/abs/physics/0512106" 
        target="_top">http://arxiv.org/abs/physics/0512106</a>.
      </p>
      <p><strong>See Also:</strong>
        modularity()
      </p>
</div></td></tr></table>
</div>
<a name="complementer"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">complementer</span>(<span class="sig-arg">loops</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.complementer">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the complementer of the graph</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>loops</code></strong> - whether to include loop edges in the complementer.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the complementer of the graph</dd>
  </dl>
</td></tr></table>
</div>
<a name="constraint"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">constraint</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.constraint">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates Burt's constraint scores for given vertices in a graph.</p>
  <p>Burt's constraint is higher if ego has less, or mutually stronger 
  related (i.e. more redundant) contacts. Burt's measure of constraint, 
  C[i], of vertex i's ego network V[i], is defined for directed and valued 
  graphs as follows:</p>
  <p>C[i] = sum( sum( (p[i,q] p[q,j])^2, q in V[i], q != i,j ), j in V[], j
  != i)</p>
  <p>for a graph of order (ie. number od vertices) N, where proportional 
  tie strengths are defined as follows:</p>
  <p>p[i,j]=(a[i,j]+a[j,i]) / sum(a[i,k]+a[k,i], k in V[i], k != i), a[i,j]
  are elements of A and the latter being the graph adjacency matrix.</p>
  <p>For isolated vertices, constraint is undefined.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices to be analysed or <code>None</code> for all 
          vertices.</li>
        <li><strong class="pname"><code>weights</code></strong> - weights associated to the edges. Can be an attribute name as 
          well. If <code>None</code>, every edge will have the same weight.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>constraint scores for all given vertices in a matrix.</dd>
  </dl>
</td></tr></table>
</div>
<a name="contract_vertices"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">contract_vertices</span>(<span class="sig-arg">mapping</span>,
        <span class="sig-arg">combine_attrs</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.contract_vertices">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Contracts some vertices in the graph, i.e. replaces groups of vertices
  with single vertices. Edges are not affected.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mapping</code></strong> - numeric vector which gives the mapping between old and new vertex
          IDs. Vertices having the same new vertex ID in this vector will 
          be remapped into a single new vertex. It is safe to pass the 
          membership vector of a <a 
          href="igraph.clustering.VertexClustering-class.html" 
          class="link">VertexClustering</a> object here.</li>
        <li><strong class="pname"><code>combine_attrs</code></strong> - specifies how to combine the attributes of the vertices being 
          collapsed into a single one. If it is <code>None</code>, all the 
          attributes will be lost. If it is a function, the attributes of 
          the vertices will be collected and passed on to that function 
          which will return the new attribute value that has to be assigned
          to the single collapsed vertex. It can also be one of the 
          following string constants which define built-in collapsing 
          functions: <code>sum</code>, <code>prod</code>, 
          <code>mean</code>, <code>median</code>, <code>max</code>, 
          <code>min</code>, <code>first</code>, <code>last</code>, 
          <code>random</code>. You can also specify different combination 
          functions for different attributes by passing a dict here which 
          maps attribute names to functions. See <a 
          href="igraph.GraphBase-class.html#simplify" 
          class="link">Graph.simplify()</a> for more details.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd><code>None</code>.</dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.GraphBase-class.html#simplify" 
        class="link">Graph.simplify()</a>
      </p>
</div></td></tr></table>
</div>
<a name="coreness"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">coreness</span>(<span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.shell_index">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Finds the coreness (shell index) of the vertices of the network.</p>
  <p>The <i class="math">k</i>-core of a graph is a maximal subgraph in 
  which each vertex has at least degree k. (Degree here means the degree in
  the subgraph of course). The coreness of a vertex is <i 
  class="math">k</i> if it is a member of the <i class="math">k</i>-core 
  but not a member of the <i class="math">k+1</i>-core.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - whether to compute the in-corenesses (<a 
          href="igraph-module.html#IN" class="link">IN</a>), the 
          out-corenesses (<a href="igraph-module.html#OUT" 
          class="link">OUT</a>) or the undirected corenesses (<a 
          href="igraph-module.html#ALL" class="link">ALL</a>). Ignored and 
          assumed to be <a href="igraph-module.html#ALL" 
          class="link">ALL</a> for undirected graphs.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the corenesses for each vertex.</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Vladimir Batagelj, Matjaz Zaversnik: <i>An <i class="math">O(m)</i>
        Algorithm for Core Decomposition of Networks.</i>
      </p>
</div></td></tr></table>
</div>
<a name="count_isomorphisms_vf2"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">count_isomorphisms_vf2</span>(<span class="sig-arg">other</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">node_compat_fn</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_compat_fn</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.count_isomorphisms_vf2">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Determines the number of isomorphisms between the graph and another 
  one</p>
  <p>Vertex and edge colors may be used to restrict the isomorphisms, as 
  only vertices and edges with the same color will be allowed to match each
  other.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the other graph. If <code>None</code>, the number of 
          automorphisms will be returned.</li>
        <li><strong class="pname"><code>color1</code></strong> - optional vector storing the coloring of the vertices of the first
          graph. If <code>None</code>, all vertices have the same color.</li>
        <li><strong class="pname"><code>color2</code></strong> - optional vector storing the coloring of the vertices of the 
          second graph. If <code>None</code>, all vertices have the same 
          color.</li>
        <li><strong class="pname"><code>edge_color1</code></strong> - optional vector storing the coloring of the edges of the first 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>edge_color2</code></strong> - optional vector storing the coloring of the edges of the second 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>node_compat_fn</code></strong> - a function that receives the two graphs and two node indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the nodes given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on node-specific criteria that are too 
          complicated to be represented by node color vectors (i.e. the 
          <code>color1</code> and <code>color2</code> parameters). 
          <code>None</code> means that every node is compatible with every 
          other node.</li>
        <li><strong class="pname"><code>edge_compat_fn</code></strong> - a function that receives the two graphs and two edge indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the edges given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on edge-specific criteria that are too 
          complicated to be represented by edge color vectors (i.e. the 
          <code>edge_color1</code> and <code>edge_color2</code> 
          parameters). <code>None</code> means that every edge is 
          compatible with every other node.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the number of isomorphisms between the two given graphs (or the 
          number of automorphisms if <code>other</code> is 
          <code>None</code>.</dd>
  </dl>
</td></tr></table>
</div>
<a name="count_multiple"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">count_multiple</span>(<span class="sig-arg">edges</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.count_multiple">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Counts the multiplicities of the given edges.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>edges</code></strong> - edge indices for which we want to count their multiplicity. If 
          <code>None</code>, all edges are counted.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the multiplicities of the given edges as a list.</dd>
  </dl>
</td></tr></table>
</div>
<a name="count_subisomorphisms_vf2"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">count_subisomorphisms_vf2</span>(<span class="sig-arg">other</span>,
        <span class="sig-arg">color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">node_compat_fn</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_compat_fn</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.count_subisomorphisms_vf2">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Determines the number of subisomorphisms between the graph and another
  one</p>
  <p>Vertex and edge colors may be used to restrict the isomorphisms, as 
  only vertices and edges with the same color will be allowed to match each
  other.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the other graph.</li>
        <li><strong class="pname"><code>color1</code></strong> - optional vector storing the coloring of the vertices of the first
          graph. If <code>None</code>, all vertices have the same color.</li>
        <li><strong class="pname"><code>color2</code></strong> - optional vector storing the coloring of the vertices of the 
          second graph. If <code>None</code>, all vertices have the same 
          color.</li>
        <li><strong class="pname"><code>edge_color1</code></strong> - optional vector storing the coloring of the edges of the first 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>edge_color2</code></strong> - optional vector storing the coloring of the edges of the second 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>node_compat_fn</code></strong> - a function that receives the two graphs and two node indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the nodes given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on node-specific criteria that are too 
          complicated to be represented by node color vectors (i.e. the 
          <code>color1</code> and <code>color2</code> parameters). 
          <code>None</code> means that every node is compatible with every 
          other node.</li>
        <li><strong class="pname"><code>edge_compat_fn</code></strong> - a function that receives the two graphs and two edge indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the edges given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on edge-specific criteria that are too 
          complicated to be represented by edge color vectors (i.e. the 
          <code>edge_color1</code> and <code>edge_color2</code> 
          parameters). <code>None</code> means that every edge is 
          compatible with every other node.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the number of subisomorphisms between the two given graphs</dd>
  </dl>
</td></tr></table>
</div>
<a name="decompose"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">decompose</span>(<span class="sig-arg">mode</span>=<span class="sig-default">STRONG</span>,
        <span class="sig-arg">maxcompno</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">minelements</span>=<span class="sig-default">1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.decompose">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Decomposes the graph into subgraphs.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - must be either STRONG or WEAK, depending on the clusters being 
          sought.</li>
        <li><strong class="pname"><code>maxcompno</code></strong> - maximum number of components to return. <code>None</code> means 
          all possible components.</li>
        <li><strong class="pname"><code>minelements</code></strong> - minimum number of vertices in a component. By setting this to 2, 
          isolated vertices are not returned as separate components.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list of the subgraphs. Every returned subgraph is a copy of the
          original.</dd>
  </dl>
</td></tr></table>
</div>
<a name="degree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">degree</span>(<span class="sig-arg">vertices</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.degree">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns some vertex degrees from the graph.</p>
  <p>This method accepts a single vertex ID or a list of vertex IDs as a 
  parameter, and returns the degree of the given vertices (in the form of a
  single integer or a list, depending on the input parameter).</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a single vertex ID or a list of vertex IDs</li>
        <li><strong class="pname"><code>mode</code></strong> - the type of degree to be returned (<a 
          href="igraph-module.html#OUT" class="link">OUT</a> for 
          out-degrees, <a href="igraph-module.html#IN" class="link">IN</a> 
          IN for in-degrees or <a href="igraph-module.html#ALL" 
          class="link">ALL</a> for the sum of them).</li>
        <li><strong class="pname"><code>loops</code></strong> - whether self-loops should be counted.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="delete_edges"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">delete_edges</span>(<span class="sig-arg">es</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.delete_edges">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Removes edges from the graph.</p>
  <p>All vertices will be kept, even if they lose all their edges. 
  Nonexistent edges will be silently ignored.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>es</code></strong> - the list of edges to be removed. Edges are identifed by edge IDs.
          <a href="igraph.EdgeSeq-class.html" class="link">EdgeSeq</a> 
          objects are also accepted here.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="delete_vertices"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">delete_vertices</span>(<span class="sig-arg">vs</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.delete_vertices">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Deletes vertices and all its edges from the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vs</code></strong> - a single vertex ID or the list of vertex IDs to be deleted.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="density"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">density</span>(<span class="sig-arg">loops</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.density">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the density of the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>loops</code></strong> - whether to take loops into consideration. If <code>True</code>, 
          the algorithm assumes that there might be some loops in the graph
          and calculates the density accordingly. If <code>False</code>, 
          the algorithm assumes that there can't be any loops.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the reciprocity of the graph.</dd>
  </dl>
</td></tr></table>
</div>
<a name="diameter"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">diameter</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">unconn</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.diameter">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the diameter of the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to consider directed paths.</li>
        <li><strong class="pname"><code>unconn</code></strong> - if <code>True</code> and the graph is unconnected, the longest 
          geodesic within a component will be returned. If 
          <code>False</code> and the graph is unconnected, the result is 
          the number of vertices if there are no weights or infinity if 
          there are weights.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the diameter</dd>
  </dl>
</td></tr></table>
</div>
<a name="disjoint_union"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">disjoint_union</span>(<span class="sig-arg">graphs</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.disjoint_union">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Creates the disjoint union of two (or more) graphs.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>graphs</code></strong> - the list of graphs to be united with the current one.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="diversity"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">diversity</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.diversity">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the structural diversity index of the vertices.</p>
  <p>The structural diversity index of a vertex is simply the (normalized) 
  Shannon entropy of the weights of the edges incident on the vertex.</p>
  <p>The measure is defined for undirected graphs only; edge directions are
  ignored.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices for which the diversity indices must be returned. If
          <code>None</code>, uses all of the vertices in the graph.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated diversity indices in a list, or a single number if
          a single vertex was supplied.</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Eagle N, Macy M and Claxton R: Network diversity and economic 
        development, Science 328, 1029--1031, 2010.
      </p>
</div></td></tr></table>
</div>
<a name="dyad_census"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">dyad_census</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.dyad_census">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Dyad census, as defined by Holland and Leinhardt</p>
  <p>Dyad census means classifying each pair of vertices of a directed 
  graph into three categories: mutual, there is an edge from <i>a</i> to 
  <i>b</i> and also from <i>b</i> to <i>a</i>; asymmetric, there is an edge
  either from <i>a</i> to <i>b</i> or from <i>b</i> to <i>a</i> but not the
  other way and null, no edges between <i>a</i> and <i>b</i>.</p>
  <dl class="fields">
    <dt>Returns:</dt>
        <dd>the number of mutual, asymmetric and null connections in a 
          3-tuple.</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a <a href="igraph.datatypes.DyadCensus-class.html" 
        class="link">DyadCensus</a> object. It is advised to use that.
      </p>
</div></td></tr></table>
</div>
<a name="eccentricity"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">eccentricity</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.eccentricity">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the eccentricities of given vertices in a graph.</p>
  <p>The eccentricity of a vertex is calculated by measuring the shortest 
  distance from (or to) the vertex, to (or from) all other vertices in the 
  graph, and taking the maximum.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices for which the eccentricity scores must be returned. 
          If <code>None</code>, uses all of the vertices in the graph.</li>
        <li><strong class="pname"><code>mode</code></strong> - must be one of <a href="igraph-module.html#IN" 
          class="link">IN</a>, <a href="igraph-module.html#OUT" 
          class="link">OUT</a> and <a href="igraph-module.html#ALL" 
          class="link">ALL</a>. <a href="igraph-module.html#IN" 
          class="link">IN</a> means that edge directions are followed; 
          <code>OUT</code> means that edge directions are followed the 
          opposite direction; <code>ALL</code> means that directions are 
          ignored. The argument has no effect for undirected graphs.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated eccentricities in a list, or a single number if a 
          single vertex was supplied.</dd>
  </dl>
</td></tr></table>
</div>
<a name="ecount"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">ecount</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.ecount">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Counts the number of edges.</p>
  <dl class="fields">
    <dt>Returns: integer</dt>
        <dd>the number of edges in the graph.</dd>
  </dl>
</td></tr></table>
</div>
<a name="edge_attributes"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">edge_attributes</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.edge_attributes">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  
  <dl class="fields">
    <dt>Returns:</dt>
        <dd>the attribute name list of the graph's edges</dd>
  </dl>
</td></tr></table>
</div>
<a name="edge_betweenness"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">edge_betweenness</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">cutoff</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.edge_betweenness">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates or estimates the edge betweennesses in a graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to consider directed paths.</li>
        <li><strong class="pname"><code>cutoff</code></strong> - if it is an integer, only paths less than or equal to this length
          are considered, effectively resulting in an estimation of the 
          betweenness values. If <code>None</code>, the exact betweennesses
          are returned.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list with the (exact or estimated) edge betweennesses of all 
          edges.</dd>
  </dl>
</td></tr></table>
</div>
<a name="edge_connectivity"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">edge_connectivity</span>(<span class="sig-arg">source</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">target</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">checks</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.edge_disjoint_paths">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the edge connectivity of the graph or between some 
  vertices.</p>
  <p>The edge connectivity between two given vertices is the number of 
  edges that have to be removed in order to disconnect the two vertices 
  into two separate components. This is also the number of edge disjoint 
  directed paths between the vertices. The edge connectivity of the graph 
  is the minimal edge connectivity over all vertex pairs.</p>
  <p>This method calculates the edge connectivity of a given vertex pair if
  both the source and target vertices are given. If none of them is given 
  (or they are both negative), the overall edge connectivity is 
  returned.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex involved in the calculation.</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex involved in the calculation.</li>
        <li><strong class="pname"><code>checks</code></strong> - if the whole graph connectivity is calculated and this is 
          <code>True</code>, igraph performs some basic checks before 
          calculation. If the graph is not strongly connected, then the 
          connectivity is obviously zero. If the minimum degree is one, 
          then the connectivity is also one. These simple checks are much 
          faster than checking the entire graph, therefore it is advised to
          set this to <code>True</code>. The parameter is ignored if the 
          connectivity between two given vertices is computed.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the edge connectivity</dd>
  </dl>
</td></tr></table>
</div>
<a name="eigenvector_centrality"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">eigenvector_centrality</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">scale</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">return_eigenvalue</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">arpack_options</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.evcent">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the eigenvector centralities of the vertices in a 
  graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to consider edge directions in a directed graph. Ignored 
          for undirected graphs.</li>
        <li><strong class="pname"><code>scale</code></strong> - whether to normalize the centralities so the largest one will 
          always be 1.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights given as a list or an edge attribute. If 
          <code>None</code>, all edges have equal weight.</li>
        <li><strong class="pname"><code>return_eigenvalue</code></strong> - whether to return the actual largest eigenvalue along with the 
          centralities</li>
        <li><strong class="pname"><code>arpack_options</code></strong> - an <a href="igraph.ARPACKOptions-class.html" 
          class="link">ARPACKOptions</a> object that can be used to 
          fine-tune the calculation. If it is omitted, the module-level 
          variable called <code>arpack_options</code> is used.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the eigenvector centralities in a list and optionally the largest
          eigenvalue (as a second member of a tuple)</dd>
  </dl>
</td></tr></table>
</div>
<a name="farthest_points"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">farthest_points</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">unconn</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.farthest_points">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns two vertex IDs whose distance equals the actual diameter of 
  the graph.</p>
  <p>If there are many shortest paths with the length of the diameter, it 
  returns the first one it found.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to consider directed paths.</li>
        <li><strong class="pname"><code>unconn</code></strong> - if <code>True</code> and the graph is unconnected, the longest 
          geodesic within a component will be returned. If 
          <code>False</code> and the graph is unconnected, the result 
          contains the number of vertices if there are no weights or 
          infinity if there are weights.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a triplet containing the two vertex IDs and their distance. The 
          IDs are <code>None</code> if the graph is unconnected and 
          <code>unconn</code> is <code>False</code>.</dd>
  </dl>
</td></tr></table>
</div>
<a name="feedback_arc_set"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">feedback_arc_set</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">method</span>=<span class="sig-default">&quot;eades&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.feedback_arc_set">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates an approximately or exactly minimal feedback arc set.</p>
  <p>A feedback arc set is a set of edges whose removal makes the graph 
  acyclic. Since this is always possible by removing all the edges, we are 
  in general interested in removing the smallest possible number of edges, 
  or an edge set with as small total weight as possible. This method 
  calculates one such edge set. Note that the task is trivial for an 
  undirected graph as it is enough to find a spanning tree and then remove 
  all the edges not in the spanning tree. Of course it is more complicated 
  for directed graphs.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name. When given, the algorithm will strive to 
          remove lightweight edges in order to minimize the total weight of
          the feedback arc set.</li>
        <li><strong class="pname"><code>method</code></strong> - the algorithm to use. <code>&quot;eades&quot;</code> uses the 
          greedy cycle breaking heuristic of Eades, Lin and Smyth, which is
          linear in the number of edges but not necessarily optimal; 
          however, it guarantees that the number of edges to be removed is 
          smaller than |E|/2 - |V|/6. <code>&quot;ip&quot;</code> uses an 
          integer programming formulation which is guaranteed to yield an 
          optimal result, but is too slow for large graphs.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the IDs of the edges to be removed, in a list.</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Eades P, Lin X and Smyth WF: A fast and effective heuristic for the
        feedback arc set problem. In: Proc Inf Process Lett 319-323, 1993.
      </p>
</div></td></tr></table>
</div>
<a name="get_adjacency"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_adjacency</span>(<span class="sig-arg">type</span>=<span class="sig-default">GET_ADJACENCY_BOTH</span>,
        <span class="sig-arg">eids</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_adjacency">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the adjacency matrix of a graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>type</code></strong> - either <code>GET_ADJACENCY_LOWER</code> (uses the lower triangle 
          of the matrix) or <code>GET_ADJACENCY_UPPER</code> (uses the 
          upper triangle) or <code>GET_ADJACENCY_BOTH</code> (uses both 
          parts). Ignored for directed graphs.</li>
        <li><strong class="pname"><code>eids</code></strong> - if <code>True</code>, the result matrix will contain zeros for 
          non-edges and the ID of the edge plus one for edges in the 
          appropriate cell. If <code>False</code>, the result matrix will 
          contain the number of edges for each vertex pair.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the adjacency matrix.</dd>
  </dl>
</td></tr></table>
</div>
<a name="get_all_shortest_paths"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_all_shortest_paths</span>(<span class="sig-arg">v</span>,
        <span class="sig-arg">to</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_all_shortest_paths">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates all of the shortest paths from/to a given node in a 
  graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>v</code></strong> - the source/destination for the calculated paths</li>
        <li><strong class="pname"><code>to</code></strong> - a vertex selector describing the destination/source for the 
          calculated paths. This can be a single vertex ID, a list of 
          vertex IDs, a single vertex name, a list of vertex names or a <a 
          href="igraph.VertexSeq-class.html" class="link">VertexSeq</a> 
          object. <code>None</code> means all the vertices.</li>
        <li><strong class="pname"><code>mode</code></strong> - the directionality of the paths. <a href="igraph-module.html#IN" 
          class="link">IN</a> means to calculate incoming paths, <a 
          href="igraph-module.html#OUT" class="link">OUT</a> means to 
          calculate outgoing paths, <a href="igraph-module.html#ALL" 
          class="link">ALL</a> means to calculate both ones.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>all of the shortest path from the given node to every other 
          reachable node in the graph in a list. Note that in case of 
          mode=<a href="igraph-module.html#IN" class="link">IN</a>, the 
          vertices in a path are returned in reversed order!</dd>
  </dl>
</td></tr></table>
</div>
<a name="get_diameter"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_diameter</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">unconn</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_diameter">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a path with the actual diameter of the graph.</p>
  <p>If there are many shortest paths with the length of the diameter, it 
  returns the first one it founds.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to consider directed paths.</li>
        <li><strong class="pname"><code>unconn</code></strong> - if <code>True</code> and the graph is unconnected, the longest 
          geodesic within a component will be returned. If 
          <code>False</code> and the graph is unconnected, the result is 
          the number of vertices if there are no weights or infinity if 
          there are weights.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the vertices in the path in order.</dd>
  </dl>
</td></tr></table>
</div>
<a name="get_eid"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_eid</span>(<span class="sig-arg">v1</span>,
        <span class="sig-arg">v2</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">error</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_eid">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the edge ID of an arbitrary edge between vertices v1 and 
  v2</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>v1</code></strong> - the first vertex ID</li>
        <li><strong class="pname"><code>v2</code></strong> - the second vertex ID</li>
        <li><strong class="pname"><code>directed</code></strong> - whether edge directions should be considered in directed graphs. 
          The default is <code>True</code>. Ignored for undirected graphs.</li>
        <li><strong class="pname"><code>error</code></strong> - if <code>True</code>, an exception will be raised when the given 
          edge does not exist. If <code>False</code>, -1 will be returned 
          in that case.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the edge ID of an arbitrary edge between vertices v1 and v2</dd>
  </dl>
</td></tr></table>
</div>
<a name="get_eids"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_eids</span>(<span class="sig-arg">pairs</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">path</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">error</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_eids">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the edge IDs of some edges between some vertices.</p>
  <p>This method can operate in two different modes, depending on which of 
  the keyword arguments <code>pairs</code> and <code>path</code> are 
  given.</p>
  <p>The method does not consider multiple edges; if there are multiple 
  edges between a pair of vertices, only the ID of one of the edges is 
  returned.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>pairs</code></strong> - a list of integer pairs. Each integer pair is considered as a 
          source-target vertex pair; the corresponding edge is looked up in
          the graph and the edge ID is returned for each pair.</li>
        <li><strong class="pname"><code>path</code></strong> - a list of vertex IDs. The list is considered as a continuous path
          from the first vertex to the last, passing through the 
          intermediate vertices. The corresponding edge IDs between the 
          first and the second, the second and the third and so on are 
          looked up in the graph and the edge IDs are returned. If both 
          <code>path</code> and <code>pairs</code> are given, the two lists
          are concatenated.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether edge directions should be considered in directed graphs. 
          The default is <code>True</code>. Ignored for undirected graphs.</li>
        <li><strong class="pname"><code>error</code></strong> - if <code>True</code>, an exception will be raised if a given edge
          does not exist. If <code>False</code>, -1 will be returned in 
          that case.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the edge IDs in a list</dd>
  </dl>
</td></tr></table>
</div>
<a name="get_incidence"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_incidence</span>(<span class="sig-arg">types</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_incidence">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Internal function, undocumented.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        Graph.get_incidence()
      </p>
</div></td></tr></table>
</div>
<a name="get_isomorphisms_vf2"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_isomorphisms_vf2</span>(<span class="sig-arg">other</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">node_compat_fn</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_compat_fn</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_isomorphisms_vf2">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns all isomorphisms between the graph and another one</p>
  <p>Vertex and edge colors may be used to restrict the isomorphisms, as 
  only vertices and edges with the same color will be allowed to match each
  other.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the other graph. If <code>None</code>, the automorphisms will be 
          returned.</li>
        <li><strong class="pname"><code>color1</code></strong> - optional vector storing the coloring of the vertices of the first
          graph. If <code>None</code>, all vertices have the same color.</li>
        <li><strong class="pname"><code>color2</code></strong> - optional vector storing the coloring of the vertices of the 
          second graph. If <code>None</code>, all vertices have the same 
          color.</li>
        <li><strong class="pname"><code>edge_color1</code></strong> - optional vector storing the coloring of the edges of the first 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>edge_color2</code></strong> - optional vector storing the coloring of the edges of the second 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>node_compat_fn</code></strong> - a function that receives the two graphs and two node indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the nodes given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on node-specific criteria that are too 
          complicated to be represented by node color vectors (i.e. the 
          <code>color1</code> and <code>color2</code> parameters). 
          <code>None</code> means that every node is compatible with every 
          other node.</li>
        <li><strong class="pname"><code>edge_compat_fn</code></strong> - a function that receives the two graphs and two edge indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the edges given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on edge-specific criteria that are too 
          complicated to be represented by edge color vectors (i.e. the 
          <code>edge_color1</code> and <code>edge_color2</code> 
          parameters). <code>None</code> means that every edge is 
          compatible with every other node.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list of lists, each item of the list containing the mapping 
          from vertices of the second graph to the vertices of the first 
          one</dd>
  </dl>
</td></tr></table>
</div>
<a name="get_shortest_paths"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_shortest_paths</span>(<span class="sig-arg">v</span>,
        <span class="sig-arg">to</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>,
        <span class="sig-arg">output</span>=<span class="sig-default">&quot;vpath&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_shortest_paths">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the shortest paths from/to a given node in a graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>v</code></strong> - the source/destination for the calculated paths</li>
        <li><strong class="pname"><code>to</code></strong> - a vertex selector describing the destination/source for the 
          calculated paths. This can be a single vertex ID, a list of 
          vertex IDs, a single vertex name, a list of vertex names or a <a 
          href="igraph.VertexSeq-class.html" class="link">VertexSeq</a> 
          object. <code>None</code> means all the vertices.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights in a list or the name of an edge attribute holding 
          edge weights. If <code>None</code>, all edges are assumed to have
          equal weight.</li>
        <li><strong class="pname"><code>mode</code></strong> - the directionality of the paths. <a href="igraph-module.html#IN" 
          class="link">IN</a> means to calculate incoming paths, <a 
          href="igraph-module.html#OUT" class="link">OUT</a> means to 
          calculate outgoing paths, <a href="igraph-module.html#ALL" 
          class="link">ALL</a> means to calculate both ones.</li>
        <li><strong class="pname"><code>output</code></strong> - determines what should be returned. If this is 
          <code>&quot;vpath&quot;</code>, a list of vertex IDs will be 
          returned, one path for each target vertex. For unconnected 
          graphs, some of the list elements may be empty. Note that in case
          of mode=<a href="igraph-module.html#IN" class="link">IN</a>, the 
          vertices in a path are returned in reversed order. If 
          <code>output=&quot;epath&quot;</code>, edge IDs are returned 
          instead of vertex IDs.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>see the documentation of the <code>output</code> parameter.</dd>
  </dl>
</td></tr></table>
</div>
<a name="get_subisomorphisms_lad"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_subisomorphisms_lad</span>(<span class="sig-arg">other</span>,
        <span class="sig-arg">domains</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">induced</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">time_limit</span>=<span class="sig-default">0</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_subisomorphisms_lad">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns all subisomorphisms between the graph and another one using 
  the LAD algorithm.</p>
  <p>The optional <code>domains</code> argument may be used to restrict 
  vertices that may match each other. You can also specify whether you are 
  interested in induced subgraphs only or not.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the pattern graph we are looking for in the graph.</li>
        <li><strong class="pname"><code>domains</code></strong> - a list of lists, one sublist belonging to each vertex in the 
          template graph. Sublist <i class="math">i</i> contains the 
          indices of the vertices in the original graph that may match 
          vertex <i class="math">i</i> in the template graph. 
          <code>None</code> means that every vertex may match every other 
          vertex.</li>
        <li><strong class="pname"><code>induced</code></strong> - whether to consider induced subgraphs only.</li>
        <li><strong class="pname"><code>time_limit</code></strong> - an optimal time limit in seconds. Only the integral part of this 
          number is taken into account. If the time limit is exceeded, the 
          method will throw an exception.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list of lists, each item of the list containing the mapping 
          from vertices of the second graph to the vertices of the first 
          one</dd>
  </dl>
</td></tr></table>
</div>
<a name="get_subisomorphisms_vf2"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">get_subisomorphisms_vf2</span>(<span class="sig-arg">other</span>,
        <span class="sig-arg">color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">node_compat_fn</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_compat_fn</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.get_subisomorphisms_vf2">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns all subisomorphisms between the graph and another one</p>
  <p>Vertex and edge colors may be used to restrict the isomorphisms, as 
  only vertices and edges with the same color will be allowed to match each
  other.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the other graph.</li>
        <li><strong class="pname"><code>color1</code></strong> - optional vector storing the coloring of the vertices of the first
          graph. If <code>None</code>, all vertices have the same color.</li>
        <li><strong class="pname"><code>color2</code></strong> - optional vector storing the coloring of the vertices of the 
          second graph. If <code>None</code>, all vertices have the same 
          color.</li>
        <li><strong class="pname"><code>edge_color1</code></strong> - optional vector storing the coloring of the edges of the first 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>edge_color2</code></strong> - optional vector storing the coloring of the edges of the second 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>node_compat_fn</code></strong> - a function that receives the two graphs and two node indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the nodes given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on node-specific criteria that are too 
          complicated to be represented by node color vectors (i.e. the 
          <code>color1</code> and <code>color2</code> parameters). 
          <code>None</code> means that every node is compatible with every 
          other node.</li>
        <li><strong class="pname"><code>edge_compat_fn</code></strong> - a function that receives the two graphs and two edge indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the edges given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on edge-specific criteria that are too 
          complicated to be represented by edge color vectors (i.e. the 
          <code>edge_color1</code> and <code>edge_color2</code> 
          parameters). <code>None</code> means that every edge is 
          compatible with every other node.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list of lists, each item of the list containing the mapping 
          from vertices of the second graph to the vertices of the first 
          one</dd>
  </dl>
</td></tr></table>
</div>
<a name="girth"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">girth</span>(<span class="sig-arg">return_shortest_circle</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.girth">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the girth of the graph.</p>
  <p>The girth of a graph is the length of the shortest circle in it.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>return_shortest_circle</code></strong> - whether to return one of the shortest circles found in the graph.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the length of the shortest circle or (if 
          <code>return_shortest_circle</code>) is true, the shortest circle
          itself as a list</dd>
  </dl>
</td></tr></table>
</div>
<a name="gomory_hu_tree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">gomory_hu_tree</span>(<span class="sig-arg">capacity</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.gomory_hu_tree">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Internal function, undocumented.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        Graph.gomory_hu_tree()
      </p>
</div></td></tr></table>
</div>
<a name="has_multiple"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">has_multiple</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.has_multiple">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether the graph has multiple edges.</p>
  <dl class="fields">
    <dt>Returns: boolean</dt>
        <dd><code>True</code> if the graph has at least one multiple edge, 
          <code>False</code> otherwise.</dd>
  </dl>
</td></tr></table>
</div>
<a name="hub_score"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">hub_score</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">scale</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">arpack_options</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">return_eigenvalue</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.hub_score">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates Kleinberg's hub score for the vertices of the graph</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
        <li><strong class="pname"><code>scale</code></strong> - whether to normalize the scores so that the largest one is 1.</li>
        <li><strong class="pname"><code>arpack_options</code></strong> - an <a href="igraph.ARPACKOptions-class.html" 
          class="link">ARPACKOptions</a> object used to fine-tune the 
          ARPACK eigenvector calculation. If omitted, the module-level 
          variable called <code>arpack_options</code> is used.</li>
        <li><strong class="pname"><code>return_eigenvalue</code></strong> - whether to return the largest eigenvalue</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the hub scores in a list and optionally the largest eigenvalue as
          a second member of a tuple</dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        authority_score()
      </p>
</div></td></tr></table>
</div>
<a name="incident"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">incident</span>(<span class="sig-arg">vertex</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.incident">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the edges a given vertex is incident on.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertex</code></strong> - a vertex ID</li>
        <li><strong class="pname"><code>mode</code></strong> - whether to return only predecessors (<a 
          href="igraph-module.html#OUT" class="link">OUT</a>), successors 
          (<a href="igraph-module.html#IN" class="link">IN</a>) or both (<a
          href="igraph-module.html#ALL" class="link">ALL</a>). Ignored for 
          undirected graphs.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="independence_number"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">independence_number</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.alpha">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the independence number of the graph.</p>
  <p>The independence number of the graph is the size of the largest 
  independent vertex set.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a 
        href="igraph.GraphBase-class.html#largest_independent_vertex_sets" 
        class="link">largest_independent_vertex_sets()</a> for the largest 
        independent vertex sets
      </p>
</div></td></tr></table>
</div>
<a name="independent_vertex_sets"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">independent_vertex_sets</span>(<span class="sig-arg">min</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">max</span>=<span class="sig-default">0</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.independent_vertex_sets">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns some or all independent vertex sets of the graph as a list of 
  tuples.</p>
  <p>Two vertices are independent if there is no edge between them. Members
  of an independent vertex set are mutually independent.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>min</code></strong> - the minimum size of sets to be returned. If zero or negative, no 
          lower bound will be used.</li>
        <li><strong class="pname"><code>max</code></strong> - the maximum size of sets to be returned. If zero or negative, no 
          upper bound will be used.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="induced_subgraph"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">induced_subgraph</span>(<span class="sig-arg">vertices</span>,
        <span class="sig-arg">implementation</span>=<span class="sig-default">&quot;auto&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.subgraph">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a subgraph spanned by the given vertices.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a list containing the vertex IDs which should be included in the 
          result.</li>
        <li><strong class="pname"><code>implementation</code></strong> - the implementation to use when constructing the new subgraph. 
          igraph includes two implementations at the moment. 
          <code>&quot;copy_and_delete&quot;</code> copies the original 
          graph and removes those vertices that are not in the given set. 
          This is more efficient if the size of the subgraph is comparable 
          to the original graph. The other implementation 
          (<code>&quot;create_from_scratch&quot;</code>) constructs the 
          result graph from scratch and then copies the attributes 
          accordingly. This is a better solution if the subgraph is 
          relatively small, compared to the original graph. 
          <code>&quot;auto&quot;</code> selects between the two 
          implementations automatically, based on the ratio of the size of 
          the subgraph and the size of the original graph.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the subgraph</dd>
  </dl>
</td></tr></table>
</div>
<a name="intersection"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">intersection</span>(<span class="sig-arg">graphs</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.intersection">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Creates the intersection of two (or more) graphs.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>graphs</code></strong> - the list of graphs to be intersected with the current one.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="is_bipartite"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_bipartite</span>(<span class="sig-arg">return_types</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_bipartite">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Decides whether the graph is bipartite or not.</p>
  <p>Vertices of a bipartite graph can be partitioned into two groups A and
  B in a way that all edges go between the two groups.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>return_types</code></strong> - if <code>False</code>, the method will simply return 
          <code>True</code> or <code>False</code> depending on whether the 
          graph is bipartite or not. If <code>True</code>, the actual group
          assignments are also returned as a list of boolean values. (Note 
          that the group assignment is not unique, especially if the graph 
          consists of multiple components, since the assignments of 
          components are independent from each other).</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd><code>True</code> if the graph is bipartite, <code>False</code> 
          if not. If <code>return_types</code> is <code>True</code>, the 
          group assignment is also returned.</dd>
  </dl>
</td></tr></table>
</div>
<a name="is_connected"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_connected</span>(<span class="sig-arg">mode</span>=<span class="sig-default">STRONG</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_connected">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Decides whether the graph is connected.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - whether we should calculate strong or weak connectivity.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd><code>True</code> if the graph is connected, <code>False</code> 
          otherwise.</dd>
  </dl>
</td></tr></table>
</div>
<a name="is_dag"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_dag</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_dag">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether the graph is a DAG (directed acyclic graph).</p>
  <p>A DAG is a directed graph with no directed cycles.</p>
  <dl class="fields">
    <dt>Returns: boolean</dt>
        <dd><code>True</code> if it is a DAG, <code>False</code> otherwise.</dd>
  </dl>
</td></tr></table>
</div>
<a name="is_loop"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_loop</span>(<span class="sig-arg">edges</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_loop">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether a specific set of edges contain loop edges</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>edges</code></strong> - edge indices which we want to check. If <code>None</code>, all 
          edges are checked.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list of booleans, one for every edge given</dd>
  </dl>
</td></tr></table>
</div>
<a name="is_minimal_separator"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_minimal_separator</span>(<span class="sig-arg">vertices</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_minimal_separator">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Decides whether the given vertex set is a minimal separator.</p>
  <p>A minimal separator is a set of vertices whose removal disconnects the
  graph, while the removal of any subset of the set keeps the graph 
  connected.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a single vertex ID or a list of vertex IDs</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd><code>True</code> is the given vertex set is a minimal separator,
          <code>False</code> otherwise.</dd>
  </dl>
</td></tr></table>
</div>
<a name="is_multiple"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_multiple</span>(<span class="sig-arg">edges</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_multiple">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether an edge is a multiple edge.</p>
  <p>Also works for a set of edges -- in this case, every edge is checked 
  one by one. Note that if there are multiple edges going between a pair of
  vertices, there is always one of them that is <i>not</i> reported as 
  multiple (only the others). This allows one to easily detect the edges 
  that have to be deleted in order to make the graph free of multiple 
  edges.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>edges</code></strong> - edge indices which we want to check. If <code>None</code>, all 
          edges are checked.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list of booleans, one for every edge given</dd>
  </dl>
</td></tr></table>
</div>
<a name="is_mutual"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_mutual</span>(<span class="sig-arg">edges</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_mutual">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether an edge has an opposite pair.</p>
  <p>Also works for a set of edges -- in this case, every edge is checked 
  one by one. The result will be a list of booleans (or a single boolean if
  only an edge index is supplied), every boolean corresponding to an edge 
  in the edge set supplied. <code>True</code> is returned for a given edge 
  <i class="math">a</i> --&gt; <i class="math">b</i> if there exists 
  another edge <i class="math">b</i> --&gt; <i class="math">a</i> in the 
  original graph (not the given edge set!). All edges in an undirected 
  graph are mutual. In case there are multiple edges between <i 
  class="math">a</i> and <i class="math">b</i>, it is enough to have at 
  least one edge in either direction to report all edges between them as 
  mutual, so the multiplicity of edges do not matter.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>edges</code></strong> - edge indices which we want to check. If <code>None</code>, all 
          edges are checked.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list of booleans, one for every edge given</dd>
  </dl>
</td></tr></table>
</div>
<a name="is_separator"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_separator</span>(<span class="sig-arg">vertices</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_separator">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Decides whether the removal of the given vertices disconnects the 
  graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a single vertex ID or a list of vertex IDs</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd><code>True</code> is the given vertex set is a separator, 
          <code>False</code> if not.</dd>
  </dl>
</td></tr></table>
</div>
<a name="is_simple"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">is_simple</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.is_simple">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether the graph is simple (no loop or multiple edges).</p>
  <dl class="fields">
    <dt>Returns: boolean</dt>
        <dd><code>True</code> if it is simple, <code>False</code> otherwise.</dd>
  </dl>
</td></tr></table>
</div>
<a name="isoclass"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">isoclass</span>(<span class="sig-arg">vertices</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.isoclass">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the isomorphy class of the graph or its subgraph.</p>
  <p>Isomorphy class calculations are implemented only for graphs with 3 or
  4 vertices.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a list of vertices if we want to calculate the isomorphy class 
          for only a subset of vertices. <code>None</code> means to use the
          full graph.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the isomorphy class of the (sub)graph</dd>
  </dl>
</td></tr></table>
</div>
<a name="isomorphic"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">isomorphic</span>(<span class="sig-arg">other</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.isomorphic">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether the graph is isomorphic to another graph.</p>
  <p>The algorithm being used is selected using a simple heuristic:</p>
  <ul>
    <li>
      If one graph is directed and the other undirected, an exception is 
      thrown.
    </li>
    <li>
      If the two graphs does not have the same number of vertices and 
      edges, it returns with <code>False</code>
    </li>
    <li>
      If the graphs have three or four vertices, then an O(1) algorithm is 
      used with precomputed data.
    </li>
    <li>
      Otherwise if the graphs are directed, then the VF2 isomorphism 
      algorithm is used (see <a 
      href="igraph.GraphBase-class.html#isomorphic_vf2" 
      class="link">Graph.isomorphic_vf2</a>).
    </li>
    <li>
      Otherwise the BLISS isomorphism algorithm is used, see <a 
      href="igraph.GraphBase-class.html#isomorphic_bliss" 
      class="link">Graph.isomorphic_bliss</a>.
    </li>
  </ul>
  <dl class="fields">
    <dt>Returns:</dt>
        <dd><code>True</code> if the graphs are isomorphic, 
          <code>False</code> otherwise.</dd>
  </dl>
</td></tr></table>
</div>
<a name="isomorphic_bliss"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">isomorphic_bliss</span>(<span class="sig-arg">other</span>,
        <span class="sig-arg">return_mapping_12</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">return_mapping_21</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">sh1</span>=<span class="sig-default">&quot;fm&quot;</span>,
        <span class="sig-arg">sh2</span>=<span class="sig-default">&quot;fm&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.isomorphic_bliss">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether the graph is isomorphic to another graph, using the 
  BLISS isomorphism algorithm.</p>
  <p>See <a href="http://www.tcs.hut.fi/Software/bliss/index.html" 
  target="_top">http://www.tcs.hut.fi/Software/bliss/index.html</a> for 
  more information about the BLISS algorithm.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the other graph with which we want to compare the graph.</li>
        <li><strong class="pname"><code>return_mapping_12</code></strong> - if <code>True</code>, calculates the mapping which maps the 
          vertices of the first graph to the second.</li>
        <li><strong class="pname"><code>return_mapping_21</code></strong> - if <code>True</code>, calculates the mapping which maps the 
          vertices of the second graph to the first.</li>
        <li><strong class="pname"><code>sh1</code></strong> - splitting heuristics for the first graph as a case-insensitive 
          string, with the following possible values:
          <ul>
            <li>
              <code>&quot;f&quot;</code>: first non-singleton cell
            </li>
            <li>
              <code>&quot;fl&quot;</code>: first largest non-singleton cell
            </li>
            <li>
              <code>&quot;fs&quot;</code>: first smallest non-singleton 
              cell
            </li>
            <li>
              <code>&quot;fm&quot;</code>: first maximally non-trivially 
              connected non-singleton cell
            </li>
            <li>
              <code>&quot;flm&quot;</code>: largest maximally non-trivially
              connected non-singleton cell
            </li>
            <li>
              <code>&quot;fsm&quot;</code>: smallest maximally 
              non-trivially connected non-singleton cell
            </li>
          </ul></li>
        <li><strong class="pname"><code>sh2</code></strong> - splitting heuristics to be used for the second graph. Accepted 
          values are as above.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>if no mapping is calculated, the result is <code>True</code> if 
          the graphs are isomorphic, <code>False</code> otherwise. If any 
          or both mappings are calculated, the result is a 3-tuple, the 
          first element being the above mentioned boolean, the second 
          element being the 1 -&gt; 2 mapping and the third element being 
          the 2 -&gt; 1 mapping. If the corresponding mapping was not 
          calculated, <code>None</code> is returned in the appropriate 
          element of the 3-tuple.</dd>
  </dl>
</td></tr></table>
</div>
<a name="isomorphic_vf2"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">isomorphic_vf2</span>(<span class="sig-arg">other</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">return_mapping_12</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">return_mapping_21</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">node_compat_fn</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_compat_fn</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">callback</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.isomorphic_vf2">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether the graph is isomorphic to another graph, using the VF2
  isomorphism algorithm.</p>
  <p>Vertex and edge colors may be used to restrict the isomorphisms, as 
  only vertices and edges with the same color will be allowed to match each
  other.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the other graph with which we want to compare the graph. If 
          <code>None</code>, the automorphjisms of the graph will be 
          tested.</li>
        <li><strong class="pname"><code>color1</code></strong> - optional vector storing the coloring of the vertices of the first
          graph. If <code>None</code>, all vertices have the same color.</li>
        <li><strong class="pname"><code>color2</code></strong> - optional vector storing the coloring of the vertices of the 
          second graph. If <code>None</code>, all vertices have the same 
          color.</li>
        <li><strong class="pname"><code>edge_color1</code></strong> - optional vector storing the coloring of the edges of the first 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>edge_color2</code></strong> - optional vector storing the coloring of the edges of the second 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>return_mapping_12</code></strong> - if <code>True</code>, calculates the mapping which maps the 
          vertices of the first graph to the second.</li>
        <li><strong class="pname"><code>return_mapping_21</code></strong> - if <code>True</code>, calculates the mapping which maps the 
          vertices of the second graph to the first.</li>
        <li><strong class="pname"><code>callback</code></strong> - if not <code>None</code>, the isomorphism search will not stop at
          the first match; it will call this callback function instead for 
          every isomorphism found. The callback function must accept four 
          arguments: the first graph, the second graph, a mapping from the 
          nodes of the first graph to the second, and a mapping from the 
          nodes of the second graph to the first. The function must return 
          <code>True</code> if the search should continue or 
          <code>False</code> otherwise.</li>
        <li><strong class="pname"><code>node_compat_fn</code></strong> - a function that receives the two graphs and two node indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the nodes given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on node-specific criteria that are too 
          complicated to be represented by node color vectors (i.e. the 
          <code>color1</code> and <code>color2</code> parameters). 
          <code>None</code> means that every node is compatible with every 
          other node.</li>
        <li><strong class="pname"><code>edge_compat_fn</code></strong> - a function that receives the two graphs and two edge indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the edges given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on edge-specific criteria that are too 
          complicated to be represented by edge color vectors (i.e. the 
          <code>edge_color1</code> and <code>edge_color2</code> 
          parameters). <code>None</code> means that every edge is 
          compatible with every other node.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>if no mapping is calculated, the result is <code>True</code> if 
          the graphs are isomorphic, <code>False</code> otherwise. If any 
          or both mappings are calculated, the result is a 3-tuple, the 
          first element being the above mentioned boolean, the second 
          element being the 1 -&gt; 2 mapping and the third element being 
          the 2 -&gt; 1 mapping. If the corresponding mapping was not 
          calculated, <code>None</code> is returned in the appropriate 
          element of the 3-tuple.</dd>
  </dl>
</td></tr></table>
</div>
<a name="knn"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">knn</span>(<span class="sig-arg">vids</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.knn">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the average degree of the neighbors for each vertex, and 
  the same quantity as the function of vertex degree.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vids</code></strong> - the vertices for which the calculation is performed. 
          <code>None</code> means all vertices.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name. If this is given, the vertex strength will 
          be used instead of the vertex degree in the calculations, but the
          &quot;ordinary&quot; vertex degree will be used for the second 
          (degree- dependent) list in the result.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>two lists in a tuple. The first list contains the average degree 
          of neighbors for each vertex, the second contains the average 
          degree of neighbors as a function of vertex degree. The zeroth 
          element of this list corresponds to vertices of degree 1.</dd>
  </dl>
</td></tr></table>
</div>
<a name="laplacian"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">laplacian</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">normalized</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.laplacian">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the Laplacian matrix of a graph.</p>
  <p>The Laplacian matrix is similar to the adjacency matrix, but the edges
  are denoted with -1 and the diagonal contains the node degrees.</p>
  <p>Normalized Laplacian matrices have 1 or 0 in their diagonals (0 for 
  vertices with no edges), edges are denoted by 1 / sqrt(d_i * d_j) where 
  d_i is the degree of node i.</p>
  <p>Multiple edges and self-loops are silently ignored. Although it is 
  possible to calculate the Laplacian matrix of a directed graph, it does 
  not make much sense.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name. When edge weights are used, the degree of a 
          node is considered to be the weight of its incident edges.</li>
        <li><strong class="pname"><code>normalized</code></strong> - whether to return the normalized Laplacian matrix.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the Laplacian matrix.</dd>
  </dl>
</td></tr></table>
</div>
<a name="largest_cliques"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">largest_cliques</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.largest_cliques">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the largest cliques of the graph as a list of tuples.</p>
  <p>Quite intuitively a clique is considered largest if there is no clique
  with more vertices in the whole graph. All largest cliques are maximal 
  (i.e. nonextendable) but not all maximal cliques are largest.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.Graph-class.html#omega" 
        class="link">clique_number()</a> for the size of the largest 
        cliques or <a href="igraph.GraphBase-class.html#maximal_cliques" 
        class="link">maximal_cliques()</a> for the maximal cliques
      </p>
</div></td></tr></table>
</div>
<a name="largest_independent_vertex_sets"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">largest_independent_vertex_sets</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.largest_independent_vertex_sets">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the largest independent vertex sets of the graph as a list of 
  tuples.</p>
  <p>Quite intuitively an independent vertex set is considered largest if 
  there is no other set with more vertices in the whole graph. All largest 
  sets are maximal (i.e. nonextendable) but not all maximal sets are 
  largest.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.Graph-class.html#alpha" 
        class="link">independence_number()</a> for the size of the largest 
        independent vertex sets or <a 
        href="igraph.GraphBase-class.html#maximal_independent_vertex_sets" 
        class="link">maximal_independent_vertex_sets()</a> for the maximal 
        (nonextendable) independent vertex sets
      </p>
</div></td></tr></table>
</div>
<a name="layout_bipartite"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_bipartite</span>(<span class="sig-arg">types</span>=<span class="sig-default">&quot;type&quot;</span>,
        <span class="sig-arg">hgap</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">vgap</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">maxiter</span>=<span class="sig-default">100</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_bipartite">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Place the vertices of a bipartite graph in two layers.</p>
  <p>The layout is created by placing the vertices in two rows, according 
  to their types. The positions of the vertices within the rows are then 
  optimized to minimize the number of edge crossings using the heuristic 
  used by the Sugiyama layout algorithm.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>types</code></strong> - an igraph vector containing the vertex types, or an attribute 
          name. Anything that evalulates to <code>False</code> corresponds 
          to vertices of the first kind, everything else to the second 
          kind.</li>
        <li><strong class="pname"><code>hgap</code></strong> - minimum horizontal gap between vertices in the same layer.</li>
        <li><strong class="pname"><code>vgap</code></strong> - vertical gap between the two layers.</li>
        <li><strong class="pname"><code>maxiter</code></strong> - maximum number of iterations to take in the crossing reduction 
          step. Increase this if you feel that you are getting too many 
          edge crossings.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_circle"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_circle</span>(<span class="sig-arg">dim</span>=<span class="sig-default">2</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_circle">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices of the graph uniformly on a circle or a 
  sphere.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>dim</code></strong> - the desired number of dimensions for the layout. dim=2 means a 2D
          layout, dim=3 means a 3D layout.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_drl"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_drl</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">fixed</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">seed</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">options</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">dim</span>=<span class="sig-default">2</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_drl">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices on a 2D plane or in the 3D space ccording to the 
  DrL layout algorithm.</p>
  <p>This is an algorithm suitable for quite large graphs, but it can be 
  surprisingly slow for small ones (where the simpler force-based layouts 
  like <code>layout_kamada_kawai()</code> or 
  <code>layout_fruchterman_reingold()</code> are more useful.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
        <li><strong class="pname"><code>seed</code></strong> - if <code>None</code>, uses a random starting layout for the 
          algorithm. If a matrix (list of lists), uses the given matrix as 
          the starting position.</li>
        <li><strong class="pname"><code>fixed</code></strong> - if a seed is given, you can specify some vertices to be kept 
          fixed at their original position in the seed by passing an 
          appropriate list here. The list must have exactly as many items 
          as the number of vertices in the graph. Items of the list that 
          evaluate to <code>True</code> denote vertices that will not be 
          moved.</li>
        <li><strong class="pname"><code>options</code></strong> - if you give a string argument here, you can select from five 
          default preset parameterisations: <code>default</code>, 
          <code>coarsen</code> for a coarser layout, <code>coarsest</code> 
          for an even coarser layout, <code>refine</code> for refining an 
          existing layout and <code>final</code> for finalizing a layout. 
          If you supply an object that is not a string, the DrL layout 
          parameters are retrieved from the respective keys of the object 
          (so it should be a dict or something else that supports the 
          mapping protocol). The following keys can be used:
          <ul>
            <li>
              <code>edge_cut</code>: edge cutting is done in the late 
              stages of the algorithm in order to achieve less dense 
              layouts. Edges are cut if there is a lot of stress on them (a
              large value in the objective function sum). The edge cutting 
              parameter is a value between 0 and 1 with 0 representing no 
              edge cutting and 1 representing maximal edge cutting.
            </li>
            <li>
              <code>init_iterations</code>: number of iterations in the 
              initialization phase
            </li>
            <li>
              <code>init_temperature</code>: start temperature during 
              initialization
            </li>
            <li>
              <code>init_attraction</code>: attraction during 
              initialization
            </li>
            <li>
              <code>init_damping_mult</code>: damping multiplier during 
              initialization
            </li>
            <li>
              <code>liquid_iterations</code>, 
              <code>liquid_temperature</code>, 
              <code>liquid_attraction</code>, 
              <code>liquid_damping_mult</code>: same parameters for the 
              liquid phase
            </li>
            <li>
              <code>expansion_iterations</code>, 
              <code>expansion_temperature</code>, 
              <code>expansion_attraction</code>, 
              <code>expansion_damping_mult</code>: parameters for the 
              expansion phase
            </li>
            <li>
              <code>cooldown_...</code>: parameters for the cooldown phase
            </li>
            <li>
              <code>crunch_...</code>: parameters for the crunch phase
            </li>
            <li>
              <code>simmer_...</code>: parameters for the simmer phase
            </li>
          </ul>
          <p>Instead of a mapping, you can also use an arbitrary Python 
          object here: if the object does not support the mapping protocol,
          an attribute of the object with the same name is looked up 
          instead. If a parameter cannot be found either as a key or an 
          attribute, the default from the <code>default</code> preset will 
          be used.</p></li>
        <li><strong class="pname"><code>dim</code></strong> - the desired number of dimensions for the layout. dim=2 means a 2D
          layout, dim=3 means a 3D layout.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_fruchterman_reingold"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_fruchterman_reingold</span>(<span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxiter</span>=<span class="sig-default">500</span>,
        <span class="sig-arg">maxdelta</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">area</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">coolexp</span>=<span class="sig-default">1.5</span>,
        <span class="sig-arg">repulserad</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">seed</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">minx</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxx</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">miny</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxy</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">minz</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxz</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">dim</span>=<span class="sig-default">2</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_fruchterman_reingold">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices on a 2D plane or in the 3D space according to the 
  Fruchterman-Reingold algorithm.</p>
  <p>This is a force directed layout, see Fruchterman, T. M. J. and 
  Reingold, E. M.: Graph Drawing by Force-directed Placement. Software -- 
  Practice and Experience, 21/11, 1129--1164, 1991</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
        <li><strong class="pname"><code>maxiter</code></strong> - the number of iterations to perform. The default is 500.</li>
        <li><strong class="pname"><code>maxdelta</code></strong> - the maximum distance to move a vertex in an iteration. The 
          default is the number of vertices.</li>
        <li><strong class="pname"><code>area</code></strong> - the area of the square on which the vertices will be placed. The 
          default is the square of the number of vertices.</li>
        <li><strong class="pname"><code>coolexp</code></strong> - the cooling exponent of the simulated annealing. The default is 
          1.5.</li>
        <li><strong class="pname"><code>repulserad</code></strong> - determines the radius at which vertex-vertex repulsion cancels 
          out attraction of adjacent vertices. The default is the number of
          vertices^3.</li>
        <li><strong class="pname"><code>minx</code></strong> - if not <code>None</code>, it must be a vector with exactly as 
          many elements as there are vertices in the graph. Each element is
          a minimum constraint on the X value of the vertex in the layout.</li>
        <li><strong class="pname"><code>maxx</code></strong> - similar to <i>minx</i>, but with maximum constraints</li>
        <li><strong class="pname"><code>miny</code></strong> - similar to <i>minx</i>, but with the Y coordinates</li>
        <li><strong class="pname"><code>maxy</code></strong> - similar to <i>maxx</i>, but with the Y coordinates</li>
        <li><strong class="pname"><code>minz</code></strong> - similar to <i>minx</i>, but with the Z coordinates. Use only for 
          3D layouts (<code>dim</code>=3).</li>
        <li><strong class="pname"><code>maxz</code></strong> - similar to <i>maxx</i>, but with the Z coordinates. Use only for 
          3D layouts (<code>dim</code>=3).</li>
        <li><strong class="pname"><code>seed</code></strong> - if <code>None</code>, uses a random starting layout for the 
          algorithm. If a matrix (list of lists), uses the given matrix as 
          the starting position.</li>
        <li><strong class="pname"><code>dim</code></strong> - the desired number of dimensions for the layout. dim=2 means a 2D
          layout, dim=3 means a 3D layout.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_graphopt"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_graphopt</span>(<span class="sig-arg">niter</span>=<span class="sig-default">500</span>,
        <span class="sig-arg">node_charge</span>=<span class="sig-default">0.001</span>,
        <span class="sig-arg">node_mass</span>=<span class="sig-default">30</span>,
        <span class="sig-arg">spring_length</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">spring_constant</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">max_sa_movement</span>=<span class="sig-default">5</span>,
        <span class="sig-arg">seed</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_graphopt">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>This is a port of the graphopt layout algorithm by Michael Schmuhl. 
  graphopt version 0.4.1 was rewritten in C and the support for layers was 
  removed.</p>
  <p>graphopt uses physical analogies for defining attracting and repelling
  forces among the vertices and then the physical system is simulated until
  it reaches an equilibrium or the maximal number of iterations is 
  reached.</p>
  <p>See <a href="http://www.schmuhl.org/graphopt/" 
  target="_top">http://www.schmuhl.org/graphopt/</a> for the original 
  graphopt.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>niter</code></strong> - the number of iterations to perform. Should be a couple of 
          hundred in general.</li>
        <li><strong class="pname"><code>node_charge</code></strong> - the charge of the vertices, used to calculate electric repulsion.</li>
        <li><strong class="pname"><code>node_mass</code></strong> - the mass of the vertices, used for the spring forces</li>
        <li><strong class="pname"><code>spring_length</code></strong> - the length of the springs</li>
        <li><strong class="pname"><code>spring_constant</code></strong> - the spring constant</li>
        <li><strong class="pname"><code>max_sa_movement</code></strong> - the maximum amount of movement allowed in a single step along a 
          single axis.</li>
        <li><strong class="pname"><code>seed</code></strong> - a matrix containing a seed layout from which the algorithm will 
          be started. If <code>None</code>, a random layout will be used.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_grid"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_grid</span>(<span class="sig-arg">width</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">height</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">dim</span>=<span class="sig-default">2</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_grid">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices of a graph in a 2D or 3D grid.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>width</code></strong> - the number of vertices in a single row of the layout. Zero or 
          negative numbers mean that the width should be determined 
          automatically.</li>
        <li><strong class="pname"><code>height</code></strong> - the number of vertices in a single column of the layout. Zero or 
          negative numbers mean that the height should be determined 
          automatically. It must not be given if the number of dimensions 
          is 2.</li>
        <li><strong class="pname"><code>dim</code></strong> - the desired number of dimensions for the layout. dim=2 means a 2D
          layout, dim=3 means a 3D layout.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_grid_fruchterman_reingold"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_grid_fruchterman_reingold</span>(<span class="sig-arg">maxiter</span>=<span class="sig-default">500</span>,
        <span class="sig-arg">maxdelta</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">area</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">coolexp</span>=<span class="sig-default">0.99</span>,
        <span class="sig-arg">repulserad</span>=<span class="sig-default">maxiter*maxdelta</span>,
        <span class="sig-arg">cellsize</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">seed</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_grid_fruchterman_reingold">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices on a 2D plane according to the 
  Fruchterman-Reingold grid algorithm.</p>
  <p>This is a modified version of a force directed layout, see 
  Fruchterman, T. M. J. and Reingold, E. M.: Graph Drawing by 
  Force-directed Placement. Software -- Practice and Experience, 21/11, 
  1129--1164, 1991. The algorithm partitions the 2D space to a grid and 
  vertex repulsion is then calculated only for vertices nearby.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>maxiter</code></strong> - the number of iterations to perform.</li>
        <li><strong class="pname"><code>maxdelta</code></strong> - the maximum distance to move a vertex in an iteration. 
          <code>None</code> means the number of vertices.</li>
        <li><strong class="pname"><code>area</code></strong> - the area of the square on which the vertices will be placed. 
          <code>None</code> means the square of <i 
          class="math">maxdelta</i>.</li>
        <li><strong class="pname"><code>coolexp</code></strong> - the cooling exponent of the simulated annealing.</li>
        <li><strong class="pname"><code>repulserad</code></strong> - determines the radius at which vertex-vertex repulsion cancels 
          out attraction of adjacent vertices. <code>None</code> means <i 
          class="math">maxiter*maxdelta</i>.</li>
        <li><strong class="pname"><code>cellsize</code></strong> - the size of the grid cells. When calculating the repulsion 
          forces, only vertices in the same or neighboring grid cells are 
          taken into account. Defaults to the fourth root of <i 
          class="math">area</i>.</li>
        <li><strong class="pname"><code>seed</code></strong> - if <code>None</code>, uses a random starting layout for the 
          algorithm. If a matrix (list of lists), uses the given matrix as 
          the starting position.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_kamada_kawai"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_kamada_kawai</span>(<span class="sig-arg">maxiter</span>=<span class="sig-default">1000</span>,
        <span class="sig-arg">sigma</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">initemp</span>=<span class="sig-default">10</span>,
        <span class="sig-arg">coolexp</span>=<span class="sig-default">0.99</span>,
        <span class="sig-arg">kkconst</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">seed</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">minx</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxx</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">miny</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxy</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">minz</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">maxz</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">dim</span>=<span class="sig-default">2</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_kamada_kawai">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices on a plane according to the Kamada-Kawai 
  algorithm.</p>
  <p>This is a force directed layout, see Kamada, T. and Kawai, S.: An 
  Algorithm for Drawing General Undirected Graphs. Information Processing 
  Letters, 31/1, 7--15, 1989.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>maxiter</code></strong> - the number of iterations to perform.</li>
        <li><strong class="pname"><code>sigma</code></strong> - the standard base deviation of the position change proposals. 
          <code>None</code> means the number of vertices / 4</li>
        <li><strong class="pname"><code>initemp</code></strong> - initial temperature of the simulated annealing.</li>
        <li><strong class="pname"><code>coolexp</code></strong> - cooling exponent of the simulated annealing.</li>
        <li><strong class="pname"><code>kkconst</code></strong> - the Kamada-Kawai vertex attraction constant. <code>None</code> 
          means the square of the number of vertices.</li>
        <li><strong class="pname"><code>minx</code></strong> - if not <code>None</code>, it must be a vector with exactly as 
          many elements as there are vertices in the graph. Each element is
          a minimum constraint on the X value of the vertex in the layout.</li>
        <li><strong class="pname"><code>maxx</code></strong> - similar to <i>minx</i>, but with maximum constraints</li>
        <li><strong class="pname"><code>miny</code></strong> - similar to <i>minx</i>, but with the Y coordinates</li>
        <li><strong class="pname"><code>maxy</code></strong> - similar to <i>maxx</i>, but with the Y coordinates</li>
        <li><strong class="pname"><code>minz</code></strong> - similar to <i>minx</i>, but with the Z coordinates. Use only for 
          3D layouts (<code>dim</code>=3).</li>
        <li><strong class="pname"><code>maxz</code></strong> - similar to <i>maxx</i>, but with the Z coordinates. Use only for 
          3D layouts (<code>dim</code>=3).</li>
        <li><strong class="pname"><code>seed</code></strong> - if <code>None</code>, uses a random starting layout for the 
          algorithm. If a matrix (list of lists), uses the given matrix as 
          the starting position.</li>
        <li><strong class="pname"><code>dim</code></strong> - the desired number of dimensions for the layout. dim=2 means a 2D
          layout, dim=3 means a 3D layout.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_lgl"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_lgl</span>(<span class="sig-arg">maxiter</span>=<span class="sig-default">150</span>,
        <span class="sig-arg">maxdelta</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">area</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">coolexp</span>=<span class="sig-default">1.5</span>,
        <span class="sig-arg">repulserad</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">cellsize</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">root</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_lgl">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices on a 2D plane according to the Large Graph 
  Layout.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>maxiter</code></strong> - the number of iterations to perform.</li>
        <li><strong class="pname"><code>maxdelta</code></strong> - the maximum distance to move a vertex in an iteration. If 
          negative, defaults to the number of vertices.</li>
        <li><strong class="pname"><code>area</code></strong> - the area of the square on which the vertices will be placed. If 
          negative, defaults to the number of vertices squared.</li>
        <li><strong class="pname"><code>coolexp</code></strong> - the cooling exponent of the simulated annealing.</li>
        <li><strong class="pname"><code>repulserad</code></strong> - determines the radius at which vertex-vertex repulsion cancels 
          out attraction of adjacent vertices. If negative, defaults to <i 
          class="math">area</i> times the number of vertices.</li>
        <li><strong class="pname"><code>cellsize</code></strong> - the size of the grid cells. When calculating the repulsion 
          forces, only vertices in the same or neighboring grid cells are 
          taken into account. Defaults to the fourth root of <i 
          class="math">area</i>.</li>
        <li><strong class="pname"><code>root</code></strong> - the root vertex, this is placed first, its neighbors in the first
          iteration, second neighbors in the second, etc. <code>None</code>
          means that a random vertex will be chosen.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_mds"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_mds</span>(<span class="sig-arg">dist</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">dim</span>=<span class="sig-default">2</span>,
        <span class="sig-arg">arpack_options</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_mds">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices in an Euclidean space with the given number of 
  dimensions using multidimensional scaling.</p>
  <p>This layout requires a distance matrix, where the intersection of row 
  <i class="math">i</i> and column <i class="math">j</i> specifies the 
  desired distance between vertex <i class="math">i</i> and vertex <i 
  class="math">j</i>. The algorithm will try to place the vertices in a way
  that approximates the distance relations prescribed in the distance 
  matrix. igraph uses the classical multidimensional scaling by Torgerson 
  (see reference below).</p>
  <p>For unconnected graphs, the method will decompose the graph into 
  weakly connected components and then lay out the components individually 
  using the appropriate parts of the distance matrix.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>dist</code></strong> - the distance matrix. It must be symmetric and the symmetry is not
          checked -- results are unspecified when a non-symmetric distance 
          matrix is used. If this parameter is <code>None</code>, the 
          shortest path lengths will be used as distances. Directed graphs 
          are treated as undirected when calculating the shortest path 
          lengths to ensure symmetry.</li>
        <li><strong class="pname"><code>dim</code></strong> - the number of dimensions. For 2D layouts, supply 2 here; for 3D 
          layouts, supply 3.</li>
        <li><strong class="pname"><code>arpack_options</code></strong> - an <a href="igraph.ARPACKOptions-class.html" 
          class="link">ARPACKOptions</a> object used to fine-tune the 
          ARPACK eigenvector calculation. If omitted, the module-level 
          variable called <code>arpack_options</code> is used.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Cox &amp; Cox: Multidimensional Scaling (1994), Chapman and Hall, 
        London.
      </p>
</div></td></tr></table>
</div>
<a name="layout_random"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_random</span>(<span class="sig-arg">dim</span>=<span class="sig-default">2</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_random">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices of the graph randomly.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>dim</code></strong> - the desired number of dimensions for the layout. dim=2 means a 2D
          layout, dim=3 means a 3D layout.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the coordinate pairs in a list.</dd>
  </dl>
</td></tr></table>
</div>
<a name="layout_reingold_tilford"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_reingold_tilford</span>(<span class="sig-arg">mode</span>=<span class="sig-default">&quot;out&quot;</span>,
        <span class="sig-arg">root</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">rootlevel</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_reingold_tilford">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Places the vertices on a 2D plane according to the Reingold-Tilford 
  layout algorithm.</p>
  <p>This is a tree layout. If the given graph is not a tree, a 
  breadth-first search is executed first to obtain a possible spanning 
  tree.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - specifies which edges to consider when builing the tree. If it is
          <code>OUT</code> then only the outgoing, if it is <code>IN</code>
          then only the incoming edges of a parent are considered. If it is
          <code>ALL</code> then all edges are used (this was the behaviour 
          in igraph 0.5 and before). This parameter also influences how the
          root vertices are calculated if they are not given. See the 
          <i>root</i> parameter.</li>
        <li><strong class="pname"><code>root</code></strong> - the index of the root vertex or root vertices. if this is a 
          non-empty vector then the supplied vertex IDs are used as the 
          roots of the trees (or a single tree if the graph is connected. 
          If this is <code>None</code> or an empty list, the root vertices 
          are automatically calculated based on topological sorting, 
          performed with the opposite of the <i>mode</i> argument.</li>
        <li><strong class="pname"><code>rootlevel</code></strong> - this argument is useful when drawing forests which are not trees.
          It specifies the level of the root vertices for every tree in the
          forest.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        layout_reingold_tilford_circular
      </p>
      <p><strong>Reference:</strong>
        EM Reingold, JS Tilford: <i>Tidier Drawings of Trees.</i> IEEE 
        Transactions on Software Engineering 7:22, 223-228, 1981.
      </p>
</div></td></tr></table>
</div>
<a name="layout_reingold_tilford_circular"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_reingold_tilford_circular</span>(<span class="sig-arg">mode</span>=<span class="sig-default">&quot;out&quot;</span>,
        <span class="sig-arg">root</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">rootlevel</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_reingold_tilford_circular">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Circular Reingold-Tilford layout for trees.</p>
  <p>This layout is similar to the Reingold-Tilford layout, but the 
  vertices are placed in a circular way, with the root vertex in the 
  center.</p>
  <p>See <a href="igraph.GraphBase-class.html#layout_reingold_tilford" 
  class="link">layout_reingold_tilford</a> for the explanation of the 
  parameters.</p>
  <dl class="fields">
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        layout_reingold_tilford
      </p>
      <p><strong>Reference:</strong>
        EM Reingold, JS Tilford: <i>Tidier Drawings of Trees.</i> IEEE 
        Transactions on Software Engineering 7:22, 223-228, 1981.
      </p>
</div></td></tr></table>
</div>
<a name="layout_star"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">layout_star</span>(<span class="sig-arg">center</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">order</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.layout_star">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates a star-like layout for the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>center</code></strong> - the ID of the vertex to put in the center</li>
        <li><strong class="pname"><code>order</code></strong> - a numeric vector giving the order of the vertices (including the 
          center vertex!). If it is <code>None</code>, the vertices will be
          placed in increasing vertex ID order.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the calculated layout.</dd>
  </dl>
</td></tr></table>
</div>
<a name="linegraph"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">linegraph</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.linegraph">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the line graph of the graph.</p>
  <p>The line graph <i class="math">L(G)</i> of an undirected graph is 
  defined as follows: <i class="math">L(G)</i> has one vertex for each edge
  in G and two vertices in <i class="math">L(G)</i> are connected iff their
  corresponding edges in the original graph share an end point.</p>
  <p>The line graph of a directed graph is slightly different: two vertices
  are connected by a directed edge iff the target of the first vertex's 
  corresponding edge is the same as the source of the second vertex's 
  corresponding edge.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="maxdegree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">maxdegree</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.maxdegree">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the maximum degree of a vertex set in the graph.</p>
  <p>This method accepts a single vertex ID or a list of vertex IDs as a 
  parameter, and returns the degree of the given vertices (in the form of a
  single integer or a list, depending on the input parameter).</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a single vertex ID or a list of vertex IDs, or <code>None</code> 
          meaning all the vertices in the graph.</li>
        <li><strong class="pname"><code>mode</code></strong> - the type of degree to be returned (<a 
          href="igraph-module.html#OUT" class="link">OUT</a> for 
          out-degrees, <a href="igraph-module.html#IN" class="link">IN</a> 
          IN for in-degrees or <a href="igraph-module.html#ALL" 
          class="link">ALL</a> for the sum of them).</li>
        <li><strong class="pname"><code>loops</code></strong> - whether self-loops should be counted.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="maxflow"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">maxflow</span>(<span class="sig-arg">source</span>,
        <span class="sig-arg">target</span>,
        <span class="sig-arg">capacity</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.maxflow">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the maximum flow between the source and target vertices.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex ID</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex ID</li>
        <li><strong class="pname"><code>capacity</code></strong> - the capacity of the edges. It must be a list or a valid attribute
          name or <code>None</code>. In the latter case, every edge will 
          have the same capacity.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a tuple containing the following: the value of the maximum flow 
          between the given vertices, the flow value on all the edges, the 
          edge IDs that are part of the corresponding minimum cut, and the 
          vertex IDs on one side of the cut. For directed graphs, the flow 
          value vector gives the flow value on each edge. For undirected 
          graphs, the flow value is positive if the flow goes from the 
          smaller vertex ID to the bigger one and negative if the flow goes
          from the bigger vertex ID to the smaller.</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a <a href="igraph.cut.Flow-class.html" 
        class="link">Flow</a> object. It is advised to use that.
      </p>
</div></td></tr></table>
</div>
<a name="maxflow_value"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">maxflow_value</span>(<span class="sig-arg">source</span>,
        <span class="sig-arg">target</span>,
        <span class="sig-arg">capacity</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.maxflow_value">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the value of the maximum flow between the source and target 
  vertices.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex ID</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex ID</li>
        <li><strong class="pname"><code>capacity</code></strong> - the capacity of the edges. It must be a list or a valid attribute
          name or <code>None</code>. In the latter case, every edge will 
          have the same capacity.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the value of the maximum flow between the given vertices</dd>
  </dl>
</td></tr></table>
</div>
<a name="maximal_cliques"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">maximal_cliques</span>(<span class="sig-arg">min</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">max</span>=<span class="sig-default">0</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.maximal_cliques">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the maximal cliques of the graph as a list of tuples.</p>
  <p>A maximal clique is a clique which can't be extended by adding any 
  other vertex to it. A maximal clique is not necessarily one of the 
  largest cliques in the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>min</code></strong> - the minimum size of maximal cliques to be returned. If zero or 
          negative, no lower bound will be used.</li>
        <li><strong class="pname"><code>max</code></strong> - the maximum size of maximal cliques to be returned. If zero or 
          negative, no upper bound will be used. If nonzero, the size of 
          every maximal clique found will be compared to this value and a 
          clique will be returned only if its size is smaller than this 
          limit.</li>
    </ul></dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.GraphBase-class.html#largest_cliques" 
        class="link">largest_cliques()</a> for the largest cliques.
      </p>
</div></td></tr></table>
</div>
<a name="maximal_independent_vertex_sets"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">maximal_independent_vertex_sets</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.maximal_independent_vertex_sets">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the maximal independent vertex sets of the graph as a list of 
  tuples.</p>
  <p>A maximal independent vertex set is an independent vertex set which 
  can't be extended by adding any other vertex to it. A maximal independent
  vertex set is not necessarily one of the largest independent vertex sets 
  in the graph.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a 
        href="igraph.GraphBase-class.html#largest_independent_vertex_sets" 
        class="link">largest_independent_vertex_sets()</a> for the largest 
        independent vertex sets
      </p>
      <p><strong>Reference:</strong>
        S. Tsukiyama, M. Ide, H. Ariyoshi and I. Shirawaka: <i>A new 
        algorithm for generating all the maximal independent sets</i>. SIAM
        J Computing, 6:505--517, 1977.
      </p>
</div></td></tr></table>
</div>
<a name="mincut"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">mincut</span>(<span class="sig-arg">source</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">target</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">capacity</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.mincut">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the minimum cut between the source and target vertices or 
  within the whole graph.</p>
  <p>The minimum cut is the minimum set of edges that needs to be removed 
  to separate the source and the target (if they are given) or to 
  disconnect the graph (if the source and target are not given). The 
  minimum is calculated using the weights (capacities) of the edges, so the
  cut with the minimum total capacity is calculated. For undirected graphs 
  and no source and target, the method uses the Stoer-Wagner algorithm. For
  a given source and target, the method uses the push-relabel algorithm; 
  see the references below.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex ID. If <code>None</code>, target must also be 
          {None} and the calculation will be done for the entire graph 
          (i.e. all possible vertex pairs).</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex ID. If <code>None</code>, source must also be 
          {None} and the calculation will be done for the entire graph 
          (i.e. all possible vertex pairs).</li>
        <li><strong class="pname"><code>capacity</code></strong> - the capacity of the edges. It must be a list or a valid attribute
          name or <code>None</code>. In the latter case, every edge will 
          have the same capacity.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the value of the minimum cut, the IDs of vertices in the first 
          and second partition, and the IDs of edges in the cut, packed in 
          a 4-tuple</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a <a href="igraph.cut.Cut-class.html" 
        class="link">Cut</a> object. It is advised to use that.
      </p>
      <strong>Reference:</strong>
      <ul class="nomargin-top">
        <li>
        M. Stoer, F. Wagner: A simple min-cut algorithm. Journal of the ACM
        44(4):585-591, 1997.
        </li>
        <li>
        A. V. Goldberg, R. E. Tarjan: A new approach to the maximum-flow 
        problem. Journal of the ACM 35(4):921-940, 1988.
        </li>
      </ul>
</div></td></tr></table>
</div>
<a name="mincut_value"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">mincut_value</span>(<span class="sig-arg">source</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">target</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">capacity</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.mincut_value">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the minimum cut between the source and target vertices or 
  within the whole graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex ID. If negative, the calculation is done for 
          every vertex except the target and the minimum is returned.</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex ID. If negative, the calculation is done for 
          every vertex except the source and the minimum is returned.</li>
        <li><strong class="pname"><code>capacity</code></strong> - the capacity of the edges. It must be a list or a valid attribute
          name or <code>None</code>. In the latter case, every edge will 
          have the same capacity.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the value of the minimum cut between the given vertices</dd>
  </dl>
</td></tr></table>
</div>
<a name="minimum_size_separators"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">minimum_size_separators</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.minimum_size_separators">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a list containing all separator vertex sets of minimum 
  size.</p>
  <p>A vertex set is a separator if its removal disconnects the graph. This
  method lists all the separators for which no smaller separator set exists
  in the given graph.</p>
  <dl class="fields">
    <dt>Returns:</dt>
        <dd>a list where each item lists the vertex indices of a given 
          separator of minimum size.</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        Arkady Kanevsky: Finding all minimum-size separating vertex sets in
        a graph. Networks 23:533--541, 1993.
      </p>
</div></td></tr></table>
</div>
<a name="modularity"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">modularity</span>(<span class="sig-arg">membership</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.modularity">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the modularity of the graph with respect to some vertex 
  types.</p>
  <p>The modularity of a graph w.r.t. some division measures how good the 
  division is, or how separated are the different vertex types from each 
  other. It is defined as <i class="math">Q=1/(2m) * 
  sum(Aij-ki*kj/(2m)delta(ci,cj),i,j)</i>. <i class="math">m</i> is the 
  number of edges, <i class="math">Aij</i> is the element of the <i 
  class="math">A</i> adjacency matrix in row <i class="math">i</i> and 
  column <i class="math">j</i>, <i class="math">ki</i> is the degree of 
  node <i class="math">i</i>, <i class="math">kj</i> is the degree of node 
  <i class="math">j</i>, and <i class="math">Ci</i> and <code>cj</code> are
  the types of the two vertices (<i class="math">i</i> and <i 
  class="math">j</i>). <i class="math">delta(x,y)</i> is one iff <i 
  class="math">x=y</i>, 0 otherwise.</p>
  <p>If edge weights are given, the definition of modularity is modified as
  follows: <i class="math">Aij</i> becomes the weight of the corresponding 
  edge, <i class="math">ki</i> is the total weight of edges incident on 
  vertex <i class="math">i</i>, <i class="math">kj</i> is the total weight 
  of edges incident on vertex <i class="math">j</i> and <i 
  class="math">m</i> is the total edge weight in the graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>membership</code></strong> - the membership vector, e.g. the vertex type index for each 
          vertex.</li>
        <li><strong class="pname"><code>weights</code></strong> - optional edge weights or <code>None</code> if all edges are 
          weighed equally.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the modularity score. Score larger than 0.3 usually indicates 
          strong community structure.</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        method overridden in <a href="igraph.Graph-class.html" 
        class="link">Graph</a> to allow <a 
        href="igraph.clustering.VertexClustering-class.html" 
        class="link">VertexClustering</a> objects as a parameter. This 
        method is not strictly necessary, since the <a 
        href="igraph.clustering.VertexClustering-class.html" 
        class="link">VertexClustering</a> class provides a variable called 
        <code>modularity</code>.
      </p>
      <p><strong>Reference:</strong>
        MEJ Newman and M Girvan: Finding and evaluating community structure
        in networks. Phys Rev E 69 026113, 2004.
      </p>
</div></td></tr></table>
</div>
<a name="motifs_randesu"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">motifs_randesu</span>(<span class="sig-arg">size</span>=<span class="sig-default">3</span>,
        <span class="sig-arg">cut_prob</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">callback</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.motifs_randesu">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Counts the number of motifs in the graph</p>
  <p>Motifs are small subgraphs of a given structure in a graph. It is 
  argued that the motif profile (ie. the number of different motifs in the 
  graph) is characteristic for different types of networks and network 
  function is related to the motifs in the graph.</p>
  <p>This function is able to find the different motifs of size three and 
  four (ie. the number of different subgraphs with three and four vertices)
  in the network.</p>
  <p>In a big network the total number of motifs can be very large, so it 
  takes a lot of time to find all of them. In such cases, a sampling method
  can be used. This function is capable of doing sampling via the 
  <i>cut_prob</i> argument. This argument gives the probability that a 
  branch of the motif search tree will not be explored.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>size</code></strong> - the size of the motifs (3 or 4).</li>
        <li><strong class="pname"><code>cut_prob</code></strong> - the cut probabilities for different levels of the search tree. 
          This must be a list of length <i>size</i> or <code>None</code> to
          find all motifs.</li>
        <li><strong class="pname"><code>callback</code></strong> - <code>None</code> or a callable that will be called for every 
          motif found in the graph. The callable must accept three 
          parameters: the graph itself, the list of vertices in the motif 
          and the isomorphy class of the motif (see <a 
          href="igraph.GraphBase-class.html#isoclass" 
          class="link">Graph.isoclass()</a>). The search will stop when the
          callback returns an object with a non-zero truth value or raises 
          an exception.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the list of motifs if <i>callback</i> is <code>None</code>, or 
          <code>None</code> otherwise</dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        S. Wernicke and F. Rasche: FANMOD: a tool for fast network motif 
        detection, Bioinformatics 22(9), 1152--1153, 2006.
      </p>
      <p><strong>See Also:</strong>
        Graph.motifs_randesu_no()
      </p>
</div></td></tr></table>
</div>
<a name="motifs_randesu_estimate"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">motifs_randesu_estimate</span>(<span class="sig-arg">size</span>=<span class="sig-default">3</span>,
        <span class="sig-arg">cut_prob</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">sample</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.motifs_randesu_estimate">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Counts the total number of motifs in the graph</p>
  <p>Motifs are small subgraphs of a given structure in a graph. This 
  function estimates the total number of motifs in a graph without 
  assigning isomorphism classes to them by extrapolating from a random 
  sample of vertices.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>size</code></strong> - the size of the motifs (3 or 4).</li>
        <li><strong class="pname"><code>cut_prob</code></strong> - the cut probabilities for different levels of the search tree. 
          This must be a list of length <i>size</i> or <code>None</code> to
          find all motifs.</li>
        <li><strong class="pname"><code>sample</code></strong> - the size of the sample or the vertex IDs of the vertices to be 
          used for sampling.</li>
    </ul></dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        S. Wernicke and F. Rasche: FANMOD: a tool for fast network motif 
        detection, Bioinformatics 22(9), 1152--1153, 2006.
      </p>
      <p><strong>See Also:</strong>
        Graph.motifs_randesu()
      </p>
</div></td></tr></table>
</div>
<a name="motifs_randesu_no"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">motifs_randesu_no</span>(<span class="sig-arg">size</span>=<span class="sig-default">3</span>,
        <span class="sig-arg">cut_prob</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.motifs_randesu_no">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Counts the total number of motifs in the graph</p>
  <p>Motifs are small subgraphs of a given structure in a graph. This 
  function counts the total number of motifs in a graph without assigning 
  isomorphism classes to them.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>size</code></strong> - the size of the motifs (3 or 4).</li>
        <li><strong class="pname"><code>cut_prob</code></strong> - the cut probabilities for different levels of the search tree. 
          This must be a list of length <i>size</i> or <code>None</code> to
          find all motifs.</li>
    </ul></dd>
  </dl>
<div class="fields">      <p><strong>Reference:</strong>
        S. Wernicke and F. Rasche: FANMOD: a tool for fast network motif 
        detection, Bioinformatics 22(9), 1152--1153, 2006.
      </p>
      <p><strong>See Also:</strong>
        Graph.motifs_randesu()
      </p>
</div></td></tr></table>
</div>
<a name="neighborhood"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">neighborhood</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">order</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.neighborhood">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>For each vertex specified by <i>vertices</i>, returns the vertices 
  reachable from that vertex in at most <i>order</i> steps.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a single vertex ID or a list of vertex IDs, or <code>None</code> 
          meaning all the vertices in the graph.</li>
        <li><strong class="pname"><code>mode</code></strong> - specifies how to take into account the direction of the edges if 
          a directed graph is analyzed. <code>&quot;out&quot;</code> means 
          that only the outgoing edges are followed, so all vertices 
          reachable from the source vertex in at most <i>order</i> steps 
          are counted. <code>&quot;in&quot;</code> means that only the 
          incoming edges are followed (in reverse direction of course), so 
          all vertices from which the source vertex is reachable in at most
          <i>order</i> steps are counted. <code>&quot;all&quot;</code> 
          treats directed edges as undirected.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a single list specifying the neighborhood if <i>vertices</i> was 
          an integer specifying a single vertex index, or a list of lists 
          if <i>vertices</i> was a list or <code>None</code>.</dd>
  </dl>
</td></tr></table>
</div>
<a name="neighborhood_size"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">neighborhood_size</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">order</span>=<span class="sig-default">1</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.neighborhood_size">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>For each vertex specified by <i>vertices</i>, returns the number of 
  vertices reachable from that vertex in at most <i>order</i> steps.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a single vertex ID or a list of vertex IDs, or <code>None</code> 
          meaning all the vertices in the graph.</li>
        <li><strong class="pname"><code>mode</code></strong> - specifies how to take into account the direction of the edges if 
          a directed graph is analyzed. <code>&quot;out&quot;</code> means 
          that only the outgoing edges are followed, so all vertices 
          reachable from the source vertex in at most <i>order</i> steps 
          are counted. <code>&quot;in&quot;</code> means that only the 
          incoming edges are followed (in reverse direction of course), so 
          all vertices from which the source vertex is reachable in at most
          <i>order</i> steps are counted. <code>&quot;all&quot;</code> 
          treats directed edges as undirected.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a single number specifying the neighborhood size if 
          <i>vertices</i> was an integer specifying a single vertex index, 
          or a list of sizes if <i>vertices</i> was a list or 
          <code>None</code>.</dd>
  </dl>
</td></tr></table>
</div>
<a name="neighbors"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">neighbors</span>(<span class="sig-arg">vertex</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.neighbors">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns adjacent vertices to a given vertex.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertex</code></strong> - a vertex ID</li>
        <li><strong class="pname"><code>mode</code></strong> - whether to return only predecessors (<a 
          href="igraph-module.html#OUT" class="link">OUT</a>), successors 
          (<a href="igraph-module.html#IN" class="link">IN</a>) or both (<a
          href="igraph-module.html#ALL" class="link">ALL</a>). Ignored for 
          undirected graphs.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="path_length_hist"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">path_length_hist</span>(<span class="sig-arg">directed</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.path_length_hist">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the path length histogram of the graph</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>directed</code></strong> - whether to consider directed paths</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a tuple. The first item of the tuple is a list of path lengths, 
          the <i class="math">i</i>th element of the list contains the 
          number of paths with length <i class="math">i+1</i>. The second 
          item contains the number of unconnected vertex pairs as a float 
          (since it might not fit into an integer)</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function is wrapped in a more convenient syntax in the derived
        class <a href="igraph.Graph-class.html" class="link">Graph</a>. It 
        is advised to use that instead of this version.
      </p>
</div></td></tr></table>
</div>
<a name="permute_vertices"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">permute_vertices</span>(<span class="sig-arg">permutation</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.permute_vertices">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Permutes the vertices of the graph according to the given permutation 
  and returns the new graph.</p>
  <p>Vertex <i class="math">k</i> of the original graph will become vertex 
  <i class="math">permutation[k]</i> in the new graph. No validity checks 
  are performed on the permutation vector.</p>
  <dl class="fields">
    <dt>Returns:</dt>
        <dd>the new graph</dd>
  </dl>
</td></tr></table>
</div>
<a name="personalized_pagerank"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">personalized_pagerank</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">directed</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">damping</span>=<span class="sig-default">0.85</span>,
        <span class="sig-arg">reset</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">reset_vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">arpack_options</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">implementation</span>=<span class="sig-default">&quot;prpack&quot;</span>,
        <span class="sig-arg">niter</span>=<span class="sig-default">1000</span>,
        <span class="sig-arg">eps</span>=<span class="sig-default">0.001</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.personalized_pagerank">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the personalized PageRank values of a graph.</p>
  <p>The personalized PageRank calculation is similar to the PageRank 
  calculation, but the random walk is reset to a non-uniform distribution 
  over the vertices in every step with probability <i 
  class="math">1-damping</i> instead of a uniform distribution.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the indices of the vertices being queried. <code>None</code> 
          means all of the vertices.</li>
        <li><strong class="pname"><code>directed</code></strong> - whether to consider directed paths.</li>
        <li><strong class="pname"><code>damping</code></strong> - the damping factor. <i class="math">1-damping</i> is the PageRank
          value for vertices with no incoming links.</li>
        <li><strong class="pname"><code>reset</code></strong> - the distribution over the vertices to be used when resetting the 
          random walk. Can be a sequence, an iterable or a vertex attribute
          name as long as they return a list of floats whose length is 
          equal to the number of vertices. If <code>None</code>, a uniform 
          distribution is assumed, which makes the method equivalent to the
          original PageRank algorithm.</li>
        <li><strong class="pname"><code>reset_vertices</code></strong> - an alternative way to specify the distribution over the vertices 
          to be used when resetting the random walk. Simply supply a list 
          of vertex IDs here, or a <a href="igraph.VertexSeq-class.html" 
          class="link">VertexSeq</a> or a <a 
          href="igraph.Vertex-class.html" class="link">Vertex</a>. 
          Resetting will take place using a uniform distribution over the 
          specified vertices.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
        <li><strong class="pname"><code>arpack_options</code></strong> - an <a href="igraph.ARPACKOptions-class.html" 
          class="link">ARPACKOptions</a> object used to fine-tune the 
          ARPACK eigenvector calculation. If omitted, the module-level 
          variable called <code>arpack_options</code> is used. This 
          argument is ignored if not the ARPACK implementation is used, see
          the <i>implementation</i> argument.</li>
        <li><strong class="pname"><code>implementation</code></strong> - which implementation to use to solve the PageRank eigenproblem. 
          Possible values are:
          <ul>
            <li>
              <code>&quot;prpack&quot;</code>: use the PRPACK library. This
              is a new implementation in igraph 0.7
            </li>
            <li>
              <code>&quot;arpack&quot;</code>: use the ARPACK library. This
              implementation was used from version 0.5, until version 0.7.
            </li>
            <li>
              <code>&quot;power&quot;</code>: use a simple power method. 
              This is the implementation that was used before igraph 
              version 0.5.
            </li>
          </ul></li>
        <li><strong class="pname"><code>niter</code></strong> - The number of iterations to use in the power method 
          implementation. It is ignored in the other implementations.</li>
        <li><strong class="pname"><code>eps</code></strong> - The power method implementation will consider the calculation as 
          complete if the difference of PageRank values between iterations 
          change less than this value for every node. It is ignored by the 
          other implementations.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a list with the personalized PageRank values of the specified 
          vertices.</dd>
  </dl>
</td></tr></table>
</div>
<a name="predecessors"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">predecessors</span>(<span class="sig-arg">vertex</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.predecessors">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the predecessors of a given vertex.</p>
  <p>Equivalent to calling the <a 
  href="igraph.GraphBase-class.html#neighbors" 
  class="link">Graph.neighbors</a> method with type=<a 
  href="igraph-module.html#IN" class="link">IN</a>.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="radius"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">radius</span>(<span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.radius">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the radius of the graph.</p>
  <p>The radius of a graph is defined as the minimum eccentricity of its 
  vertices (see <a href="igraph.GraphBase-class.html#eccentricity" 
  class="link">eccentricity()</a>).</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - what kind of paths to consider for the calculation in case of 
          directed graphs. <code>OUT</code> considers paths that follow 
          edge directions, <code>IN</code> considers paths that follow the 
          opposite edge directions, <code>ALL</code> ignores edge 
          directions. The argument is ignored for undirected graphs.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the radius@see: <a 
          href="igraph.GraphBase-class.html#eccentricity" 
          class="link">Graph.eccentricity()</a></dd>
  </dl>
</td></tr></table>
</div>
<a name="reciprocity"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">reciprocity</span>(<span class="sig-arg">ignore_loops</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">&quot;default&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.reciprocity">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Reciprocity defines the proportion of mutual connections in a directed
  graph. It is most commonly defined as the probability that the opposite 
  counterpart of a directed edge is also included in the graph. This 
  measure is calculated if <code>mode</code> is 
  <code>&quot;default&quot;</code>.</p>
  <p>Prior to igraph 0.6, another measure was implemented, defined as the 
  probability of mutual connection between a vertex pair if we know that 
  there is a (possibly non-mutual) connection between them. In other words,
  (unordered) vertex pairs are classified into three groups: (1) 
  disconnected, (2) non-reciprocally connected and (3) reciprocally 
  connected. The result is the size of group (3), divided by the sum of 
  sizes of groups (2) and (3). This measure is calculated if 
  <code>mode</code> is <code>&quot;ratio&quot;</code>.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>ignore_loops</code></strong> - whether loop edges should be ignored.</li>
        <li><strong class="pname"><code>mode</code></strong> - the algorithm to use to calculate the reciprocity; see above for 
          more details.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the reciprocity of the graph</dd>
  </dl>
</td></tr></table>
</div>
<a name="rewire"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">rewire</span>(<span class="sig-arg">n</span>=<span class="sig-default">1000</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">&quot;simple&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.rewire">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Randomly rewires the graph while preserving the degree 
  distribution.</p>
  <p>Please note that the rewiring is done &quot;in-place&quot;, so the 
  original graph will be modified. If you want to preserve the original 
  graph, use the <a href="igraph.GraphBase-class.html#copy" 
  class="link">copy</a> method before.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>n</code></strong> - the number of rewiring trials.</li>
        <li><strong class="pname"><code>mode</code></strong> - the rewiring algorithm to use. It can either be 
          <code>&quot;simple&quot;</code> or 
          <code>&quot;loops&quot;</code>; the former does not create or 
          destroy loop edges while the latter does.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="rewire_edges"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">rewire_edges</span>(<span class="sig-arg">prob</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">multiple</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.rewire_edges">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Rewires the edges of a graph with constant probability.</p>
  <p>Each endpoint of each edge of the graph will be rewired with a 
  constant probability, given in the first argument.</p>
  <p>Please note that the rewiring is done &quot;in-place&quot;, so the 
  original graph will be modified. If you want to preserve the original 
  graph, use the <a href="igraph.GraphBase-class.html#copy" 
  class="link">copy</a> method before.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>prob</code></strong> - rewiring probability</li>
        <li><strong class="pname"><code>loops</code></strong> - whether the algorithm is allowed to create loop edges</li>
        <li><strong class="pname"><code>multiple</code></strong> - whether the algorithm is allowed to create multiple edges.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="shortest_paths"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">shortest_paths</span>(<span class="sig-arg">source</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">target</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.shortest_paths_dijkstra">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates shortest path lengths for given vertices in a graph.</p>
  <p>The algorithm used for the calculations is selected automatically: a 
  simple BFS is used for unweighted graphs, Dijkstra's algorithm is used 
  when all the weights are positive. Otherwise, the Bellman-Ford algorithm 
  is used if the number of requested source vertices is larger than 100 and
  Johnson's algorithm is used otherwise.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - a list containing the source vertex IDs which should be included 
          in the result. If <code>None</code>, all vertices will be 
          considered.</li>
        <li><strong class="pname"><code>target</code></strong> - a list containing the target vertex IDs which should be included 
          in the result. If <code>None</code>, all vertices will be 
          considered.</li>
        <li><strong class="pname"><code>weights</code></strong> - a list containing the edge weights. It can also be an attribute 
          name (edge weights are retrieved from the given attribute) or 
          <code>None</code> (all edges have equal weight).</li>
        <li><strong class="pname"><code>mode</code></strong> - the type of shortest paths to be used for the calculation in 
          directed graphs. <a href="igraph-module.html#OUT" 
          class="link">OUT</a> means only outgoing, <a 
          href="igraph-module.html#IN" class="link">IN</a> means only 
          incoming paths. <a href="igraph-module.html#ALL" 
          class="link">ALL</a> means to consider the directed graph as an 
          undirected one.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the shortest path lengths for given vertices in a matrix</dd>
  </dl>
</td></tr></table>
</div>
<a name="similarity_dice"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">similarity_dice</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">pairs</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">IGRAPH_ALL</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.similarity_dice">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Dice similarity coefficient of vertices.</p>
  <p>The Dice similarity coefficient of two vertices is twice the number of
  their common neighbors divided by the sum of their degrees. This 
  coefficient is very similar to the Jaccard coefficient, but usually gives
  higher similarities than its counterpart.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices to be analysed. If <code>None</code> and 
          <i>pairs</i> is also <code>None</code>, all vertices will be 
          considered.</li>
        <li><strong class="pname"><code>pairs</code></strong> - the vertex pairs to be analysed. If this is given, 
          <i>vertices</i> must be <code>None</code>, and the similarity 
          values will be calculated only for the given pairs. Vertex pairs 
          must be specified as tuples of vertex IDs.</li>
        <li><strong class="pname"><code>mode</code></strong> - which neighbors should be considered for directed graphs. Can be 
          <a href="igraph-module.html#ALL" class="link">ALL</a>, <a 
          href="igraph-module.html#IN" class="link">IN</a> or <a 
          href="igraph-module.html#OUT" class="link">OUT</a>, ignored for 
          undirected graphs.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether vertices should be considered adjacent to themselves. 
          Setting this to <code>True</code> assumes a loop edge for all 
          vertices even if none is present in the graph. Setting this to 
          <code>False</code> may result in strange results: nonadjacent 
          vertices may have larger similarities compared to the case when 
          an edge is added between them -- however, this might be exactly 
          the result you want to get.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the pairwise similarity coefficients for the vertices specified, 
          in the form of a matrix if <code>pairs</code> is 
          <code>None</code> or in the form of a list if <code>pairs</code> 
          is not <code>None</code>.</dd>
  </dl>
</td></tr></table>
</div>
<a name="similarity_inverse_log_weighted"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">similarity_inverse_log_weighted</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">IGRAPH_ALL</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.similarity_inverse_log_weighted">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Inverse log-weighted similarity coefficient of vertices.</p>
  <p>Each vertex is assigned a weight which is 1 / log(degree). The 
  log-weighted similarity of two vertices is the sum of the weights of 
  their common neighbors.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices to be analysed. If <code>None</code>, all vertices 
          will be considered.</li>
        <li><strong class="pname"><code>mode</code></strong> - which neighbors should be considered for directed graphs. Can be 
          <a href="igraph-module.html#ALL" class="link">ALL</a>, <a 
          href="igraph-module.html#IN" class="link">IN</a> or <a 
          href="igraph-module.html#OUT" class="link">OUT</a>, ignored for 
          undirected graphs. <a href="igraph-module.html#IN" 
          class="link">IN</a> means that the weights are determined by the 
          out-degrees, <a href="igraph-module.html#OUT" 
          class="link">OUT</a> means that the weights are determined by the
          in-degrees.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the pairwise similarity coefficients for the vertices specified, 
          in the form of a matrix (list of lists).</dd>
  </dl>
</td></tr></table>
</div>
<a name="similarity_jaccard"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">similarity_jaccard</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">pairs</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">IGRAPH_ALL</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.similarity_jaccard">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Jaccard similarity coefficient of vertices.</p>
  <p>The Jaccard similarity coefficient of two vertices is the number of 
  their common neighbors divided by the number of vertices that are 
  adjacent to at least one of them.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - the vertices to be analysed. If <code>None</code> and 
          <i>pairs</i> is also <code>None</code>, all vertices will be 
          considered.</li>
        <li><strong class="pname"><code>pairs</code></strong> - the vertex pairs to be analysed. If this is given, 
          <i>vertices</i> must be <code>None</code>, and the similarity 
          values will be calculated only for the given pairs. Vertex pairs 
          must be specified as tuples of vertex IDs.</li>
        <li><strong class="pname"><code>mode</code></strong> - which neighbors should be considered for directed graphs. Can be 
          <a href="igraph-module.html#ALL" class="link">ALL</a>, <a 
          href="igraph-module.html#IN" class="link">IN</a> or <a 
          href="igraph-module.html#OUT" class="link">OUT</a>, ignored for 
          undirected graphs.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether vertices should be considered adjacent to themselves. 
          Setting this to <code>True</code> assumes a loop edge for all 
          vertices even if none is present in the graph. Setting this to 
          <code>False</code> may result in strange results: nonadjacent 
          vertices may have larger similarities compared to the case when 
          an edge is added between them -- however, this might be exactly 
          the result you want to get.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the pairwise similarity coefficients for the vertices specified, 
          in the form of a matrix if <code>pairs</code> is 
          <code>None</code> or in the form of a list if <code>pairs</code> 
          is not <code>None</code>.</dd>
  </dl>
</td></tr></table>
</div>
<a name="simplify"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">simplify</span>(<span class="sig-arg">multiple</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">combine_edges</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.simplify">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Simplifies a graph by removing self-loops and/or multiple edges.</p>
  <p>For example, suppose you have a graph with an edge attribute named 
  <code>weight</code>. <code>graph.simplify(combine_edges=max)</code> will 
  take the maximum of the weights of multiple edges and assign that weight 
  to the collapsed edge. <code>graph.simplify(combine_edges=sum)</code> 
  will take the sum of the weights. You can also write 
  <code>graph.simplify(combine_edges=dict(weight=&quot;sum&quot;))</code> 
  or <code>graph.simplify(combine_edges=dict(weight=sum))</code>, since 
  <code>sum</code> is recognised both as a Python built-in function and as 
  a string constant.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>multiple</code></strong> - whether to remove multiple edges.</li>
        <li><strong class="pname"><code>loops</code></strong> - whether to remove loops.</li>
        <li><strong class="pname"><code>combine_edges</code></strong> - specifies how to combine the attributes of multiple edges between
          the same pair of vertices into a single attribute. If it is 
          <code>None</code>, only one of the edges will be kept and all the
          attributes will be lost. If it is a function, the attributes of 
          multiple edges will be collected and passed on to that function 
          which will return the new attribute value that has to be assigned
          to the single collapsed edge. It can also be one of the following
          string constants:
          <ul>
            <li>
              <code>&quot;ignore&quot;</code>: all the edge attributes will
              be ignored.
            </li>
            <li>
              <code>&quot;sum&quot;</code>: the sum of the edge attribute 
              values will be used for the new edge.
            </li>
            <li>
              <code>&quot;product&quot;</code>: the product of the edge 
              attribute values will be used for the new edge.
            </li>
            <li>
              <code>&quot;mean&quot;</code>: the mean of the edge attribute
              values will be used for the new edge.
            </li>
            <li>
              <code>&quot;median&quot;</code>: the median of the edge 
              attribute values will be used for the new edge.
            </li>
            <li>
              <code>&quot;min&quot;</code>: the minimum of the edge 
              attribute values will be used for the new edge.
            </li>
            <li>
              <code>&quot;max&quot;</code>: the maximum of the edge 
              attribute values will be used for the new edge.
            </li>
            <li>
              <code>&quot;first&quot;</code>: the attribute value of the 
              first edge in the collapsed set will be used for the new 
              edge.
            </li>
            <li>
              <code>&quot;last&quot;</code>: the attribute value of the 
              last edge in the collapsed set will be used for the new edge.
            </li>
            <li>
              <code>&quot;random&quot;</code>: a randomly selected value 
              will be used for the new edge
            </li>
            <li>
              <code>&quot;concat&quot;</code>: the attribute values will be
              concatenated for the new edge.
            </li>
          </ul>
          <p>You can also use a dict mapping edge attribute names to 
          functions or the above string constants if you want to make the 
          behaviour of the simplification process depend on the name of the
          attribute. <code>None</code> is a special key in this dict, its 
          value will be used for all the attributes not specified 
          explicitly in the dictionary.</p></li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="st_mincut"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">st_mincut</span>(<span class="sig-arg">source</span>,
        <span class="sig-arg">target</span>,
        <span class="sig-arg">capacity</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.st_mincut">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the minimum cut between the source and target vertices in a
  graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex ID</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex ID</li>
        <li><strong class="pname"><code>capacity</code></strong> - the capacity of the edges. It must be a list or a valid attribute
          name or <code>None</code>. In the latter case, every edge will 
          have the same capacity.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the value of the minimum cut, the IDs of vertices in the first 
          and second partition, and the IDs of edges in the cut, packed in 
          a 4-tuple</dd>
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a list of <a href="igraph.cut.Cut-class.html" 
        class="link">Cut</a> objects. It is advised to use that.
      </p>
</div></td></tr></table>
</div>
<a name="strength"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">strength</span>(<span class="sig-arg">vertices</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>,
        <span class="sig-arg">loops</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.strength">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the strength (weighted degree) of some vertices from the 
  graph</p>
  <p>This method accepts a single vertex ID or a list of vertex IDs as a 
  parameter, and returns the strength (that is, the sum of the weights of 
  all incident edges) of the given vertices (in the form of a single 
  integer or a list, depending on the input parameter).</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a single vertex ID or a list of vertex IDs</li>
        <li><strong class="pname"><code>mode</code></strong> - the type of degree to be returned (<a 
          href="igraph-module.html#OUT" class="link">OUT</a> for 
          out-degrees, <a href="igraph-module.html#IN" class="link">IN</a> 
          IN for in-degrees or <a href="igraph-module.html#ALL" 
          class="link">ALL</a> for the sum of them).</li>
        <li><strong class="pname"><code>loops</code></strong> - whether self-loops should be counted.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="subcomponent"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">subcomponent</span>(<span class="sig-arg">v</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">ALL</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.subcomponent">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Determines the indices of vertices which are in the same component as 
  a given vertex.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>v</code></strong> - the index of the vertex used as the source/destination</li>
        <li><strong class="pname"><code>mode</code></strong> - if equals to <a href="igraph-module.html#IN" class="link">IN</a>,
          returns the vertex IDs from where the given vertex can be 
          reached. If equals to <a href="igraph-module.html#OUT" 
          class="link">OUT</a>, returns the vertex IDs which are reachable 
          from the given vertex. If equals to <a 
          href="igraph-module.html#ALL" class="link">ALL</a>, returns all 
          vertices within the same component as the given vertex, ignoring 
          edge directions. Note that this is not equal to calculating the 
          union of the results of <a href="igraph-module.html#IN" 
          class="link">IN</a> and <a href="igraph-module.html#OUT" 
          class="link">OUT</a>.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the indices of vertices which are in the same component as a 
          given vertex.</dd>
  </dl>
</td></tr></table>
</div>
<a name="subgraph_edges"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">subgraph_edges</span>(<span class="sig-arg">edges</span>,
        <span class="sig-arg">delete_vertices</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.subgraph_edges">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns a subgraph spanned by the given edges.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>edges</code></strong> - a list containing the edge IDs which should be included in the 
          result.</li>
        <li><strong class="pname"><code>delete_vertices</code></strong> - if <code>True</code>, vertices not incident on any of the 
          specified edges will be deleted from the result. If 
          <code>False</code>, all vertices will be kept.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the subgraph</dd>
  </dl>
</td></tr></table>
</div>
<a name="subisomorphic_lad"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">subisomorphic_lad</span>(<span class="sig-arg">other</span>,
        <span class="sig-arg">domains</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">induced</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">time_limit</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">return_mapping</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.subisomorphic_lad">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether a subgraph of the graph is isomorphic to another 
  graph.</p>
  <p>The optional <code>domains</code> argument may be used to restrict 
  vertices that may match each other. You can also specify whether you are 
  interested in induced subgraphs only or not.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the pattern graph we are looking for in the graph.</li>
        <li><strong class="pname"><code>domains</code></strong> - a list of lists, one sublist belonging to each vertex in the 
          template graph. Sublist <i class="math">i</i> contains the 
          indices of the vertices in the original graph that may match 
          vertex <i class="math">i</i> in the template graph. 
          <code>None</code> means that every vertex may match every other 
          vertex.</li>
        <li><strong class="pname"><code>induced</code></strong> - whether to consider induced subgraphs only.</li>
        <li><strong class="pname"><code>time_limit</code></strong> - an optimal time limit in seconds. Only the integral part of this 
          number is taken into account. If the time limit is exceeded, the 
          method will throw an exception.</li>
        <li><strong class="pname"><code>return_mapping</code></strong> - when <code>True</code>, the function will return a tuple, where 
          the first element is a boolean denoting whether a subisomorphism 
          has been found or not, and the second element describes the 
          mapping of the vertices from the template graph to the original 
          graph. When <code>False</code>, only the boolean is returned.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>if no mapping is calculated, the result is <code>True</code> if 
          the graph contains a subgraph that is isomorphic to the given 
          template, <code>False</code> otherwise. If the mapping is 
          calculated, the result is a tuple, the first element being the 
          above mentioned boolean, and the second element being the mapping
          from the target to the original graph.</dd>
  </dl>
</td></tr></table>
</div>
<a name="subisomorphic_vf2"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">subisomorphic_vf2</span>(<span class="sig-arg">other</span>,
        <span class="sig-arg">color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color1</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_color2</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">return_mapping_12</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">return_mapping_21</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">callback</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">node_compat_fn</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">edge_compat_fn</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.subisomorphic_vf2">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Checks whether a subgraph of the graph is isomorphic to another 
  graph.</p>
  <p>Vertex and edge colors may be used to restrict the isomorphisms, as 
  only vertices and edges with the same color will be allowed to match each
  other.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>other</code></strong> - the other graph with which we want to compare the graph.</li>
        <li><strong class="pname"><code>color1</code></strong> - optional vector storing the coloring of the vertices of the first
          graph. If <code>None</code>, all vertices have the same color.</li>
        <li><strong class="pname"><code>color2</code></strong> - optional vector storing the coloring of the vertices of the 
          second graph. If <code>None</code>, all vertices have the same 
          color.</li>
        <li><strong class="pname"><code>edge_color1</code></strong> - optional vector storing the coloring of the edges of the first 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>edge_color2</code></strong> - optional vector storing the coloring of the edges of the second 
          graph. If <code>None</code>, all edges have the same color.</li>
        <li><strong class="pname"><code>return_mapping_12</code></strong> - if <code>True</code>, calculates the mapping which maps the 
          vertices of the first graph to the second. The mapping can 
          contain -1 if a given node is not mapped.</li>
        <li><strong class="pname"><code>return_mapping_21</code></strong> - if <code>True</code>, calculates the mapping which maps the 
          vertices of the second graph to the first. The mapping can 
          contain -1 if a given node is not mapped.</li>
        <li><strong class="pname"><code>callback</code></strong> - if not <code>None</code>, the subisomorphism search will not stop
          at the first match; it will call this callback function instead 
          for every subisomorphism found. The callback function must accept
          four arguments: the first graph, the second graph, a mapping from
          the nodes of the first graph to the second, and a mapping from 
          the nodes of the second graph to the first. The function must 
          return <code>True</code> if the search should continue or 
          <code>False</code> otherwise.</li>
        <li><strong class="pname"><code>node_compat_fn</code></strong> - a function that receives the two graphs and two node indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the nodes given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on node-specific criteria that are too 
          complicated to be represented by node color vectors (i.e. the 
          <code>color1</code> and <code>color2</code> parameters). 
          <code>None</code> means that every node is compatible with every 
          other node.</li>
        <li><strong class="pname"><code>edge_compat_fn</code></strong> - a function that receives the two graphs and two edge indices (one
          from the first graph, one from the second graph) and returns 
          <code>True</code> if the edges given by the two indices are 
          compatible (i.e. they could be matched to each other) or 
          <code>False</code> otherwise. This can be used to restrict the 
          set of isomorphisms based on edge-specific criteria that are too 
          complicated to be represented by edge color vectors (i.e. the 
          <code>edge_color1</code> and <code>edge_color2</code> 
          parameters). <code>None</code> means that every edge is 
          compatible with every other node.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>if no mapping is calculated, the result is <code>True</code> if 
          the graph contains a subgraph that's isomorphic to the given one,
          <code>False</code> otherwise. If any or both mappings are 
          calculated, the result is a 3-tuple, the first element being the 
          above mentioned boolean, the second element being the 1 -&gt; 2 
          mapping and the third element being the 2 -&gt; 1 mapping. If the
          corresponding mapping was not calculated, <code>None</code> is 
          returned in the appropriate element of the 3-tuple.</dd>
  </dl>
</td></tr></table>
</div>
<a name="successors"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">successors</span>(<span class="sig-arg">vertex</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.successors">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Returns the successors of a given vertex.</p>
  <p>Equivalent to calling the <a 
  href="igraph.GraphBase-class.html#neighbors" 
  class="link">Graph.neighbors</a> method with type=<a 
  href="igraph-module.html#OUT" class="link">OUT</a>.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="to_directed"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">to_directed</span>(<span class="sig-arg">mutual</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.to_directed">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Converts an undirected graph to directed.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mutual</code></strong> - <code>True</code> if mutual directed edges should be created for 
          every undirected edge. If <code>False</code>, a directed edge 
          with arbitrary direction is created.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="to_undirected"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">to_undirected</span>(<span class="sig-arg">mode</span>=<span class="sig-default">&quot;collapse&quot;</span>,
        <span class="sig-arg">combine_edges</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.to_undirected">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Converts a directed graph to undirected.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - specifies what to do with multiple directed edges going between 
          the same vertex pair. <code>True</code> or 
          <code>&quot;collapse&quot;</code> means that only a single edge 
          should be created from multiple directed edges. 
          <code>False</code> or <code>&quot;each&quot;</code> means that 
          every edge will be kept (with the arrowheads removed). 
          <code>&quot;mutual&quot;</code> creates one undirected edge for 
          each mutual directed edge pair.</li>
        <li><strong class="pname"><code>combine_edges</code></strong> - specifies how to combine the attributes of multiple edges between
          the same pair of vertices into a single attribute. See <a 
          href="igraph.GraphBase-class.html#simplify" 
          class="link">Graph.simplify()</a> for more details.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="topological_sorting"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">topological_sorting</span>(<span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.topological_sorting">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates a possible topological sorting of the graph.</p>
  <p>Returns a partial sorting and issues a warning if the graph is not a 
  directed acyclic graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - if <a href="igraph-module.html#OUT" class="link">OUT</a>, 
          vertices are returned according to the forward topological order 
          -- all vertices come before their successors. If <a 
          href="igraph-module.html#IN" class="link">IN</a>, all vertices 
          come before their ancestors.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>a possible topological ordering as a list</dd>
  </dl>
</td></tr></table>
</div>
<a name="transitivity_avglocal_undirected"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">transitivity_avglocal_undirected</span>(<span class="sig-arg">mode</span>=<span class="sig-default">&quot;nan&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.transitivity_avglocal_undirected">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the average of the vertex transitivities of the graph.</p>
  <p>The transitivity measures the probability that two neighbors of a 
  vertex are connected. In case of the average local transitivity, this 
  probability is calculated for each vertex and then the average is taken. 
  Vertices with less than two neighbors require special treatment, they 
  will either be left out from the calculation or they will be considered 
  as having zero transitivity, depending on the <i>mode</i> parameter.</p>
  <p>Note that this measure is different from the global transitivity 
  measure (see <a 
  href="igraph.GraphBase-class.html#transitivity_undirected" 
  class="link">transitivity_undirected()</a>) as it simply takes the 
  average local transitivity across the whole network.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - defines how to treat vertices with degree less than two. If 
          <code>TRANSITIVITT_ZERO</code> or <code>&quot;zero&quot;</code>, 
          these vertices will have zero transitivity. If 
          <code>TRANSITIVITY_NAN</code> or <code>&quot;nan&quot;</code>, 
          these vertices will be excluded from the average.</li>
    </ul></dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.GraphBase-class.html#transitivity_undirected" 
        class="link">transitivity_undirected()</a>, <a 
        href="igraph.GraphBase-class.html#transitivity_local_undirected" 
        class="link">transitivity_local_undirected()</a>
      </p>
      <p><strong>Reference:</strong>
        D. J. Watts and S. Strogatz: <i>Collective dynamics of small-world 
        networks</i>. Nature 393(6884):440-442, 1998.
      </p>
</div></td></tr></table>
</div>
<a name="transitivity_local_undirected"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">transitivity_local_undirected</span>(<span class="sig-arg">vertices</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">&quot;nan&quot;</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.transitivity_local_undirected">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the local transitivity (clustering coefficient) of the 
  given vertices in the graph.</p>
  <p>The transitivity measures the probability that two neighbors of a 
  vertex are connected. In case of the local transitivity, this probability
  is calculated separately for each vertex.</p>
  <p>Note that this measure is different from the global transitivity 
  measure (see <a 
  href="igraph.GraphBase-class.html#transitivity_undirected" 
  class="link">transitivity_undirected()</a>) as it calculates a 
  transitivity value for each vertex individually.</p>
  <p>The traditional local transitivity measure applies for unweighted 
  graphs only. When the <code>weights</code> argument is given, this 
  function calculates the weighted local transitivity proposed by Barrat et
  al (see references).</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>vertices</code></strong> - a list containing the vertex IDs which should be included in the 
          result. <code>None</code> means all of the vertices.</li>
        <li><strong class="pname"><code>mode</code></strong> - defines how to treat vertices with degree less than two. If 
          <code>TRANSITIVITT_ZERO</code> or <code>&quot;zero&quot;</code>, 
          these vertices will have zero transitivity. If 
          <code>TRANSITIVITY_NAN</code> or <code>&quot;nan&quot;</code>, 
          these vertices will have <code>NaN</code> (not a number) as their
          transitivity.</li>
        <li><strong class="pname"><code>weights</code></strong> - edge weights to be used. Can be a sequence or iterable or even an
          edge attribute name.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the transitivities for the given vertices in a list</dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.GraphBase-class.html#transitivity_undirected" 
        class="link">transitivity_undirected()</a>, <a 
        href="igraph.GraphBase-class.html#transitivity_avglocal_undirected"
        class="link">transitivity_avglocal_undirected()</a>
      </p>
      <strong>Reference:</strong>
      <ul class="nomargin-top">
        <li>
        Watts DJ and Strogatz S: <i>Collective dynamics of small-world 
        networks</i>. Nature 393(6884):440-442, 1998.
        </li>
        <li>
        Barrat A, Barthelemy M, Pastor-Satorras R and Vespignani A: <i>The 
        architecture of complex weighted networks</i>. PNAS 101, 3747 
        (2004). <a href="http://arxiv.org/abs/cond-mat/0311416" 
        target="_top">http://arxiv.org/abs/cond-mat/0311416</a>.
        </li>
      </ul>
</div></td></tr></table>
</div>
<a name="transitivity_undirected"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">transitivity_undirected</span>(<span class="sig-arg">mode</span>=<span class="sig-default">&quot;nan&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.transitivity_undirected">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the global transitivity (clustering coefficient) of the 
  graph.</p>
  <p>The transitivity measures the probability that two neighbors of a 
  vertex are connected. More precisely, this is the ratio of the triangles 
  and connected triplets in the graph. The result is a single real number. 
  Directed graphs are considered as undirected ones.</p>
  <p>Note that this measure is different from the local transitivity 
  measure (see <a 
  href="igraph.GraphBase-class.html#transitivity_local_undirected" 
  class="link">transitivity_local_undirected()</a>) as it calculates a 
  single value for the whole graph.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>mode</code></strong> - if <code>TRANSITIVITY_ZERO</code> or 
          <code>&quot;zero&quot;</code>, the result will be zero if the 
          graph does not have any triplets. If <code>&quot;nan&quot;</code>
          or <code>TRANSITIVITY_NAN</code>, the result will be 
          <code>NaN</code> (not a number).</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the transitivity</dd>
  </dl>
<div class="fields">      <p><strong>See Also:</strong>
        <a href="igraph.GraphBase-class.html#transitivity_local_undirected"
        class="link">transitivity_local_undirected()</a>, <a 
        href="igraph.GraphBase-class.html#transitivity_avglocal_undirected"
        class="link">transitivity_avglocal_undirected()</a>
      </p>
      <p><strong>Reference:</strong>
        S. Wasserman and K. Faust: <i>Social Network Analysis: Methods and 
        Applications</i>. Cambridge: Cambridge University Press, 1994.
      </p>
</div></td></tr></table>
</div>
<a name="triad_census"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">triad_census</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.triad_census">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Triad census, as defined by Davis and Leinhardt</p>
  <p>Calculating the triad census means classifying every triplets of 
  vertices in a directed graph. A triplet can be in one of 16 states, these
  are listed in the documentation of the C interface of igraph.</p>
  <dl class="fields">
  </dl>
<div class="fields">      <p><strong>Attention:</strong>
        this function has a more convenient interface in class <a 
        href="igraph.Graph-class.html" class="link">Graph</a> which wraps 
        the result in a <a href="igraph.datatypes.TriadCensus-class.html" 
        class="link">TriadCensus</a> object. It is advised to use that. The
        name of the triplet classes are also documented there.
      </p>
</div></td></tr></table>
</div>
<a name="unfold_tree"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">unfold_tree</span>(<span class="sig-arg">sources</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mode</span>=<span class="sig-default">OUT</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.unfold_tree">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Unfolds the graph using a BFS to a tree by duplicating vertices as 
  necessary.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>sources</code></strong> - the source vertices to start the unfolding from. It should be a 
          list of vertex indices, preferably one vertex from each connected
          component. You can use <a 
          href="igraph.GraphBase-class.html#topological_sorting" 
          class="link">Graph.topological_sorting()</a> to determine a 
          suitable set. A single vertex index is also accepted.</li>
        <li><strong class="pname"><code>mode</code></strong> - which edges to follow during the BFS. <code>OUT</code> follows 
          outgoing edges, <code>IN</code> follows incoming edges, 
          <code>ALL</code> follows both. Ignored for undirected graphs.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the unfolded tree graph and a mapping from the new vertex indices
          to the old ones.</dd>
  </dl>
</td></tr></table>
</div>
<a name="union"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">union</span>(<span class="sig-arg">graphs</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.union">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Creates the union of two (or more) graphs.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>graphs</code></strong> - the list of graphs to be united with the current one.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="vcount"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">vcount</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.vcount">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Counts the number of vertices.</p>
  <dl class="fields">
    <dt>Returns: integer</dt>
        <dd>the number of vertices in the graph.</dd>
  </dl>
</td></tr></table>
</div>
<a name="vertex_attributes"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">vertex_attributes</span>()</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.vertex_attributes">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  
  <dl class="fields">
    <dt>Returns:</dt>
        <dd>the attribute name list of the graph's vertices</dd>
  </dl>
</td></tr></table>
</div>
<a name="vertex_connectivity"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">vertex_connectivity</span>(<span class="sig-arg">source</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">target</span>=<span class="sig-default">-1</span>,
        <span class="sig-arg">checks</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">neighbors</span>=<span class="sig-default">&quot;error&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#Graph.cohesion">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculates the vertex connectivity of the graph or between some 
  vertices.</p>
  <p>The vertex connectivity between two given vertices is the number of 
  vertices that have to be removed in order to disconnect the two vertices 
  into two separate components. This is also the number of vertex disjoint 
  directed paths between the vertices (apart from the source and target 
  vertices of course). The vertex connectivity of the graph is the minimal 
  vertex connectivity over all vertex pairs.</p>
  <p>This method calculates the vertex connectivity of a given vertex pair 
  if both the source and target vertices are given. If none of them is 
  given (or they are both negative), the overall vertex connectivity is 
  returned.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>source</code></strong> - the source vertex involved in the calculation.</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex involved in the calculation.</li>
        <li><strong class="pname"><code>checks</code></strong> - if the whole graph connectivity is calculated and this is 
          <code>True</code>, igraph performs some basic checks before 
          calculation. If the graph is not strongly connected, then the 
          connectivity is obviously zero. If the minimum degree is one, 
          then the connectivity is also one. These simple checks are much 
          faster than checking the entire graph, therefore it is advised to
          set this to <code>True</code>. The parameter is ignored if the 
          connectivity between two given vertices is computed.</li>
        <li><strong class="pname"><code>neighbors</code></strong> - tells igraph what to do when the two vertices are connected. 
          <code>&quot;error&quot;</code> raises an exception, 
          <code>&quot;infinity&quot;</code> returns infinity, 
          <code>&quot;ignore&quot;</code> ignores the edge.</li>
    </ul></dd>
    <dt>Returns:</dt>
        <dd>the vertex connectivity</dd>
  </dl>
</td></tr></table>
</div>
<a name="write_dimacs"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_dimacs</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">source</span>,
        <span class="sig-arg">target</span>,
        <span class="sig-arg">capacity</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_dimacs">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the graph in DIMACS format to the given file.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
        <li><strong class="pname"><code>source</code></strong> - the source vertex ID</li>
        <li><strong class="pname"><code>target</code></strong> - the target vertex ID</li>
        <li><strong class="pname"><code>capacity</code></strong> - the capacities of the edges in a list. If it is not a list, the 
          corresponding edge attribute will be used to retrieve capacities.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="write_dot"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_dot</span>(<span class="sig-arg">f</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_dot">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the graph in DOT format to the given file.</p>
  <p>DOT is the format used by the <a href="http://www.graphviz.org" 
  target="_top">GraphViz</a> software package.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="write_edgelist"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_edgelist</span>(<span class="sig-arg">f</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_edgelist">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the edge list of a graph to a file.</p>
  <p>Directed edges are written in (from, to) order.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="write_gml"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_gml</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">creator</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">ids</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_gml">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the graph in GML format to the given file.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
        <li><strong class="pname"><code>creator</code></strong> - optional creator information to be written to the file. If 
          <code>None</code>, the current date and time is added.</li>
        <li><strong class="pname"><code>ids</code></strong> - optional numeric vertex IDs to use in the file. This must be a 
          list of integers or <code>None</code>. If <code>None</code>, the 
          <code>id</code> attribute of the vertices are used, or if they 
          don't exist, numeric vertex IDs will be generated automatically.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="write_graphml"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_graphml</span>(<span class="sig-arg">f</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_graphml">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the graph to a GraphML file.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="write_leda"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_leda</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">names</span>=<span class="sig-default">&quot;name&quot;</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">&quot;weights&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_leda">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the graph to a file in LEDA native format.</p>
  <p>The LEDA format supports at most one attribute per vertex and edge. 
  You can specify which vertex and edge attribute you want to use. Note 
  that the name of the attribute is not saved in the LEDA file.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
        <li><strong class="pname"><code>names</code></strong> - the name of the vertex attribute to be stored along with the 
          vertices. It is usually used to store the vertex names (hence the
          name of the keyword argument), but you may also use a numeric 
          attribute. If you don't want to store any vertex attributes, 
          supply <code>None</code> here.</li>
        <li><strong class="pname"><code>weights</code></strong> - the name of the edge attribute to be stored along with the edges.
          It is usually used to store the edge weights (hence the name of 
          the keyword argument), but you may also use a string attribute. 
          If you don't want to store any edge attributes, supply 
          <code>None</code> here.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="write_lgl"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_lgl</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">names</span>=<span class="sig-default">&quot;name&quot;</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">&quot;weights&quot;</span>,
        <span class="sig-arg">isolates</span>=<span class="sig-default">True</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_lgl">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the edge list of a graph to a file in .lgl format.</p>
  <p>Note that multiple edges and/or loops break the LGL software, but 
  igraph does not check for this condition. Unless you know that the graph 
  does not have multiple edges and/or loops, it is wise to call <a 
  href="igraph.GraphBase-class.html#simplify" class="link">simplify()</a> 
  before saving.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
        <li><strong class="pname"><code>names</code></strong> - the name of the vertex attribute containing the name of the 
          vertices. If you don't want to store vertex names, supply 
          <code>None</code> here.</li>
        <li><strong class="pname"><code>weights</code></strong> - the name of the edge attribute containing the weight of the 
          vertices. If you don't want to store weights, supply 
          <code>None</code> here.</li>
        <li><strong class="pname"><code>isolates</code></strong> - whether to include isolated vertices in the output.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="write_ncol"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_ncol</span>(<span class="sig-arg">f</span>,
        <span class="sig-arg">names</span>=<span class="sig-default">&quot;name&quot;</span>,
        <span class="sig-arg">weights</span>=<span class="sig-default">&quot;weights&quot;</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_ncol">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the edge list of a graph to a file in .ncol format.</p>
  <p>Note that multiple edges and/or loops break the LGL software, but 
  igraph does not check for this condition. Unless you know that the graph 
  does not have multiple edges and/or loops, it is wise to call <a 
  href="igraph.GraphBase-class.html#simplify" class="link">simplify()</a> 
  before saving.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
        <li><strong class="pname"><code>names</code></strong> - the name of the vertex attribute containing the name of the 
          vertices. If you don't want to store vertex names, supply 
          <code>None</code> here.</li>
        <li><strong class="pname"><code>weights</code></strong> - the name of the edge attribute containing the weight of the 
          vertices. If you don't want to store weights, supply 
          <code>None</code> here.</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<a name="write_pajek"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">write_pajek</span>(<span class="sig-arg">f</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="igraph-pysrc.html#GraphBase.write_pajek">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Writes the graph in Pajek format to the given file.</p>
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>f</code></strong> - the name of the file to be written or a Python file handle</li>
    </ul></dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="epynavbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="igraph-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="epynavbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="epynavbar" align="center"
            ></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Thu Apr 24 11:45:15 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</div>



  </div>
  </div>
</div>
</div>


    </div>

    <div id="footer"><div class="container">
      <div class="row">
        <div class="col-xs-12">
          <p class="text-muted credit">
	    &copy; 2003 &ndash; 2015 The igraph core team.
            &bull; Code licensed under 
            <a href="http://www.gnu.org/licenses/gpl-2.0.html">
            GNU GPL 2</a> or later, documentation under 
            <a href="http://www.gnu.org/copyleft/fdl.html">
              GNU FDL.</a>
            <!-- &bull; <a href="">Download old releases</a> -->
            <!-- &bull; <a href="">Old documentation</a> -->
            <!-- &bull; <a href="">Authors</a> -->
            <!-- &bull; <a href="">Tools</a> -->
            <!-- &bull; <a href="">Papers citing igraph</a> -->
          </p>
        </div>
      </div>
    </div></div>

    <!-- Bootstrap core JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="/js/jquery-2.0.3.min.js"></script>
    <script src="/bootstrap/js/bootstrap.min.js"></script>
    <script src="/js/affix.js"></script>
    <script src="/js/tooltips.js"></script>
    <script src="/js/toggle.js"></script>

    

    <!-- Google analytics -->
    <script>
      (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
      })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

      ga('create', 'UA-1392972-5', 'igraph.org');
      ga('send', 'pageview');
    </script>

  </body>
</html>
